---
title: "Figure 1 Sketchpad"
author: "docmanny"
date: "2023-05-05"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


```{r library, include=F, echo=F, message=F, warning=F, error=F}
library(tidyverse)
library(ggtree)
library(ape)
library(tidytree)
library(MCMCtreeR)
library(ggpubr)
library(patchwork)
library(grid)
library(cowplot)
library(sf)
library(maps)
library(rnaturalearth)
library(rnaturalearthdata)
library(ggspatial)
library(ggchicklet)
library(gridGraphics)
library(grid)


## From: https://stackoverflow.com/questions/77188670/ggplot2-fix-absolute-size-of-ggplot-on-onscreen-device-rstudio-plots-pane
draw <- function(plot, x_in = 2, y_in = 2) {
  
  grid::grid.newpage()
  
  grid::rectGrob(gp = grid::gpar(fill = "gray")) |>
  grid::grid.draw()
  
  grid::viewport(width  = ggplot2::unit(x_in, "in"), 
                 height = ggplot2::unit(y_in, "in")) |>
    grid::pushViewport()
  
  ggplot2::ggplot_build(plot) |>
    ggplot2::ggplot_gtable() |>
    grid::grid.draw()
}

print.ggplot <- draw

```



# Candidate Subplots

## Figure 1A: State of the bat sequencing union


```{r, include=F, echo=F}
### Doesn't work
# knitr::read_chunk("ng_curves.Rmd", labels = c("assembly_stats", "metadata", "phylogeny", "setup phylo plus auNG", "phylo auNG circular map color", "with T2T line"))
# knitr::read_chunk("ng_curves.Rmd")
```

```{r, include=F, echo=F, results='hide', child="ng_curves.Rmd" }
```

```{r F1A prep}
tr.cancerSucceptibility <- treeio::read.beast('../output/Upham2019_newChiroptera-RICR-Data.nexus')
tr.cancerSucceptibility@data <- tr.cancerSucceptibility@data %>% 
  mutate_at(vars(-label.y, -node,-Ancestor),as.numeric)

lq <- read_tsv('../data/lifehistory/manny_mammal_agedata_harmonized.withLQ_hasAli_hasNTiles_withRICR.tsv') %>% 
  select(label, LQ, lifespan) %>% 
  filter(label %in% tr.genomes$tip.label)

node.chiroptera = MRCA(tr.cancerSucceptibility, 'Myotis_lucifugus', 'Pteropus_vampyrus')

tipnode.chiroptera = offspring(tr.cancerSucceptibility, node.chiroptera, type='tips')
tips.chiroptera = tr.cancerSucceptibility %>% as_tibble %>% filter(node %in% tipnode.chiroptera) %>% pull(label) %>% intersect(., auNG.fancy.color$label)
tips.other = tr.cancerSucceptibility %>% as_tibble %>% filter(label %in% c('Homo_sapiens', 'Mus_musculus','Bos_taurus')) %>% pull(label)

tr.genomes.lq <- tr.cancerSucceptibility %>% tidytree::keep.tip(c(tips.chiroptera, tips.other))
```


```{r F1A: LQ}
p.trNew.auNG.circ.altcolor.lq <-
  tr.genomes.lq %>% 
  ggtree(
    aes(color=LQ.median),
    layout = "fan", 
    open.angle=180, 
    # branch.length = 'none'
  ) +
  scale_color_viridis_c(
    'Longevity Quotient', option='D',na.value = 'black',
    guide=guide_colorbar(title.position='top', title.hjust = 0.5,
    title.theme = element_text(face='bold', hjust = 0.5, size=5))
  ) +
  ggnewscale::new_scale_color() + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      # color=NA,
      orientation = 'y',
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA,
      fontface='italic',
      size=5/.pt,
      # offset=0.2,
      label.padding = unit(rep(0,4), "pt"),
    ) + 
  # ) +
  scale_color_manual("Origin", values = species_color, guide="none") +
  scale_fill_manual("Origin", values = species_color, guide="none") #+
  # theme(plot.margin = unit(c(0.3,0.5,0,0.5),"in"))
```

```{r F1A: Life}

p.trNew.auNG.circ.altcolor.life <-
  tr.genomes.lq %>% 
  ggtree(
    aes(color=lnLife.median),
    layout = "fan", 
    open.angle=180, 
    branch.length = 'none'
  ) +
  scale_color_viridis_c(
    'Log Lifespan', option='D',na.value = 'black',
    # na.value="#fde725ff",
    guide=guide_colorbar(title.position='top', title.hjust = 0.5,
    title.theme = element_text(face='bold', hjust = 0.5, size=5))
  ) +
  ggnewscale::new_scale_color() + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      # color=NA,
      orientation = 'y',
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA,
      fontface='italic',
      size=5/.pt,
      # offset=0.2,
      label.padding = unit(rep(0,4), "pt"),
    ) + 
  # ) +
  scale_color_manual("Origin", values = species_color, guide="none") +
  scale_fill_manual("Origin", values = species_color, guide="none") #+
  # theme(plot.margin = unit(c(0.3,0.5,0,0.5),"in"))

```

```{r F1A: RICR}

p.trNew.auNG.circ.altcolor.ricr <-
  tr.genomes.lq %>% 
  ggtree(
    aes(color=log2CancerSucceptabilityChange),
    layout = "fan", 
    open.angle=180, 
    branch.length = 'none'
  ) +
  scale_color_viridis_c(
    'Relative Cancer Risk', option='D',na.value = 'black',
    guide=guide_colorbar(title.position='top', title.hjust = 0.5,
    title.theme = element_text(face='bold', hjust = 0.5, size=5))
  ) +
  ggnewscale::new_scale_color() + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      # color=NA,
      orientation = 'y',
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA,
      fontface='italic',
      size=5/.pt,
      # offset=0.2,
      label.padding = unit(rep(0,4), "pt"),
    ) + 
  # ) +
  scale_color_manual("Origin", values = species_color, guide="none") +
  scale_fill_manual("Origin", values = species_color, guide="none") #+
  # theme(plot.margin = unit(c(0.3,0.5,0,0.5),"in"))

```

```{r F1A}
t2t.base = p.trNew.auNG.circ.altcolor.lq$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.bartop = p.trNew.auNG.circ.altcolor.lq$layers[[3]]$data %>% filter(label == 'Homo_sapiens') %>% pull(new_xtmp) +0.6#+   p.trNew.auNG.circ.altcolor$layers[[3]]$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.start = min(p.trNew.auNG.circ.altcolor.lq$data$y)
t2t.end = max(p.trNew.auNG.circ.altcolor.lq$data$y)

f1a <- p.trNew.auNG.circ.altcolor.lq + 
  geom_segment(
    aes(
      x = t2t.base + t2t.bartop,
      xend = t2t.base + t2t.bartop,
      y = t2t.start, 
      yend = t2t.end
    ),
    linewidth=0.1, 
    # lty='dashed'
  ) + 
  theme_void() + 
  theme(
    text=element_text(
      family='Helvetica', 
      size=5
    ),
    plot.background = element_blank(),
    legend.background = element_blank(),
    legend.position = c(0.5,0.45),
    legend.key.width = unit(0.2, 'in'),
    legend.key.height = unit(0.1, 'in'),
    legend.direction = 'horizontal',
    legend.box.margin = margin(0,0,0,0, unit='in'),
    # legend.title = element_text(face='bold', hjust = 0.5, size=1)
  )

draw(f1a, x_in = 7, y_in = 4)

```
```{r}
f1a %>% 
  ggsave(plot = ., "../output/subfigs/phylo_auNG_circ_col.pdf")
f1a %>% 
  ggsave(plot = ., "../output/subfigs/phylo_auNG_circ_col.svg")
f1a %>% 
  ggsave(plot = ., "../output/subfigs/phylo_auNG_circ_col.png")
```


## Figure 1B: Time-calibrated Myotis phylo

```{r F1B}
phy <- readMCMCtree('../data/timetree/Upham_NewBats/Runs_real_combined/FigTree.tre')


df.nodeages <- phy$nodeAges %>% 
  as.data.frame %>% 
  rownames_to_column('node') %>% 
  as_tibble() %>% 
  mutate_all(as.numeric)

df.phy <- phy$apePhy %>% 
  as_tibble()# %>% 
  # as.treedata()



trdat.phy <- full_join(df.phy, df.nodeages, by='node') %>% 
  as.treedata()
trdat.phy.onlyOurMyotis <- trdat.phy %>% 
  tidytree::drop.tip(trdat.phy@phylo$tip.label[!(trdat.phy@phylo$tip.label %in% c(our_genomes, "Myotis_brandtii", "Myotis_myotis"))])
  
phy.onlyOurMyotis <- trdat.phy.onlyOurMyotis@phylo
df.nodeages.onlyOurMyotis <- trdat.phy.onlyOurMyotis@data %>% 
  filter(!is.na(mean)) %>% 
  column_to_rownames('node') %>% 
  as.matrix()

test = list(
  "apePhy" = phy.onlyOurMyotis,
  "nodeAges" = df.nodeages.onlyOurMyotis
)

class(test) <- "MCMCtreer"

MCMC.tree.plot(
  test, cex.tips = 0.5, time.correction = 100, scale.res = c("Eon",
"Period"), plot.type = "phylogram", cex.age = 0.5, cex.labels = 0.5,lwd.bar = 2,
relative.height = 0.08, col.tree = "black", node.method = "bar", col.age = "navy",
no.margin = TRUE, label.offset = 0,
# pos.age = 11.2
)


p.f1b <- recordPlot()
plot.new() ## clean up device
p.f1b # redraw

## grab the scene as a grid object
grid.echo()
f1b <- grid.grab()

## draw it, changes optional
grid.newpage()
f1b <- editGrob(f1b, vp=viewport(width=unit(2.6,"in"), height = unit(2,"in")), gp=gpar(fontsize=10))
# grid.draw(f1b)
```

```{r pictures}
# pic.myoLuc <- image_read(path = '../data/img/batCon-myoLuc.jpg') %>% image_ggplot()
# pic.myoYum <- image_read("../data/img/batCon-myoYum.jpg") %>% image_ggplot()
# pic.myoCai <- image_read("../data/img/batCon-myoCai.jpg") %>% image_ggplot()
# pic.myoVol <- image_read("../data/img/batCon-myoVol.jpg") %>% image_ggplot()
# pic.myoEvo <- image_read("../data/img/batCon-myoEvo-Michael-DurhamMinden-PIctures.jpeg") %>% image_ggplot()
# pic.myoThy <- image_read("../data/img/batCon-myoThy.jpg") %>% image_ggplot()
# pic.myoOcc <- image_read("../data/img/batCon-myoOcc-Rick-and-Nora-Bowers-Alamy.jpeg") %>% image_ggplot()
# pic.myoVel <- image_read("../data/img/batCon-myoVel.jpg") %>% image_ggplot()
# pic.myoAui <- image_read("../data/img/batCon-myoAui.jpg") %>% image_ggplot()
grid.newpage()

f1b.pics <- 
  (pic.myoVol + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_volans'], linewidth=1))) /
  (pic.myoCai + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_californicus'], linewidth=1))) /
  (pic.myoAui + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_auriculus'], linewidth=1))) /
  (pic.myoEvo + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_evotis'], linewidth=1))) /
  (pic.myoThy + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_thysanodes'], linewidth=1))) /
  (pic.myoVel + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_velifer'], linewidth=1))) /
  (pic.myoYum + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_yumanensis'], linewidth=1))) /
  (pic.myoLuc + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_lucifugus'], linewidth=1))) /
  (pic.myoOcc + 
  theme(rect=element_rect(fill=NA, color=species_color['Myotis_occultus'], linewidth=1)))

f1b.pics <- patchworkGrob(f1b.pics)

f1b.pics <- editGrob(f1b.pics, vp=viewport(width=unit(1,"in"), height = unit(1.5,"in")), gp=gpar(margins=c(0,0,0,0)))

# grid.draw(f1b.pics)

```



## Figure 1C: Collection Sites

```{r, include=F, echo=F, results='hide', child="sample_map.Rmd" }
```

```{r F1C}

datasheet.known.noYum <- 
  datasheet.known %>% 
  filter(species != "Myotis yumanensis")

f1c <- ggplot() + 
  geom_sf(
    data= caaznv, 
    color = "#ABB0B8", 
    fill = "#6F7378", 
    size = 0.1
  ) +
  geom_spatial_point(
    data= datasheet.known.noYum %>% filter(!species %in% c('Myotis auriculus', 'Myotis volans')), 
    aes(
      x=Longitude, 
      y=Lattitude, 
      group=species, 
      color=species
  ), 
  size=2.5
  ) + 
  geom_spatial_point(
    data= datasheet.known.noYum %>% filter(species == 'Myotis auriculus'), 
    aes(
      x=Longitude, 
      y=Lattitude, 
      group=species, 
      color=species
  ), 
  size=2.5, 
  shape="\u25D7"
  ) + 
  geom_spatial_point(
    data= datasheet.known.noYum %>% filter(species == 'Myotis volans'), 
    aes(
      x=Longitude, 
      y=Lattitude, 
      group=species, 
      color=species
  ), 
  size=2.5, 
  shape='\u25D6',
  position=position_nudge(x=-0.4, y=0.02)
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis lucifugus'), 
    aes(label=species, group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=8.5,
    nudge_y=1
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis californicus'), 
    aes(label=species, group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=8,
    nudge_y=2
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis occultus'), 
    aes(label=species %>% str_pad(width = str_length('Myotis auriculus')-1,side = 'both'), 
        group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=-8.75,
    nudge_y=1.17
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis velifer'), 
    aes(label=species %>% str_pad(width = str_length('Myotis auriculus')+1,side = 'both'), 
        group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=-9.4,
    nudge_y=2.4
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis volans'), 
    aes(label=species %>% str_pad(width = str_length('Myotis auriculus'),side = 'both'), 
        group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, size=2, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=-11.4,
    nudge_y=1.3
  ) + 
  geom_spatial_label(
    data= datasheet.known.noYum %>% filter(species == 'Myotis auriculus'), 
    aes(label=species %>% str_pad(width = str_length('Myotis auriculus')), 
        group=species, y=Lattitude, x=Longitude, fill=species,
    ),
    show.legend=F, size=2, color='black', size=2, family='Helvetica', fontface='bold.italic',
    nudge_x=-11.5,
    nudge_y=-0.25
  ) + 
  scale_color_manual(
    values = species_color %>% 
      set_names(
        ., 
        names(.) %>% 
          str_replace_all("_", " ")
      )
  ) +
  scale_fill_manual(
    values = species_color %>% 
      set_names(., names(.) %>% str_replace_all("_", " "))
  ) +
  theme_pubr() + 
  labs_pubr() + 
  coord_sf(clip = 'off') + 
  theme(
    legend.position = "none",
    text=element_text(
      family='Helvetica', 
      size=5
    ),
    plot.background = element_blank(),
    plot.margin = margin(0,0.0,0,0.0,unit='in'),
  )

draw(f1c, x_in = 2, y_in = 2)
```

```{r}
p.map.collections.caaznv %>% 
  ggsave(plot = ., "../output/subfigs/Collection_map_CAAZNV.pdf")
p.map.collections.caaznv %>% 
  ggsave(plot = ., "../output/subfigs/Collection_map_CAAZNV.svg")
p.map.collections.caaznv %>% 
  ggsave(plot = ., "../output/subfigs/Collection_map_CAAZNV.png")
```

## Figure 1D: BUSCOs

```{r, include=F, echo=F, results='hide', child="annotations.Rmd" }
```

```{r F1D}
f1d <- busco %>% 
  filter(!name %in% c("N", "Complete")) %>% 
  filter(Genome != 'mMyoYum1') %>% 
  ggplot(
    aes(
      x = Genome,
      fill = name,
      y = value
    )
  ) + 
  geom_bar(position="stack", stat = "identity") + 
  geom_text(
    data = busco %>% filter(name == "Complete") %>% 
  filter(Genome != 'mMyoYum1'), 
    mapping = aes(label=value %>% str_c(., "%"), y=0),
    hjust=0,
    family='Helvetica',
    fontface='bold',
    size=2
  ) + 
  geom_errorbar(
    data = busco %>% filter(name == "Complete") %>% 
  filter(Genome != 'mMyoYum1'), 
    mapping = aes(
      x = Genome, 
      color=name,
      y=value,
      ymin=value,
      ymax=value
    ),
    stat= 'identity',
    fill=NA,
    inherit.aes = F,
    show.legend = F
  ) + 
  scale_fill_manual(
    "Category", values = col.busco,
    guide=guide_legend(
      direction = 'vertical',
      title.position='top', 
      title.vjust = 0.5,
      title.theme = element_text(
        face='bold', 
        hjust = 0.5, 
        size=5
      ),
      # nrow = 2,
      # ncol=3,
      label.position = 'left'
    )
    ) + 
  scale_y_continuous("Percentage", labels = scales::percent_format(scale = 1, accuracy=1)) + 
  coord_flip() + 
  theme_pubr() +
  labs_pubr() + 
  theme(
    text=element_text(
      family='Helvetica', 
      size=5
    ),
    plot.margin = margin(0,0.0,0,0.0,unit='in'),
    plot.background = element_blank(),
    legend.background = element_blank(),
    legend.position = 'left',
    # legend.position = c(0,-0.5),
    legend.key.width = unit(0.1, 'in'),
    legend.key.height = unit(0.1, 'in'),
    legend.margin = margin(0,0,0,0, unit='in'),
    legend.justification = 'right'
  )

f1d
```

```{r}
p.busco %>% 
  ggsave(plot = ., "../output/subfigs/BUSCOs.pdf")
p.busco %>% 
  ggsave(plot = ., "../output/subfigs/BUSCOs.svg")
p.busco %>% 
  ggsave(plot = ., "../output/subfigs/BUSCOs.png")
```

## Figure 1E: Chromosome completeness

```{r F1E}

completion_status = readxl::read_xlsx('../data/SupplementTable1_GenomeStats.xlsx', sheet='GapsAndScaffold')

chr_naming = tibble(
  Scaffold = c(1,     2,      3,      5,     6,     7,     8,     9,     10, 
               11,    12,     13,     14,    15,    16,    17,    18,    19,    20, 
               21,    22,     4,     23),
  chr      = c("1/2", "3/4", "5/6",   "7",   "8",   "9",   "10",  "11",  "12",
               "13",  "14",  "15/16", "17",  "18",  "19",  "20",  "21",  "22",  "23",
               "24",  "25",  "X",     "Y") %>% factor(., levels=.),
  size     = c(2,     2,      2,      1,     1,     1,     1,     1,     1,
               1,     1,      1,      1,     1,     1,     1,     1,     0.5,   0.5,
               0.5,   0.5,    2,      0.5)
)

pc.complete.chr <- completion_status %>% 
  filter(Genome != "mMyoYum1") %>% 
  left_join(chr_naming) %>% 
  group_by(chr, size) %>% 
  summarise(
    n = n(),
    n.Complete = sum(Complete == "Complete"),
    n.Fragmented = sum(Complete == "Frac"),
    n.Gap = sum(Complete == "Gap"),
    pc.Complete = n.Complete/n,
    pc.Fragmented = n.Fragmented/n,
    pc.Gap = n.Gap/n
  ) %>% 
  arrange(chr) %>% 
  ungroup %>% 
  group_by(chr, size, n) %>% 
  pivot_longer(c(n.Complete, n.Fragmented, n.Gap, pc.Complete, pc.Fragmented, pc.Gap))

f1e <- pc.complete.chr %>% 
  filter(
    str_starts(name, 'pc.')
  ) %>% 
  mutate(name=name %>% str_remove('pc.')) %>% 
  ggplot(
    aes(
      x = chr, 
      y = value*size, 
      fill=name
    )
  ) +
  geom_chicklet(width = 1) + 
  geom_text(
    data = . %>% filter(name == "Complete"),
    aes(
      y=size,
      label = value %>% scales::label_percent()(.)
    ),
    vjust=1,
    size=2,
    fontface='bold'
  ) +
  xlab("Chr") + 
  labs_pubr() + 
  theme_void() + 
  scale_y_continuous(
    # expand=c(0,0.1)
    expand=c(0,0)
  )+
  scale_fill_manual(
    "Status",
    labels = c(
      Complete = "T2T",
      Fragmented = "1T",
      Gap = "Near-T2T (1 gap)"
    ),
    values=c(
      Complete = '#332288',
      Fragmented = "#882255",
      Gap = "#FEA82F"
    )
  ) + 
  theme(
    axis.text.x = element_text(size=8, angle=-45, vjust=0),
    legend.position = 'bottom',
    legend.title = element_text(size=8),
    legend.text = element_text(size=5),
    axis.title.x = element_text(size=5, face='bold')
  )

pc.complete.genome <- completion_status %>% 
  filter(Genome != "mMyoYum1") %>% 
  group_by(Genome) %>% 
  summarise(
    n = n(),
    n.Complete = sum(Complete == "Complete"),
    n.Fragmented = sum(Complete == "Frac"),
    n.Gap = sum(Complete == "Gap"),
    pc.Complete = n.Complete/n,
    pc.Fragmented = n.Fragmented/n,
    pc.Gap = n.Gap/n
  ) %>% 
  pivot_longer(c(n.Complete, n.Fragmented, n.Gap, pc.Complete, pc.Fragmented, pc.Gap))

f1e2 <- pc.complete.genome %>% 
  filter(
    str_starts(name, 'pc.'),
    Genome != 'mMyoYum1'
  ) %>% 
  mutate(name=name %>% str_remove('pc.')) %>% 
  ggplot(
    aes(
      x = 0,
      y = value, 
      fill=name
    )
  ) +
  geom_bar(stat='identity') + 
  geom_text(
    aes(
      label = value %>% round(digits = 2) %>% scales::label_percent()(.)
    ),
    fontface='bold', 
    size=1,
    position = position_stack(vjust = 0.5),
  ) + 
  labs_pubr() + 
  theme_void() + 
  scale_fill_manual(
    "Status",
    values=c(
      Complete = '#332288',
      Fragmented = "#882255",
      Gap = "#FEA82F"
    )
  ) + 
  coord_polar(theta='y',clip = 'off') + 
  facet_wrap(~Genome, nrow = 1, ncol=8) + 
  theme(
    strip.text = element_text(size=5, face = 'bold'),
    legend.position = 'none'
  )
```


# Figure 1:

## Version 1: using grid graphics

```{r Figure 1 via Grid}

vp.f1 <- viewport(
  x=unit(0,'in'),
  y=unit(0,'in'),
  just = c("left", "bottom"),
  name = "Fig1",
  width = unit(5, 'in'),
  height = unit(4, 'in'),
)

f1.blank <- rectGrob(
  x=unit(0,'in'),
  y=unit(0,'in'),
  width = unit(5, 'in'),
  height = unit(4, 'in'),
  just = c("left", "bottom"),
  gp = gpar(color='white', fill='white')
)

inset.f1a <- viewport(
  name="f1a",
  x = unit(0, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(5, 'in'), 
  height = unit(4, 'in')
)



inset.f1b <- viewport(
  name="f1b",
  x = unit(0, 'in'), 
  y = unit(-0.15, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.1, 'in'), 
  height = unit(2.1, 'in')
)

inset.f1c <- viewport(
  name="f1c",
  x = unit(2.14, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.85, 'in'), 
  height = unit(1.8, 'in')
)


lab.A <- grid.text(
  name = 'labA',
    label= 'A',
    x = unit(0, 'in'), 
    y = unit(4, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.B <- grid.text(
  name = 'labB',
    label= 'B',
    x = unit(0, 'in'), 
    y = unit(1.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.C <- grid.text(
  name = 'labC',
    label= 'C',
    x = unit(3.1, 'in'), 
    y = unit(1.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

cairo_pdf('../output/draft_figs/Draft_Figure1v1.pdf',width = 5, height = 4, onefile = T)
# f1 <- grid.grabExpr({
grid.newpage()

vp_all <- vpTree(inset.f1a, vpList(inset.f1b, inset.f1c))
pushViewport(vp_all)
seekViewport('f1a')
grid.draw(f1.blank)
grid.draw(ggplotGrob(f1a+theme(plot.margin = margin(0.4,0.1,0.5,0,unit='in'))))

grid.draw(lab.A)

grid.draw(lab.C)

grid.draw(lab.D)

seekViewport('f1b')

grid.draw(ggplotGrob(f1c))

seekViewport('f1c')

grid.draw(ggplotGrob(f1d))

upViewport()

# })

# grid.newpage()
# grid.draw(f1)
dev.off()
```



## Version 2: with pseudo-ideogram

```{r Figure 1 with completeness stats}

vp.f1 <- viewport(
  x=unit(0,'in'),
  y=unit(0,'in'),
  just = c("left", "bottom"),
  name = "Fig1",
  width = unit(5, 'in'),
  height = unit(6, 'in'),
)

f1.blank <- rectGrob(
  x=unit(0,'in'),
  y=unit(0,'in'),
  width = unit(5, 'in'),
  height = unit(6, 'in'),
  just = c("left", "bottom"),
  gp = gpar(color='white', fill='white')
)

inset.f1a <- viewport(
  name="f1a",
  x = unit(0, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(5, 'in'), 
  height = unit(6, 'in')
)


inset.f1b <- viewport(
  name="f1b",
  x = unit(0, 'in'), 
  # y = unit(-0.15, 'in'), 
  y = unit(1.85, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.1, 'in'), 
  height = unit(2.1, 'in')
)

inset.f1c <- viewport(
  name="f1c",
  x = unit(2.14, 'in'), 
  y = unit(2, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.85, 'in'), 
  height = unit(1.8, 'in')
)

inset.f1d <- viewport(
  name="f1d",
  x = unit(0, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(5, 'in'), 
  height = unit(2, 'in')
)


inset.f1d2 <- viewport(
  name="f1d2",
  x = unit(1, 'in'), 
  y = unit(1.25, 'in'), 
  just = c("left", "bottom"),
  width = unit(3, 'in'), 
  height = unit(1, 'in')
)


lab.A <- grid.text(
  name = 'labA',
    label= 'A',
    x = unit(0, 'in'), 
    y = unit(6, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.B <- grid.text(
  name = 'labB',
    label= 'B',
    x = unit(0, 'in'), 
    y = unit(3.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.C <- grid.text(
  name = 'labC',
    label= 'C',
    x = unit(3.1, 'in'), 
    y = unit(3.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.D <- grid.text(
  name = 'labD',
    label= 'D',
    x = unit(0, 'in'), 
    y = unit(2, 'in'), 
    just = c("left", "bottom"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )


cairo_pdf('../output/draft_figs/Draft_Figure1.pdf',width = 5, height = 6, onefile = T)
# png('../output/draft_figs/Draft_Figure1.png',width = 5, height = 6, units = 'in', res = 300)
# f1 <- grid.grabExpr({
grid.newpage()

vp_all <- vpTree(inset.f1a, vpList(inset.f1b, inset.f1c, inset.f1d, inset.f1d2))
pushViewport(vp_all)
seekViewport('f1a')
grid.draw(f1.blank)
grid.draw(ggplotGrob(f1a+theme(plot.margin = margin(0.5,0.1,2.5,0.1,unit='in'))))

grid.draw(lab.A)

grid.draw(lab.B)

grid.draw(lab.C)

grid.draw(lab.D)

seekViewport('f1b')

grid.draw(ggplotGrob(f1c))

seekViewport('f1c')

grid.draw(ggplotGrob(f1d))

upViewport()

seekViewport('f1d')

grid.draw(ggplotGrob(f1e))

upViewport()

seekViewport('f1d2')

grid.draw(ggplotGrob(f1e2))

upViewport()

# })

# grid.newpage()
# grid.draw(f1)
dev.off()
```

## Version 3: with timetree and photos

```{r Figure 1 with completeness stats}
grid.newpage()
vp.f1 <- viewport(
  x=unit(0,'in'),
  y=unit(0,'in'),
  just = c("left", "bottom"),
  name = "Fig1",
  width = unit(6, 'in'),
  height = unit(6, 'in'),
)

f1.blank <- rectGrob(
  x=unit(0,'in'),
  y=unit(0,'in'),
  width = unit(6, 'in'),
  height = unit(6, 'in'),
  just = c("left", "bottom"),
  gp = gpar(color='white', fill='white')
)

inset.f1a <- viewport(
  name="f1a",
  x = unit(0, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(3, 'in'), 
  height = unit(6, 'in')
)


inset.f1b <- viewport(
  name="f1b",
  x = unit(3, 'in'), 
  y = unit(3.4, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.6, 'in'), 
  height = unit(2, 'in')
)

inset.f1bpics <- viewport(
  name="f1bpics",
  x = unit(5, 'in'), 
  y = unit(3.4, 'in'), 
  just = c("left", "bottom"),
  width = unit(1, 'in'), 
  height = unit(1.5, 'in')
)

inset.f1c <- viewport(
  name="f1c",
  x = unit(0, 'in'), 
  # y = unit(-0.15, 'in'), 
  y = unit(1.85, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.1, 'in'), 
  height = unit(2.1, 'in')
)

inset.f1d <- viewport(
  name="f1d",
  x = unit(2.14, 'in'), 
  y = unit(2, 'in'), 
  just = c("left", "bottom"),
  width = unit(2.85, 'in'), 
  height = unit(1.8, 'in')
)

inset.f1e <- viewport(
  name="f1e",
  x = unit(0, 'in'), 
  y = unit(0, 'in'), 
  just = c("left", "bottom"),
  width = unit(5, 'in'), 
  height = unit(2, 'in')
)


inset.f1e2 <- viewport(
  name="f1e2",
  x = unit(1, 'in'), 
  y = unit(1.25, 'in'), 
  just = c("left", "bottom"),
  width = unit(3, 'in'), 
  height = unit(1, 'in')
)


lab.A <- grid.text(
  name = 'labA',
    label= 'A',
    x = unit(0, 'in'), 
    y = unit(6, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.B <- grid.text(
  name = 'labB',
    label= 'B',
    x = unit(3, 'in'), 
    y = unit(6, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.C <- grid.text(
  name = 'labC',
    label= 'C',
    x = unit(0, 'in'), 
    y = unit(3.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.D <- grid.text(
  name = 'labD',
    label= 'D',
    x = unit(3.1, 'in'), 
    y = unit(3.95, 'in'), 
    just = c("left", "top"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )

lab.E <- grid.text(
  name = 'labE',
    label= 'E',
    x = unit(0, 'in'), 
    y = unit(2, 'in'), 
    just = c("left", "bottom"),
    gp = gpar(
      fontsize = 10,
      fontfamily = 'Helvetica',
      fontface = 'bold'
    )
  )


cairo_pdf('../output/draft_figs/Draft_Figure1v3.pdf',width = 6, height = 6, onefile = T)
# png('../output/draft_figs/Draft_Figure1.png',width = 5, height = 6, units = 'in', res = 300)
# f1 <- grid.grabExpr({
grid.newpage()

vp_all <- vpTree(inset.f1a, vpList(inset.f1b, inset.f1bpics, inset.f1c, inset.f1d, inset.f1e, inset.f1e2))
pushViewport(vp_all)
seekViewport('f1a')
grid.draw(f1.blank)
grid.draw(ggplotGrob(f1a+theme(plot.margin = margin(0.5,0.1,2.5,0.1,unit='in'))))

grid.draw(lab.A)

grid.draw(lab.B)

grid.draw(lab.C)

grid.draw(lab.D)

grid.draw(lab.E)


seekViewport('f1b')

grid.draw(f1b)

seekViewport('f1bpics')

grid.draw(f1b.pics)

seekViewport('f1c')

grid.draw(ggplotGrob(f1c))

seekViewport('f1d')

grid.draw(ggplotGrob(f1d))

upViewport()

seekViewport('f1e')

grid.draw(ggplotGrob(f1e))

upViewport()

seekViewport('f1e2')

grid.draw(ggplotGrob(f1e2))

upViewport()

# })

# grid.newpage()
# grid.draw(f1)
dev.off()
```
