---
title: "QC Plots"
author: "Juan Manuel Vazquez"
date: "6/27/2022"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE, warning=F, echo=F}
knitr::opts_chunk$set(echo = TRUE)
knitr::knit_hooks$set(crop = knitr::hook_pdfcrop)
knitr::opts_chunk$set(crop = TRUE)
library(tidyverse)
library(ape)
library(magrittr)
library(ggpubr)
library(patchwork)
library(ggtree)
library(tidytree)
library(ggtreeExtra)
library(ggtext)
library(ggsci)
library(gridExtra)
library(ragg)
library(ggchicklet)
# library(gghighlight)
# library(plotly)
```

## Assembly Stats

```{r}
our_genomes = c(
  "Myotis_auriculus",
  "Myotis_californicus",
  "Myotis_occultus",
  "Myotis_lucifugus",
  "Myotis_yumanensis",
  "Myotis_volans",
  "Myotis_velifer",
  "Myotis_evotis",
  "Myotis_thysanodes"
)
```

```{r assembly_stats, eval=F}
assembly_stats <- read_tsv("../data/assembly_stats/all_assembly_stats.contigs.tsv", col_names = c("genomeName", "stat", "v1", "v2")) %>% 
  mutate(contig = str_detect(genomeName, "contig")) %>% 
  filter(str_detect(genomeName, "M_|mmyo|\\.chr", negate=T)) %>%
  filter(str_detect(genomeName, "mMyo.+1(?!_unmasked)", negate=T)) %>%
  separate(genomeName, c("name", "scaffold"), sep="\\.") %>% 
  arrange(name, contig) %>% 
  select(-scaffold) %>% 
  mutate(name=name %>% str_remove("_unmasked"))
```

```{r assembly_stats from file}
assembly_stats <- read_tsv('../data/assembly_stats/all_assembly_stats.cleaned.tsv')
```


```{r metadata}
meta <- read_tsv("../data/chiropteraGenomes-20230427.tsv") %>% 
  mutate(`Organism Name` = `Organism Name` %>% str_replace_all(" ", "_"))

meta.origin <- meta %>% 
  select(name=`Organism Name`, Origin=Group, Citation) %>% 
  group_by(name) %>% 
  mutate(
    Origin = ifelse(Origin %in% c('Zoonomia', 'Bat1K', 'DNAZoo', 'CCGP'), Origin, 'Other')
  ) %>% 
  bind_rows(assembly_stats %>% filter(str_detect(name, "mMyo")) %>% select(name) %>% distinct %>% mutate(Origin="This Paper", Citation = "This Paper", name = name %>% str_remove("_unmasked")))

assembly_stats.meta <- assembly_stats %>% 
  left_join(meta.origin, by="name") %>% 
  replace_na(list(Origin="Other"))%>% mutate(
    species = name %>% 
      str_replace("mMyoAui.*", "Myotis_auriculus") %>% 
      str_replace("mMyoCai.*", "Myotis_californicus") %>% 
      str_replace("mMyoOcc.*", "Myotis_occultus") %>% 
      str_replace("mMyoLuc.*", "Myotis_lucifugus") %>% 
      str_replace("mMyoYum.*", "Myotis_yumanensis") %>% 
      str_replace("mMyoVol.*", "Myotis_volans") %>% 
      str_replace("mMyoVel.*", "Myotis_velifer") %>% 
      str_replace("mMyoEvo.*", "Myotis_evotis") %>% 
      str_replace("mMyoThy.*", "Myotis_thysanodes") %>% 
      str_replace("_T2T","") %>% 
      str_replace("myoLuc2", "Myotis_lucifugus")
  )

assembly_stats.meta[str_detect(assembly_stats.meta$name, "T2T"),"Origin"] <- "T2T"


```

### Table

```{r}
table.auNG <- assembly_stats.meta %>% 
  filter(stat=='auNG') %>% 
  arrange(v2) %>% 
  mutate(v2=scales::label_comma()(v2),
         species= species %>% str_replace_all('_', " ")) %>%
  select(-v1, -contig, -Citation, -name) %>% 
  select(Species=species, everything()) %>% 
  dplyr::rename(auNG='v2')
table.LG50 <- assembly_stats.meta %>% 
  filter(stat=='LGx', v1==50) %>% 
  arrange(v2) %>% 
  mutate(v2=scales::label_comma()(v2),
         species= species %>% str_replace_all('_', " ")) %>%
  select(-v1, -contig, -Citation, -name) %>% 
  select(Species=species, everything()) %>% 
  dplyr::rename(LG50='v2')
table.NG50 <- assembly_stats.meta %>% 
  filter(stat=='NGx', v1==50) %>% 
  arrange(desc(v2)) %>% 
  mutate(v2=scales::label_comma()(v2),
         species= species %>% str_replace_all('_', " ")) %>%
  select(-v1, -contig, -Citation, -name) %>% 
  select(Species=species, everything()) %>% 
  dplyr::rename(NG50='v2')

## TODO:
# table.NScaff <- assembly_stats.meta %>% 
#   filter(stat=='N', v1=='Scaffold') %>% 
#   arrange(desc(v2)) %>% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %>% str_replace_all('_', " ")) %>%
#   select(-v1, -contig, -Citation, -name) %>% 
#   select(species, everything()) %>% 
#   dplyr::rename(Scaffolds='v2')
# table.NGap <- assembly_stats.meta %>% 
#   filter(stat=='N', v1=='Gap') %>% 
#   arrange(desc(v2)) %>% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %>% str_replace_all('_', " ")) %>%
#   select(-v1, -contig, -Citation, -name) %>% 
#   select(species, everything()) %>% 
#   dplyr::rename(Gaps='v2')
# table.NST2T <- assembly_stats.meta %>% 
#   filter(stat=='N', v1=='Scaffolds_T2T') %>% 
#   arrange(desc(v2)) %>% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %>% str_replace_all('_', " ")) %>%
#   select(-v1, -contig, -Citation, -name) %>% 
#   select(species, everything()) %>% 
#   dplyr::rename(Scaffolds_T2T='v2')
# table.NCT2T <- assembly_stats.meta %>% 
#   arrange(desc(v2)) %>% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %>% str_replace_all('_', " ")) %>%
#   select(-v1, -contig, -Citation, -name) %>% 
#   select(species, everything()) %>% 
#   dplyr::rename(Complete_Scaffolds='v2')

table.stats <- table.auNG %>%
  # full_join(table.NScaff) %>% 
  # full_join(table.NGap) %>% 
  # full_join(table.NST2T) %>% 
  # full_join(table.NCT2T) %>% 
  full_join(table.NG50) %>% 
  full_join(table.LG50)

```


### Notes about the tree

In order to get TimeTree to play nicely with the phylogeny, a few changes were needed:
  - Murina aurata feae (replaced with Murina aurata)
  - Pteropus pselaphon (replaced with Pteropus mariannus)
  - Miniopterus schreibersii (replace with Miniopterus schreibersii orianae)
  - Myotis occultus (replaced with Myotis lucifugus lucifugus)
  - Myotis lucifugus (replaced with Myotis lucifugus carissima)
  - Hipposideros pendleburyi (replaced with Hipposideros turpis)
  - TimetreeV5 has split Eptesicus and broken its nomenclature/link to NCBI, so you can't find "Eptesicus fuscus" anymore.
    Instead, you have to look for all of Chiroptera or Vespertilliodinae and then look for Eptesicus fuscus hispaniolae.

If you do all these changes, then undo the changes in post, you get this tree. 
Note that it was very annoying to splice the two trees together by eye so avoid needing to redo this.

```{r phylogeny}
tr <- treeio::read.newick("../data/trees/TimeTree.org/genomeSpecies.timetree.nwk")

# setdiff(tr$tip.label, assembly_stats.meta$species)
# setdiff(assembly_stats.meta$species, tr$tip.label)
```

## NGx curves

```{r NGx}
plot.NGx <- assembly_stats.meta %>% 
  filter(stat == "NGx",
         contig==T
         ) %>% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x="NGx", y = "bp", title = "Contig Contiguity") + 
  geom_point() + 
  geom_line() + 
  ylim(1e4,NA) + 
  scale_color_brewer(palette = "Dark2") + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = "bottom")
plot.NGx
```

```{r NGx alt log}

col.source <- c(
  "This Paper" = "#FFBA08",
  Bat1K = "#B33951",
  T2T = "#d1b1c8",
  Zoonomia = "#4357AD",
  Other = "#6f7c12"
)

plot.NGx.altCol.log <- assembly_stats.meta %>% 
  mutate(Origin = Origin %>% factor(levels=names(col.source))) %>% 
  filter(stat == "NGx",
         contig==T
         ) %>% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x="NGx", y = "bp", title = "Contig Contiguity") + 
  # geom_point() + 
  geom_line(linewidth=0.5) + 
  scale_y_log10(lim=c(1e4,NA))+
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = "bottom")
plot.NGx.altCol.log
```

```{r NGx alt linear}
plot.NGx.altCol <- assembly_stats.meta %>% 
  mutate(Origin = Origin %>% factor(levels=names(col.source))) %>% 
  filter(stat == "NGx",
         contig==T
         ) %>% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x="NGx", y = "bp", title = "Contig Contiguity") + 
  # geom_point() + 
  geom_line(linewidth=0.5) + 
  ylim(1e4,NA) +
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = "bottom")
plot.NGx.altCol
```


```{r eval=F}
plot.NGx.altCol %>% ggsave(plot=., filename = "../output/NGx_curves.pdf", width = 6, height = 3.71, dpi=900, units = "in")
```

```{r onlyT2T}
assembly_stats.plotme <- assembly_stats.meta %>% 
  mutate(Origin = Origin %>% factor(levels=names(col.source))) %>% 
  filter(stat == "NGx",
         contig==T
         )
plot.NGx.altCol.t2tzoo <- assembly_stats.plotme %>% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x="NGx", y = "bp", title = "Contig Contiguity") + 
  # geom_point() + 
  geom_line(
    data=assembly_stats.plotme %>% filter(Origin %in% c("T2T", "Zoonomia")),
    linewidth=0.5
  ) + 
  ylim(1e4,NA) +
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = "bottom")
plot.NGx.altCol.t2tzoo
```


```{r eval=F}
plot.NGx.altCol.t2tzoo %>% ggsave(plot=., filename = "../output/NGx_curves_onlyT2TZoonomia.pdf", width = 6, height = 3.71, dpi=900, units = "in")

```

## auNG

```{r AuNG}
plot.auNG <- assembly_stats.meta %>% 
  filter(stat == "auNG",
         contig==T
         ) %>% 
  arrange(desc(v2)) %>% 
  mutate(name = factor(name, levels = name %>% unique)) %>% 
  ggplot(aes(x=v2, y=name, color=Origin)) + 
  labs(x="auNG", y = "Assembly", title = "auNG") + 
  geom_point() + 
  scale_color_brewer(palette = "Dark2") + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = "bottom")
plot.auNG
```

## Tree plus auNG

```{r setup phylo plus auNG}
auNG <- assembly_stats.meta %>% 
  filter(stat == "auNG",
         contig==T
         ) %>% 
  select(species, Origin, v2, v1) %>% 
  rename(label=species, auNG=v2, genomeSize=v1) %>% 
  filter(!(str_detect(label, "lucifugus") & Origin=="Other")) %>% 
  # a bug lead to an accidental dup when there was both a contig and a scaffold
  group_by(label, Origin) %>% 
  filter(auNG==max(auNG)) %>% 
  ungroup
  

color_scale_origin = RColorBrewer::brewer.pal(n=auNG %>% pull(Origin) %>% unique %>% length, name="Dark2") %>% set_names(., auNG %>% pull(Origin) %>% unique)

# tr.auNG <- full_join(tr,auNG)

## Sometimes I hate ggtree
nodes_ourgenomes <- tr %>% 
  as_tibble %>% 
  filter(label %in% our_genomes) %>% 
  select(label, node) %>% 
  deframe()
```


```{r phylo plus auNG}
p.tr.auNG <- tr %>% 
  ggtree() + 
  geom_tiplab(
    aes(
      subset = (node %in% nodes_ourgenomes)
    ),
    align = T,
    fontface="bold"
    ) +
  geom_tiplab(
    aes(
      subset = !(node %in% nodes_ourgenomes)
    ),
    align = T
    ) +
  xlim_tree(c(NA,250)) +
  geom_facet(
    panel = "auNG (bar)",
    data = auNG,
    geom = geom_col,
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = 'y',
    width = .6
  )+
  geom_facet(
    panel = "auNG (point)", 
    data = auNG, 
    geom = geom_point, 
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ), 
    size=2
  )+
  theme_tree2() +
  scale_color_manual("Origin", values = color_scale_origin) +
  scale_fill_manual("Origin", values = color_scale_origin) +
  # geom_facet(panel = "Genome Size", data = auNG %>% select(label, genomeSize, Origin), geom = geom_point, 
  #              mapping=aes(x = genomeSize, color=Origin))+ 
  theme(legend.position = "bottom")

p.tr.auNG
```

```{r auNG.fancy}
auNG.fancy <- auNG %>% 
  mutate(
    label_fancy = sapply(label, . %>% if_else(. %in% our_genomes, str_c("**",.,"**"), .)) %>% 
      str_remove("_mesoamericanus") %>% 
      str_replace_all("_", " ")
  )
```


```{r phylo auNG circular}
angle_rotate <- function(angle){
    # subset1 <- "(angle < 90 | angle > 270)"
    # subset2 <- "(angle >= 90 & angle <=270)"
    if (angle < 90 | angle > 270){
      return(angle)
    } else {
      return(angle+180)
    }
}
h_readjust <- function(angle){
    if (angle < 90 | angle > 270){
      return(0)
    } else {
      return(1)
    }
}

p.tr.auNG.circ <-
  tr %>%
  ggtree(layout = "fan", open.angle=180) +
  xlim_tree(c(NA,100))+ 
  geom_fruit(
    data = auNG.fancy,
    geom = geom_col,
    mapping=aes(
      y= label,
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = 'y',
    width = .6
  ) +
  geom_fruit(
    data = auNG.fancy,
    geom = geom_richtext, 
    mapping = aes(
      y=label,
      label=label_fancy,
      color=Origin,
      angle=sapply(angle,angle_rotate),
      hjust=sapply(angle,h_readjust)
    ),
    fill = NA, 
    label.color = NA, 
    label.padding = grid::unit(rep(0, 4), "pt"),
    
  ) + 
  scale_color_manual("Origin", values = color_scale_origin, guide="none") +
  scale_fill_manual("Origin", values = color_scale_origin,
                    guide = guide_legend(nrow = 2, ncol=6, title.position = "left")) +
  theme(legend.position = c(0.6,0.3), plot.margin = unit(c(4,0,0,0),"cm"))
  
p.tr.auNG.circ

```

```{r prep stuff}
species_color = ggsci::pal_d3(palette = "category20")(length(our_genomes)+1) %>% 
  set_names(., c(our_genomes, "Other"))

species_color["Myotis_evotis"] = "#17BECFFF"
species_color["Other"] = "#7F7F7FFF"

auNG.fancy.color <- auNG.fancy %>% 
  mutate(color_me = sapply(label, . %>% ifelse(. %in% our_genomes, ., "Other")))
```


```{r phylo auNG circular map color}
p.tr.auNG.circ.altcolor <-
  tr %>%
  ggtree(layout = "fan", open.angle=180) +
  xlim_tree(c(NA,100))+ 
  # geom_fruit_list(
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      orientation = 'y',
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA, 
      # offset=0.2,
      label.padding = grid::unit(rep(0,4), "pt"),
    ) + 
  # ) +
  scale_color_manual("Origin", values = species_color, guide="none") +
  scale_fill_manual("Origin", values = species_color, guide="none") +
  theme(plot.margin = unit(c(4,0,0,0),"cm"))
  
p.tr.auNG.circ.altcolor

```

```{r with T2T line}

t2t.base = p.tr.auNG.circ.altcolor$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.bartop = p.tr.auNG.circ.altcolor$layers[[4]]$data %>% filter(label == 'Homo_sapiens') %>% pull(new_xtmp) +
  p.tr.auNG.circ.altcolor$layers[[4]]$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.start = min(p.tr.auNG.circ.altcolor$data$y)
t2t.end = max(p.tr.auNG.circ.altcolor$data$y)

p.tr.auNG.circ.altcolor.withLine <- p.tr.auNG.circ.altcolor + 
  geom_segment(
    aes(
      x = t2t.base + t2t.bartop,
      xend = t2t.base + t2t.bartop,
      y = t2t.start, 
      yend = t2t.end
    )
  )
```

## Updated Phylo

```{r new phylo}
tr.536 <- read.tree('../data/trees/536Mammals-rooted-nodeNames.nwk')

tr.536$tip.label <- tr.536$tip.label %>% 
            str_replace('Canis_lupus_familiaris', 'Canis_familiaris') #%>% 
            # 

auNG.fancy.color$label <- auNG.fancy.color$label %>% 
  str_replace('Murina_aurata_feae', 'Murina_aurata') %>% 
  str_replace('Pteronotus_parnellii_mesoamericanus', 'Pteronotus_parnellii')
  
t.tr.536 <- tr.536 %>% as_tibble

node.chiroptera = t.tr.536%>% filter(label=='Chiroptera') %>% pull(node)

tipnodes.chiroptera = tidytree::offspring(
  tr.536, 
  .node=node.chiroptera, 
  type = 'tips'
)

tipnodes.chiroptera <- t.tr.536 %>% 
    filter(node %in% tipnodes.chiroptera) %>% 
    pull(label)


tips = c(
  # t.tr.536 %>% 
  #   filter(node %in% tipnodes.chiroptera) %>% 
  #   pull(label),
  tipnodes.chiroptera,
  "Homo_sapiens",
  "Mus_musculus",
  "Canis_familiaris",
  "Bos_taurus"
)

tr.genomes <- tr.536 %>% 
  keep.tip(
    tips
  )

```


```{r newPhylo}

p.trNew.auNG.circ.altcolor <-
  tr.genomes %>%
  ggtree(layout = "fan", open.angle=180, branch.length = 'none') +
  # xlim_tree(c(NA,10))+ 
  # geom_fruit_list(
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      orientation = 'y',
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA,
      fontface='italic',
      size=8/.pt,
      # offset=0.2,
      label.padding = grid::unit(rep(0,4), "pt"),
    ) + 
  # ) +
  scale_color_manual("Origin", values = species_color, guide="none") +
  scale_fill_manual("Origin", values = species_color, guide="none") +
  theme(plot.margin = unit(c(4,0,0,0),"cm"))

t2t.base = p.trNew.auNG.circ.altcolor$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.bartop = p.trNew.auNG.circ.altcolor$layers[[3]]$data %>% filter(label == 'Homo_sapiens') %>% pull(new_xtmp) +0.6#+   p.trNew.auNG.circ.altcolor$layers[[3]]$data %>% filter(label == 'Homo_sapiens') %>% pull(x)
t2t.start = min(p.trNew.auNG.circ.altcolor$data$y)
t2t.end = max(p.trNew.auNG.circ.altcolor$data$y)

p.trNew.auNG.circ.altcolor.withLine <- p.trNew.auNG.circ.altcolor + 
  geom_segment(
    aes(
      x = t2t.base + t2t.bartop,
      xend = t2t.base + t2t.bartop,
      y = t2t.start, 
      yend = t2t.end
    ),
    linewidth=0.1, 
    # lty='dashed'
  )

p.trNew.auNG.circ.altcolor.withLine
```

```{r}

# g <- ggplotGrob(p.trNew.auNG.circ.altcolor.withLine)

p.trNew.auNG.circ.altcolor.withLine %>% 
  ggsave(
    filename = '../output/subfigs/phylo_auNG_circ_col_withLine.pdf',
    plot= ., 
    width = 7.5, 
    height = 4, 
    units = 'in', 
    dpi = 300
  )
```

```{r}
completion_status = readxl::read_xlsx('../data/SupplementTable1_GenomeStats.xlsx', sheet='GapsAndScaffold')

completion_status %>% 
  select(Genome, Scaffold, Complete) %>% 
  pivot_wider(names_from=Scaffold, values_from=Complete)

order_genomes <- completion_status %>% 
  group_by(Genome) %>% 
  summarize(n.complete = sum(Complete == 'Complete')) %>% 
  arrange(desc(n.complete)) %>% 
  pull(Genome)

completion_status %>% 
  mutate(
    Complete=Complete %>% sapply(function(x){ifelse(x=='Frac', 'Frag', x)}),
    Genome = factor(Genome, levels=order_genomes)
  ) %>% 
  filter(Scaffold<=23) %>% 
  ggplot(
    aes(
      x=Scaffold,
      y=Genome,
      fill=Complete
    )
  ) + 
  geom_raster() + 
  labs(x='Chr', y='Genome') + 
  scale_fill_manual("Status", values=c(Gap='yellow', Frag='pink', Complete='green')) + 
  scale_x_continuous(
    breaks=1:23,
    labels = c(1:3, "X", 5:22, "Y")
  ) + 
  theme_pubr() +
  labs_pubr()
```

```{r}


chr_naming = tibble(
  Scaffold = c(1,     2,      3,      5,     6,     7,     8,     9,     10, 
               11,    12,     13,     14,    15,    16,    17,    18,    19,    20, 
               21,    22,     4,     23),
  chr      = c("1/2", "3/4", "5/6",   "7",   "8",   "9",   "10",  "11",  "12",
               "13",  "14",  "15/16", "17",  "18",  "19",  "20",  "21",  "22",  "23",
               "24",  "25",  "X",     "Y") %>% factor(., levels=.),
  size     = c(2,     2,      2,      1,     1,     1,     1,     1,     1,
               1,     1,      1,      1,     1,     1,     1,     1,     0.5,   0.5,
               0.5,   0.5,    2,      0.5)
)

pc.complete.chr <- completion_status %>% 
  filter(Genome != "mMyoYum1") %>% 
  left_join(chr_naming) %>% 
  group_by(chr, size) %>% 
  summarise(
    n = n(),
    n.Complete = sum(Complete == "Complete"),
    n.Fragmented = sum(Complete == "Frac"),
    n.Gap = sum(Complete == "Gap"),
    pc.Complete = n.Complete/n,
    pc.Fragmented = n.Fragmented/n,
    pc.Gap = n.Gap/n
  ) %>% 
  arrange(chr) %>% 
  ungroup %>% 
  group_by(chr, size, n) %>% 
  pivot_longer(c(n.Complete, n.Fragmented, n.Gap, pc.Complete, pc.Fragmented, pc.Gap))

pc.complete.chr %>% 
  filter(
    str_starts(name, 'pc.')
  ) %>% 
  mutate(name=name %>% str_remove('pc.')) %>% 
  ggplot(
    aes(
      x = chr, 
      y = value*size, 
      fill=name
    )
  ) +
  geom_chicklet(width = 1) + 
  geom_text(
    data = . %>% filter(name == "Complete"),
    aes(
      y=size,
      label = value %>% scales::label_percent()(.)
    ),
    vjust=1,
    fontface='bold'
  ) + 
  labs_pubr() + 
  theme_void() + 
  scale_y_continuous(
    # expand=c(0,0.1)
    expand=c(0,0)
  )+
  scale_fill_manual(
    "Status",
    values=c(
      Complete = '#0B5D1E',
      Fragmented = "#FF2E00",
      Gap = "#FEA82F"
    )
  ) + 
  theme(
    axis.text.x = element_text(size=12, angle=-45, vjust=0)
  )
  
```

```{r}
pc.complete.genome <- completion_status %>% 
  filter(Genome != "mMyoYum1") %>% 
  group_by(Genome) %>% 
  summarise(
    n = n(),
    n.Complete = sum(Complete == "Complete"),
    n.Fragmented = sum(Complete == "Frac"),
    n.Gap = sum(Complete == "Gap"),
    pc.Complete = n.Complete/n,
    pc.Fragmented = n.Fragmented/n,
    pc.Gap = n.Gap/n
  ) %>% 
  pivot_longer(c(n.Complete, n.Fragmented, n.Gap, pc.Complete, pc.Fragmented, pc.Gap))

pc.complete.genome %>% 
  filter(
    str_starts(name, 'pc.')
  ) %>% 
  mutate(name=name %>% str_remove('pc.')) %>% 
  # filter(Genome == 'mMyoLuc1') %>% 
  ggplot(
    aes(
      x = 0,
      y = value, 
      fill=name
    )
  ) +
  geom_bar(stat='identity') + 
  geom_text(
    aes(
      label = value %>% round(digits = 2) %>% scales::label_percent()(.)
    ),
    fontface='bold', 
    position = position_stack(vjust = 0.5),
  ) + 
  labs_pubr() + 
  theme_void() + 
  scale_fill_manual(
    "Status",
    values=c(
      Complete = '#0B5D1E',
      Fragmented = "#FF2E00",
      Gap = "#FEA82F"
    )
  ) + 
  coord_polar(theta='y') + 
  facet_wrap(~Genome) + 
  theme(
    strip.text = element_text(size=10, face = 'bold'),
    legend.position = 'none'
  )
```

