---
title: "Orthogroup Duplications"
author: "Juan Manuel Vazquez"
date: "2023-05-02"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## A very simple attempt at looking at gene duplications

```{r, warning=F, echo=F, message=F}
library(tidyverse)
```


### Method


All of this is based on `OrthofinderResults/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv`. 

First, a list of all Orthogroups associated with a COSMIC Tier 1 gene was generated by subsetting one of the Ortholog files by COSMIC ENSGs with the command:

```
grep -f COSMIC_ENSG.txt ../../Results_Mar11/Orthologues/Orthologues_human/human__v__auriculus.tsv | cut -f1 | sort -Vu > COSMIC_Myotis_orthogroups.txt
```

Then, I subsetted the `Duplications_per_Orthogroup` file to only include the COSMIC orthogroups:

```
grep -f COSMIC_Myotis_orthogroups.txt ../../Results_Mar11/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv > Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv
```

For all counts related to COSMIC, I used `cut` and `grep` to look for duplication event counts in the 3rd column, "Duplications (50% support)":

```
### No dups
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -w "0" | wc -l

### 1 dup
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -w "1" | wc -l

### 2+ dup
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -vw "0" | grep -vw "1" | wc -l

### Total
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | wc -l
```

I repeated this exactly using the original `Duplications_per_Orthogroup.tsv` file as a baseline.

To make it a more fair fight, I subsetted the original `Duplications_per_Orthogroup.tsv` file to match the total group count of the COSMIC file and repeated the counts:

```
tail -n +2 ../../Results_Mar11/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv | shuf -n 537 > Orthofinder_DuplicationsPerOrthogroup_All.tsv
```


### Table of counts

```{r full counts}
dup <- read_table('../../data/tmp_Orthogroup_duplication.tsv')

```

```{r, results='asis', echo=F}
dup %>% knitr::kable()
```


### COSMIC vs All

### Dups vs no Dups

```{r contingency table: COSMIC vs All}
dup.table1 <- data.frame(row.names = c("COSMIC","OrthoGroup"), notDup = c(445,18106), dup = c(59+33, 2527+1812))

```

```{r, results='asis', echo=F}
dup.table1 %>% knitr::kable()
```

```{r COSMICvAll Mosaic Plot}
mosaicplot(dup.table1,
           main = "COSMIC vs All",
           sub = "Any duplications vs no duplications",
           color = TRUE
)
```

```{r COSMICvAll Fisher Test}
test1 <- fisher.test(dup.table1, alternative = "g")
```

```{r, echo=F}
print(test1)
```

#### oneDup to moreDup

```{r contingency table: COSMIC vs All 2}
dup.table1b <- data.frame(row.names = c("COSMIC","OrthoGroup"), oneDup = c(59,2527), moreDup = c(33, 1812))

```

```{r, results='asis', echo=F}
dup.table1b %>% knitr::kable()
```

```{r COSMICvAll Mosaic Plot 2}
mosaicplot(dup.table1b,
           main = "COSMIC vs All",
           sub = "One duplication vs many duplications",
           color = TRUE
)
```

```{r COSMICvAll Fisher Test 2}
test1b <- fisher.test(dup.table1b, alternative = "g")
```

```{r, echo=F}
print(test1b)
```


### COSMIC vs equal-size subsample;


#### Dup vs No Dups

```{r contingency table: COSMIC vs Sub}
dup.table2 <- data.frame(row.names = c("COSMIC","OrthoGroup_537"), notDup = c(445,421), dup = c(59+33, 62+54))
```

```{r, results='asis', echo=F}
dup.table2 %>% knitr::kable()
```

```{r COSMICvSub Mosaic Plot}
mosaicplot(dup.table2,
           main = "COSMIC vs Subsample",
           sub = "Any duplications vs no duplications",
           color = TRUE
)
```

```{r COSMICvSub Fisher Test}
test2 <- fisher.test(dup.table2, alternative = "g")
```

```{r, echo=F}
print(test2)
```


#### oneDup vs manyDup

```{r contingency table: COSMIC vs Sub 2}
dup.table2b <- data.frame(row.names = c("COSMIC","OrthoGroup"), oneDup = c(59,62), moreDup = c(33, 54))

```

```{r, results='asis', echo=F}
dup.table2b %>% knitr::kable()
```

```{r COSMICvSub Mosaic Plot 2}
mosaicplot(dup.table2b,
           main = "COSMIC vs Subsample",
           sub = "One duplication vs many duplications",
           color = TRUE
)
```

```{r COSMICvSub Fisher Test 2}
test2b <- fisher.test(dup.table2b, alternative = "g")
```

```{r, echo=F}
print(test2b)
```

### Conclusion

As an overall rule, genes in COSMIC orthogroups are ___less___ likely to have duplicated in our _Myotis_ species relative to all other orthogroups. 

On the one hand, this shouldn't be terribly surprising: bats have non-allometric differences in lifespan, and so they shouldn't have a need for an expanded repetoire of TSGs. Furthermore, bat genomes are small relative to other clades [\@citeme], which in combination with our overall gene counts for our species suggests that we shouldn't expect rampant gene duplication. 

On the other hand, based on the multistage model of cancer, cancer risk scales linearly with body size and to the sixth-power for lifespan; thus in cases where there's increases in lifespan, duplication should be the fastest way to deal with it. However, this assumes that 1) there weren't already duplications or other optimizations that preempt the increase in cancer risk in the common ancestor of long- and short-lived myotis; and 2) that the short-lived Myotis really are that short-lived. There is one thing that moves faster than duplication of genes: rewiring of gene regulatory networks. It could very well be that the TEs are changing about and driving things vs the genes.

Then there is the matter of the sensitivity of this analysis. Here, I am lumping all COSMIC genes together, both oncogenes and TSGs. I'm also lumping together all duplications everywhere. A more nuanced approach is called for, that looks at TSGs and oncogenes separately; and that also look at where in the phylogeny things change.


## To Do

- Look at what happens if you do this for another clade with a human outgroup. Ideally, another group of 10 species in the same genus. 
- Break up COSMIC genes into pro- and anti-cancerous, and/or by pro- and anti- cancer hallmark. 
- Redo this using the properly-segregated COSMIC orthogroups.
- Refine the topology of duplications using NOTUNG and CAFE.




