<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Juan M Vazquez (docmanny)" />

<meta name="date" content="2022-12-15" />

<title>mash-dist</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myotis Genomes Interactive Workbook</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">mash-dist</h1>
<h4 class="author">Juan M Vazquez (docmanny)</h4>
<h4 class="date">2022-12-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>R_workflowr/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221115code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221115)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221115code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221115)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytreeab2573219718f1edf48b51314ec37dc1aad8d751targetblankab25732a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/ab2573219718f1edf48b51314ec37dc1aad8d751" target="_blank">ab25732</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytreeab2573219718f1edf48b51314ec37dc1aad8d751targetblankab25732a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/ab2573219718f1edf48b51314ec37dc1aad8d751" target="_blank">ab25732</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    R_workflowr/.RData
    Ignored:    R_workflowr/.Rhistory
    Ignored:    R_workflowr/.Rprofile
    Ignored:    R_workflowr/.Rproj.user/
    Ignored:    R_workflowr/_workflowr.yml
    Ignored:    analyses/annotation/output/
    Ignored:    analyses/cactus/data/
    Ignored:    analyses/pangenome/output/
    Ignored:    data/USGS_SpeciesRanges/
    Ignored:    data/genomes/
    Ignored:    data/gff/
    Ignored:    data/tree/

Untracked files:
    Untracked:  R_workflowr/.gitattributes
    Untracked:  R_workflowr/.gitignore
    Untracked:  R_workflowr/R_workflowr.Rproj
    Untracked:  R_workflowr/analysis/
    Untracked:  R_workflowr/ng_curves.html
    Untracked:  R_workflowr/renv.lock
    Untracked:  R_workflowr/renv/
    Untracked:  analyses/mMyoLuc1/
    Untracked:  analyses/makeHub/
    Untracked:  analyses/tmp-mMyoLuc1/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(ape)
library(phangorn)
library(tidyverse)</code></pre>
<pre><code>Warning in system(&quot;timedatectl&quot;, intern = TRUE): running command &#39;timedatectl&#39;
had status 1</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.4.0      ✔ purrr   0.3.5 
✔ tibble  3.1.8      ✔ dplyr   1.0.10
✔ tidyr   1.2.1      ✔ stringr 1.4.1 
✔ readr   2.1.3      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(ggtree)</code></pre>
<pre><code>ggtree v3.7.1 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

S Xu, Z Dai, P Guo, X Fu, S Liu, L Zhou, W Tang, T Feng, M Chen, L
Zhan, T Wu, E Hu, Y Jiang, X Bo, G Yu. ggtreeExtra: Compact
visualization of richly annotated phylogenetic data. Molecular Biology
and Evolution. 2021, 38(9):4039-4042. doi: 10.1093/molbev/msab166

G Yu. Data Integration, Manipulation and Visualization of Phylogenetic
Trees (1st ed.). Chapman and Hall/CRC. 2022. ISBN: 9781032233574

Attaching package: &#39;ggtree&#39;

The following object is masked from &#39;package:tidyr&#39;:

    expand

The following object is masked from &#39;package:ape&#39;:

    rotate</code></pre>
<pre class="r"><code>library(ggnewscale)
library(ComplexHeatmap)</code></pre>
<pre><code>Loading required package: grid
========================================
ComplexHeatmap version 2.14.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
<pre class="r"><code>species = c(
  &quot;Myotis_occultus&quot;,
  &quot;Myotis_lucifugus&quot;,
  &quot;Myotis_evotis&quot;,
  &quot;Myotis_thysanodes&quot;,
  &quot;Myotis_auriculus&quot;,
  &quot;Myotis_californicus&quot;,
  &quot;Myotis_yumanensis&quot;,
  &quot;Myotis_velifer&quot;,
  &quot;Myotis_volans&quot;
  )

species_genome = c(
  &quot;Myotis_auriculus&quot; = &quot;mMyoAui1.cleaned&quot;,
  &quot;Myotis_californicus&quot; = &quot;mMyoCai1.cleaned&quot;,
  &quot;Myotis_occultus&quot; = &quot;mMyoOcc1.cleaned&quot;,
  &quot;Myotis_lucifugus&quot; = &quot;mMyoLuc1.cleaned&quot;,
  &quot;Myotis_yumanensis&quot; = &quot;mMyoYum1.cleaned&quot;,
  &quot;Myotis_volans&quot; = &quot;mMyoVol1.cleaned&quot;,
  &quot;Myotis_velifer&quot; = &quot;mMyoVel1.cleaned&quot;,
  &quot;Myotis_evotis&quot; = &quot;mMyoEvo1.cleaned&quot;,
  &quot;Myotis_thysanodes&quot; = &quot;mMyoThy1.cleaned&quot;
)

genome_species = c(
  &quot;mMyoAui1.cleaned&quot; = &quot;Myotis_auriculus&quot;,
  &quot;mMyoCai1.cleaned&quot; = &quot;Myotis_californicus&quot;,
  &quot;mMyoOcc1.cleaned&quot; = &quot;Myotis_occultus&quot;,
  &quot;mMyoLuc1.cleaned&quot; = &quot;Myotis_lucifugus&quot;,
  &quot;mMyoYum1.cleaned&quot; = &quot;Myotis_yumanensis&quot;,
  &quot;mMyoVol1.cleaned&quot; = &quot;Myotis_volans&quot;,
  &quot;mMyoVel1.cleaned&quot; = &quot;Myotis_velifer&quot;,
  &quot;mMyoEvo1.cleaned&quot; = &quot;Myotis_evotis&quot;,
  &quot;mMyoThy1.cleaned&quot; = &quot;Myotis_thysanodes&quot;
)</code></pre>
<pre class="r"><code>tree.timetree &lt;- read.tree(&#39;../../data/tree/species_timetree.nwk&#39;) %&gt;% 
  keep.tip(species)

p.tree.timetree &lt;- ape::rotateConstr(tree.timetree, species) %&gt;% 
  ggtree()

p.tree.timetree + 
  theme_tree() + 
  xlim_tree(c(NA,20)) + 
  geom_tiplab() + 
  theme_tree()</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dist_mat &lt;- read_tsv(file = &#39;../../data/mash_matrix_full.tsv&#39;) %&gt;% 
  rename(all_of(species_genome)) %&gt;% 
  mutate(`...1`= genome_species[`...1`]) %&gt;% 
  column_to_rownames(&#39;...1&#39;) %&gt;% 
  as.matrix() %&gt;% 
  as.dist()</code></pre>
<pre><code>New names:
Rows: 9 Columns: 10
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;\t&quot; chr
(1): ...1 dbl (9): mMyoAui1.cleaned, mMyoCai1.cleaned, mMyoEvo1.cleaned,
mMyoLuc1.clea...
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>tree.nj &lt;- ape::nj(dist_mat)

ggtree(tree.nj, layout = &quot;ape&quot;) +
  geom_tiplab()</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tree.upgma &lt;- phangorn::upgma(dist_mat)

p.tree.upgma &lt;- ape::rotateConstr(tree.upgma, species) %&gt;% 
  ggtree()

p.tree.upgma +
  geom_tiplab() + 
  theme_tree() + 
  xlim_tree(c(NA, 0.02))</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cotree.upgma &lt;- ape::rotateConstr(tree.upgma, species)
cotree.timetree &lt;- ape::rotateConstr(tree.timetree, species)

assoc.mat &lt;- cbind(cotree.timetree$tip.label,cotree.timetree$tip.label)
ape::cophyloplot(x = cotree.upgma, y= cotree.timetree, assoc = assoc.mat)</code></pre>
<p><img src="figure/mash-dist.Rmd/cophyloplot%20with%20ape-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p.tree.timetree.cladeogram &lt;- tree.timetree %&gt;% 
  # ape::rotateConstr(species) %&gt;% 
  ggtree(branch.length = &#39;none&#39;) %&gt;% 
  rotate(tree_view = ., node = 15) %&gt;% 
  rotate(tree_view = ., node = 16) %&gt;% 
  rotate(tree_view = ., node = 11)

p.tree.upgma.cladeogram &lt;- ape::rotateConstr(tree.upgma, species) %&gt;% 
  ggtree(branch.length = &#39;none&#39;)

cotree.upgma.dat &lt;- p.tree.upgma.cladeogram$data
cotree.timetree.dat &lt;- p.tree.timetree.cladeogram$data

cotree.upgma.dat$x &lt;- max(cotree.upgma.dat$x) - cotree.upgma.dat$x + max(cotree.timetree.dat$x) + 50

dd &lt;- bind_rows(cotree.timetree.dat, cotree.upgma.dat) %&gt;% 
  filter(!is.na(label)) %&gt;% 
  arrange(label, x,y)

color.subclades = c(
  &quot;Myotis_lucifugus&quot; = &quot;#1B9E77&quot;,
  &quot;Myotis_occultus&quot; = &quot;#1B9E77&quot;,
  &quot;Myotis_volans&quot; = &quot;#1B9E77&quot;,
  &quot;Myotis_evotis&quot; = &quot;#D95F02&quot;,
  &quot;Myotis_thysanodes&quot; = &quot;#D95F02&quot;,
  &quot;Myotis_auriculus&quot; = &quot;#7570B3&quot;,
  &quot;Myotis_californicus&quot; = &quot;#7570B3&quot;,
  &quot;Myotis_yumanensis&quot; = &quot;#E7298A&quot;,
  &quot;Myotis_velifer&quot; = &quot;#E7298A&quot;
  )

p.cotree &lt;- p.tree.timetree.cladeogram +
  geom_line(aes(x,y, group=label, color=label), data=dd) + 
  geom_tree(data=cotree.upgma.dat) + 
  geom_tiplab() + 
  geom_tiplab(data=cotree.upgma.dat, hjust = 1) + 
  scale_color_manual(values=color.subclades) + 
  guides(color=guide_none())
  # ggnewscale::new_scale_fill() +
  # geom_hilight(
  #        data = d2,
  #        mapping = aes(
  #           subset = node %in% c(38, 48, 58),
  #           node=node,
  #           fill=as.factor(node))
  # ) +
  # labs(fill = &quot;clades for tree in right&quot; )</code></pre>
<pre class="r"><code>tree.upgma %&gt;% keep.tip(c(&quot;Myotis_volans&quot;, &quot;Myotis_evotis&quot;, &quot;Myotis_thysanodes&quot;)) %&gt;% ggtree() + geom_tiplab() + xlim_tree(c(NA,0.05))</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ape::root.phylo(tree.nj, node = 16) %&gt;% drop.tip(c(&quot;Myotis_californicus&quot;, &quot;Myotis_auriculus&quot;)) %&gt;% ggtree() + geom_tiplab() + geom_nodelab(aes(label=node), hjust=-1) + xlim_tree(c(NA,0.05))</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="by-chromosome" class="section level2">
<h2>By chromosome:</h2>
<pre class="r"><code>dist.chr &lt;- read_csv(&quot;../../analyses/pangenome/output/mash-triangle/neartic_myotis.mash_triangle.all.individualChrom.csv&quot;) %&gt;% 
  column_to_rownames(&#39;...1&#39;) %&gt;% 
  as.matrix() %&gt;% 
  as.dist()</code></pre>
<pre><code>New names:
Rows: 6409 Columns: 6410
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; chr
(1): ...1 dbl (6408): mMyoAui1.0.SUPER__1, mMyoAui1.0.SUPER__2,
mMyoAui1.0.SUPER__3, m... lgl (1): mMyoSep1.0.HiC_scaffold_4948
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>df.chr &lt;- dist.chr %&gt;% 
  as.matrix() %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column(&quot;genome.hap.scaffold&quot;) %&gt;% 
  as_tibble %&gt;% 
  separate(&quot;genome.hap.scaffold&quot;, c(&quot;genome&quot;, &quot;hap&quot;, &quot;scaffold&quot;), sep=&quot;\\.&quot;, remove = F)</code></pre>
<pre class="r"><code>Heatmap(
  dist_mat %&gt;% as.matrix,
  name = &quot;mash&quot;)</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dist.chr.mat &lt;- dist.chr %&gt;% as.matrix

Heatmap(
  df.chr %&gt;% 
    # select(-starts_with(&quot;mMyoAui1&quot;), -starts_with(&quot;mMyoYum1&quot;, -starts_with)) %&gt;% 
    filter(genome == &quot;mMyoAui1&quot;) %&gt;% 
    select(starts_with(&quot;mMyoSep&quot;), genome.hap.scaffold) %&gt;% 
    select_if(function(x){any(x!=1, na.rm = T)}) %&gt;% 
    column_to_rownames(&quot;genome.hap.scaffold&quot;) %&gt;% 
    as.matrix,
  name = &quot;mash&quot;, show_row_dend = F, show_column_names = F, show_column_dend = F)</code></pre>
<pre><code>`use_raster` is automatically set to TRUE for a matrix with more than
2000 columns You can control `use_raster` argument by explicitly
setting TRUE/FALSE to it.

Set `ht_opt$message = FALSE` to turn off this message.</code></pre>
<p><img src="figure/mash-dist.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df.chr %&gt;% 
    # select(-starts_with(&quot;mMyoAui1&quot;), -starts_with(&quot;mMyoYum1&quot;, -starts_with)) %&gt;% 
    filter(genome == &quot;mMyoAui1&quot;) %&gt;% 
    select(starts_with(&quot;mMyoSep&quot;), genome, hap, scaffold) %&gt;% 
  pivot_longer(-c(genome,hap,scaffold)) %&gt;% 
  group_by(genome, hap, scaffold) %&gt;% 
  filter(value == min(value)) %&gt;% 
  summarize(closest = name, value=value)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;genome&#39;, &#39;hap&#39;, &#39;scaffold&#39;. You can
override using the `.groups` argument.</code></pre>
<pre><code># A tibble: 148 × 5
# Groups:   genome, hap, scaffold [136]
   genome   hap   scaffold   closest                       value
   &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;                         &lt;dbl&gt;
 1 mMyoAui1 0     SUPER__1   mMyoSep1.0.HiC_scaffold_18   0.0131
 2 mMyoAui1 0     SUPER__10  mMyoSep1.0.HiC_scaffold_8    0.0137
 3 mMyoAui1 0     SUPER__101 mMyoSep1.0.HiC_scaffold_107  0.0328
 4 mMyoAui1 0     SUPER__102 mMyoSep1.0.HiC_scaffold_107  0.0223
 5 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_1971 0.225 
 6 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_2015 0.225 
 7 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_2188 0.225 
 8 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_2343 0.225 
 9 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_2345 0.225 
10 mMyoAui1 0     SUPER__103 mMyoSep1.0.HiC_scaffold_2605 0.225 
# … with 138 more rows</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 Patched (2022-11-10 r83330)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.14.0 ggnewscale_0.4.8      ggtree_3.7.1         
 [4] forcats_0.5.2         stringr_1.4.1         dplyr_1.0.10         
 [7] purrr_0.3.5           readr_2.1.3           tidyr_1.2.1          
[10] tibble_3.1.8          ggplot2_3.4.0         tidyverse_1.3.2      
[13] phangorn_2.10.0       ape_5.6-2            

loaded via a namespace (and not attached):
  [1] googledrive_2.0.0   colorspace_2.0-3    rjson_0.2.21       
  [4] ellipsis_0.3.2      rprojroot_2.0.3     circlize_0.4.15    
  [7] GlobalOptions_0.1.2 fs_1.5.2            aplot_0.1.8        
 [10] clue_0.3-63         rstudioapi_0.14     farver_2.1.1       
 [13] bit64_4.0.5         fansi_1.0.3         lubridate_1.9.0    
 [16] xml2_1.3.3          codetools_0.2-18    doParallel_1.0.17  
 [19] cachem_1.0.6        knitr_1.40          jsonlite_1.8.3     
 [22] workflowr_1.7.0     broom_1.0.1         cluster_2.1.4      
 [25] dbplyr_2.2.1        png_0.1-7           compiler_4.2.2     
 [28] httr_1.4.4          backports_1.4.1     assertthat_0.2.1   
 [31] Matrix_1.5-1        fastmap_1.1.0       lazyeval_0.2.2     
 [34] gargle_1.2.1        cli_3.4.1           later_1.3.0        
 [37] htmltools_0.5.4     tools_4.2.2         igraph_1.3.5       
 [40] gtable_0.3.1        glue_1.6.2          fastmatch_1.1-3    
 [43] Rcpp_1.0.9          cellranger_1.1.0    jquerylib_0.1.4    
 [46] vctrs_0.5.0         nlme_3.1-160        iterators_1.0.14   
 [49] xfun_0.34           rvest_1.0.3         timechange_0.1.1   
 [52] lifecycle_1.0.3     googlesheets4_1.0.1 scales_1.2.1       
 [55] vroom_1.6.0         hms_1.1.2           promises_1.2.0.1   
 [58] parallel_4.2.2      RColorBrewer_1.1-3  yaml_2.3.6         
 [61] ggfun_0.0.8         yulab.utils_0.0.5   sass_0.4.2         
 [64] stringi_1.7.8       highr_0.9           S4Vectors_0.36.1   
 [67] foreach_1.5.2       tidytree_0.4.1      BiocGenerics_0.44.0
 [70] shape_1.4.6         rlang_1.0.6         pkgconfig_2.0.3    
 [73] matrixStats_0.63.0  evaluate_0.18       lattice_0.20-45    
 [76] labeling_0.4.2      treeio_1.23.0       patchwork_1.1.2    
 [79] bit_4.0.4           tidyselect_1.2.0    magrittr_2.0.3     
 [82] R6_2.5.1            magick_2.7.3        IRanges_2.32.0     
 [85] generics_0.1.3      DBI_1.1.3           pillar_1.8.1       
 [88] haven_2.5.1         withr_2.5.0         modelr_0.1.9       
 [91] crayon_1.5.2        utf8_1.2.2          tzdb_0.3.0         
 [94] rmarkdown_2.17      GetoptLong_1.0.5    readxl_1.4.1       
 [97] git2r_0.30.1        reprex_2.0.2        digest_0.6.30      
[100] httpuv_1.6.6        gridGraphics_0.5-1  stats4_4.2.2       
[103] munsell_0.5.0       ggplotify_0.1.0     bslib_0.4.1        
[106] quadprog_1.5-8     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
