<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Juan Manuel Vazquez" />


<title>QC Plots</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">R_workflowr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QC Plots</h1>
<h4 class="author">Juan Manuel Vazquez</h4>
<h4 class="date">6/27/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>R_workflowr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221115code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221115)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221115code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221115)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree5c92bd2a13493dbae32f7c8b3e046d097f503d47targetblank5c92bd2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/5c92bd2a13493dbae32f7c8b3e046d097f503d47" target="_blank">5c92bd2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree5c92bd2a13493dbae32f7c8b3e046d097f503d47targetblank5c92bd2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/5c92bd2a13493dbae32f7c8b3e046d097f503d47" target="_blank">5c92bd2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    R_workflowr/.Rproj.user/
    Ignored:    data/USGS_SpeciesRanges/
    Ignored:    data/tree/

Untracked files:
    Untracked:  R_workflowr/analysis/ng_curves.Rmd
    Untracked:  R_workflowr/analysis/sample_map.Rmd
    Untracked:  R_workflowr/ng_curves.html
    Untracked:  R_workflowr/renv.lock
    Untracked:  R_workflowr/renv/

Unstaged changes:
    Modified:   .gitignore
    Modified:   R_workflowr/README.md
    Modified:   R_workflowr/_workflowr.yml
    Deleted:    R_workflowr/code/README.md
    Modified:   data/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>assembly_stats &lt;- read_tsv(&quot;~/projects/MyotisGenomeAssembly/data/chiroptera.contig.assembly_stats.tsv&quot;, col_names = c(&quot;genomeName&quot;, &quot;stat&quot;, &quot;v1&quot;, &quot;v2&quot;)) %&gt;% 
  mutate(contig = str_detect(genomeName, &quot;contig&quot;)) %&gt;% 
  filter(str_detect(genomeName, &quot;M_|mmyo|\\.chr&quot;, negate=T)) %&gt;%
  filter(str_detect(genomeName, &quot;mMyo.+1(?!_unmasked)&quot;, negate=T)) %&gt;%
  separate(genomeName, c(&quot;name&quot;, &quot;scaffold&quot;), sep=&quot;\\.&quot;) %&gt;% 
  arrange(name, contig) %&gt;% 
  select(-scaffold)</code></pre>
<pre><code>Rows: 27336 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): genomeName, stat
dbl (2): v1, v2

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>our_genomes = c(
  &quot;Myotis_auriculus&quot;,
  &quot;Myotis_californicus&quot;,
  &quot;Myotis_occultus&quot;,
  &quot;Myotis_lucifugus&quot;,
  &quot;Myotis_yumanensis&quot;,
  &quot;Myotis_volans&quot;,
  &quot;Myotis_velifer&quot;,
  &quot;Myotis_evotis&quot;,
  &quot;Myotis_thysanodes&quot;
)</code></pre>
<pre class="r"><code>meta &lt;- read_tsv(&quot;../data/chiroptera_genomes_20220627.tsv&quot;) %&gt;% 
  filter(!is.na(`Assembly BioProject Lineage Parent Accessions`)) %&gt;% 
  mutate(`Organism Name` = `Organism Name` %&gt;% str_replace_all(&quot; &quot;, &quot;_&quot;))</code></pre>
<pre><code>Rows: 129 Columns: 29
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr  (15): Assembly Name, Assembly Accession, Organism Name, Organism Infras...
dbl  (10): Annotation BUSCO Complete, Annotation BUSCO Duplicated, Annotatio...
lgl   (3): Organism Infraspecific Names Strain, Organism Infraspecific Names...
date  (1): Assembly Submission Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>meta.origin &lt;- meta %&gt;% 
  filter(str_detect(`Assembly BioProject Lineage Title`, &quot;[Aa]ssembl&quot;, negate = T)) %&gt;%
  filter(str_detect(`Assembly BioProject Lineage Title`, &quot;Myotis myotis&quot;, negate = T)) %&gt;% 
  filter(str_detect(`Assembly BioProject Lineage Title`, &quot;Pipistrellus&quot;, negate = T)) %&gt;% 
  select(name=`Organism Name`, `Assembly BioProject Lineage Title`) %&gt;% 
  group_by(name) %&gt;% 
  summarize(Origin = `Assembly BioProject Lineage Title` %&gt;% unique %&gt;% sort %&gt;%  paste(collapse=&quot;,&quot;)) %&gt;% 
  mutate(Origin = Origin %&gt;% 
           str_replace(&quot;The European Reference Genome Atlas,Vertebrate Genomes Project&quot;, &quot;Vertebrate Genomes Project&quot;) %&gt;% 
           str_replace(&quot;Bat1K Project,Vertebrate Genomes Project&quot;, &quot;Bat1K Project&quot;) %&gt;% 
           str_replace(&quot;Genome sequencing for Pteropus vampyrus&quot;, &quot;Other&quot;) %&gt;% 
           str_replace(&quot;Eptesicus fuscus Genome Sequencing&quot;, &quot;Other&quot;)
         ) %&gt;% 
  bind_rows(assembly_stats %&gt;% filter(str_detect(name, &quot;mMyo&quot;)) %&gt;% select(name) %&gt;% distinct %&gt;% mutate(Origin=&quot;This Paper&quot;))

assembly_stats.meta &lt;- assembly_stats %&gt;% 
  left_join(meta.origin, by=&quot;name&quot;) %&gt;% 
  replace_na(list(Origin=&quot;Other&quot;)) %&gt;% 
  mutate(
    species = name %&gt;% 
      str_replace(&quot;mMyoAui.*&quot;, &quot;Myotis_auriculus&quot;) %&gt;% 
      str_replace(&quot;mMyoCai.*&quot;, &quot;Myotis_californicus&quot;) %&gt;% 
      str_replace(&quot;mMyoOcc.*&quot;, &quot;Myotis_occultus&quot;) %&gt;% 
      str_replace(&quot;mMyoLuc.*&quot;, &quot;Myotis_lucifugus&quot;) %&gt;% 
      str_replace(&quot;mMyoYum.*&quot;, &quot;Myotis_yumanensis&quot;) %&gt;% 
      str_replace(&quot;mMyoVol.*&quot;, &quot;Myotis_volans&quot;) %&gt;% 
      str_replace(&quot;mMyoVel.*&quot;, &quot;Myotis_velifer&quot;) %&gt;% 
      str_replace(&quot;mMyoEvo.*&quot;, &quot;Myotis_evotis&quot;) %&gt;% 
      str_replace(&quot;mMyoThy.*&quot;, &quot;Myotis_thysanodes&quot;) %&gt;% 
      str_replace(&quot;_T2T&quot;,&quot;&quot;) %&gt;% 
      str_replace(&quot;myoLuc2&quot;, &quot;Myotis_lucifugus&quot;)
  )

assembly_stats.meta[str_detect(assembly_stats.meta$name, &quot;T2T&quot;),&quot;Origin&quot;] &lt;- &quot;T2T&quot;
assembly_stats.meta[str_detect(assembly_stats.meta$name, &quot;Yum&quot;),&quot;Origin&quot;] &lt;- &quot;CCGP&quot;</code></pre>
<pre class="r"><code>tr &lt;- treeio::read.newick(&quot;~/projects/MyotisGenomeAssembly/data/tree/species_timetree2.nwk&quot;) %&gt;% 
  ape::drop.tip(&quot;Rhinolophus_sinicus&quot;)

setdiff(tr$tip.label, assembly_stats.meta$species)</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>setdiff(assembly_stats.meta$species, tr$tip.label)</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>plot.NGx &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;NGx&quot;,
         contig==T
         ) %&gt;% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x=&quot;NGx&quot;, y = &quot;bp&quot;, title = &quot;Contig Contiguity&quot;) + 
  geom_point() + 
  geom_line() + 
  ylim(1e4,NA) + 
  scale_color_brewer(palette = &quot;Dark2&quot;) + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.NGx</code></pre>
<pre><code>Warning: Removed 697 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 697 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/ng_curves.Rmd/NGx-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.auNG &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;auNG&quot;,
         contig==T
         ) %&gt;% 
  arrange(desc(v2)) %&gt;% 
  mutate(name = factor(name, levels = name %&gt;% unique)) %&gt;% 
  ggplot(aes(x=v2, y=name, color=Origin)) + 
  labs(x=&quot;auNG&quot;, y = &quot;Assembly&quot;, title = &quot;auNG&quot;) + 
  geom_point() + 
  scale_color_brewer(palette = &quot;Dark2&quot;) + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.auNG</code></pre>
<p><img src="figure/ng_curves.Rmd/AuNG-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>auNG &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;auNG&quot;,
         contig==T
         ) %&gt;% 
  select(species, Origin, v2, v1) %&gt;% 
  rename(label=species, auNG=v2, genomeSize=v1) %&gt;% 
  filter(!(str_detect(label, &quot;lucifugus&quot;) &amp; Origin==&quot;Other&quot;))

color_scale_origin = RColorBrewer::brewer.pal(n=auNG %&gt;% pull(Origin) %&gt;% unique %&gt;% length, name=&quot;Dark2&quot;) %&gt;% set_names(., auNG %&gt;% pull(Origin) %&gt;% unique)

# tr.auNG &lt;- full_join(tr,auNG)

## Sometimes I hate ggtree
nodes_ourgenomes &lt;- tr %&gt;% 
  as_tibble %&gt;% 
  filter(label %in% our_genomes) %&gt;% 
  select(label, node) %&gt;% 
  deframe()


p.tr.auNG &lt;- tr %&gt;% 
  ggtree() + 
  geom_tiplab(
    aes(
      subset = (node %in% nodes_ourgenomes)
    ),
    align = T,
    fontface=&quot;bold&quot;
    ) +
  geom_tiplab(
    aes(
      subset = !(node %in% nodes_ourgenomes)
    ),
    align = T
    ) +
  xlim_tree(c(NA,250)) +
  geom_facet(
    panel = &quot;auNG (bar)&quot;,
    data = auNG,
    geom = geom_col,
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = &#39;y&#39;,
    width = .6
  )+
  geom_facet(
    panel = &quot;auNG (point)&quot;, 
    data = auNG, 
    geom = geom_point, 
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ), 
    size=2
  )+
  theme_tree2() +
  scale_color_manual(&quot;Origin&quot;, values = color_scale_origin) +
  scale_fill_manual(&quot;Origin&quot;, values = color_scale_origin) +
  # geom_facet(panel = &quot;Genome Size&quot;, data = auNG %&gt;% select(label, genomeSize, Origin), geom = geom_point, 
  #              mapping=aes(x = genomeSize, color=Origin))+ 
  theme(legend.position = &quot;bottom&quot;)

p.tr.auNG</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20plus%20auNG-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>auNG.fancy &lt;- auNG %&gt;% 
  mutate(
    label_fancy = sapply(label, . %&gt;% if_else(. %in% our_genomes, str_c(&quot;**&quot;,.,&quot;**&quot;), .)) %&gt;% 
      str_remove(&quot;_mesoamericanus&quot;) %&gt;% 
      str_replace_all(&quot;_&quot;, &quot; &quot;)
  )

angle_rotate &lt;- function(angle){
    # subset1 &lt;- &quot;(angle &lt; 90 | angle &gt; 270)&quot;
    # subset2 &lt;- &quot;(angle &gt;= 90 &amp; angle &lt;=270)&quot;
    if (angle &lt; 90 | angle &gt; 270){
      return(angle)
    } else {
      return(angle+180)
    }
}
h_readjust &lt;- function(angle){
    if (angle &lt; 90 | angle &gt; 270){
      return(0)
    } else {
      return(1)
    }
}

p.tr.auNG.circ &lt;-
  tr %&gt;%
  ggtree(layout = &quot;fan&quot;, open.angle=180) +
  xlim_tree(c(NA,100))+ 
  geom_fruit(
    data = auNG.fancy,
    geom = geom_col,
    mapping=aes(
      y= label,
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = &#39;y&#39;,
    width = .6
  ) +
  geom_fruit(
    data = auNG.fancy,
    geom = geom_richtext, 
    mapping = aes(
      y=label,
      label=label_fancy,
      color=Origin,
      angle=sapply(angle,angle_rotate),
      hjust=sapply(angle,h_readjust)
    ),
    fill = NA, 
    label.color = NA, 
    label.padding = grid::unit(rep(0, 4), &quot;pt&quot;),
    
  ) + 
  scale_color_manual(&quot;Origin&quot;, values = color_scale_origin, guide=&quot;none&quot;) +
  scale_fill_manual(&quot;Origin&quot;, values = color_scale_origin,
                    guide = guide_legend(nrow = 2, ncol=6, title.position = &quot;left&quot;)) +
  theme(legend.position = c(0.6,0.3), plot.margin = unit(c(4,0,0,0),&quot;cm&quot;))</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.</code></pre>
<pre class="r"><code>p.tr.auNG.circ</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20auNG%20circular-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>species_color = ggsci::pal_d3(palette = &quot;category20&quot;)(length(our_genomes)+1) %&gt;% 
  set_names(., c(our_genomes, &quot;Other&quot;))

species_color[&quot;Myotis_evotis&quot;] = &quot;#17BECFFF&quot;
species_color[&quot;Other&quot;] = &quot;#7F7F7FFF&quot;

auNG.fancy.color &lt;- auNG.fancy %&gt;% 
  mutate(color_me = sapply(label, . %&gt;% ifelse(. %in% our_genomes, ., &quot;Other&quot;)))

p.tr.auNG.circ.altcolor &lt;-
  tr %&gt;%
  ggtree(layout = &quot;fan&quot;, open.angle=180) +
  xlim_tree(c(NA,100))+ 
  geom_fruit_list(
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      orientation = &#39;y&#39;,
      width = .6
    ),
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_point,
      mapping=aes(
        y= label,
        x = log(17.7),
        angle=angle
      ),
      color=&quot;black&quot;,
      shape = &quot;-&quot;,
      size=2,
      vjust = 1,
      hjust = 1
    ),
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_point,
      mapping=aes(
        y= label,
        x = log(max(auNG)),
        angle=angle
      ),
      color=&quot;blue&quot;,
      shape = &quot;-&quot;,
      size=2,
      vjust = 1,
      hjust = 1
    )
  ) +
  geom_fruit(
    data = auNG.fancy.color,
    geom = geom_richtext, 
    mapping = aes(
      y=label,
      label=label_fancy,
      color=color_me,
      angle=sapply(angle,angle_rotate),
      hjust=sapply(angle,h_readjust)
    ),
    fill = NA, 
    label.color = NA, 
    label.padding = grid::unit(rep(0, 4), &quot;pt&quot;),
  ) + 
  scale_color_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  scale_fill_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  theme(plot.margin = unit(c(4,0,0,0),&quot;cm&quot;))</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.</code></pre>
<pre><code>Warning in (function (mapping = NULL, data = NULL, stat = &quot;identity&quot;, position =
&quot;identity&quot;, : Ignoring unknown parameters: `vjust` and `hjust`</code></pre>
<pre><code>Warning in (function (mapping = NULL, data = NULL, stat = &quot;identity&quot;, position =
&quot;identity&quot;, : Ignoring unknown aesthetics: angle</code></pre>
<pre><code>Warning in (function (mapping = NULL, data = NULL, stat = &quot;identity&quot;, position =
&quot;identity&quot;, : Ignoring unknown parameters: `vjust` and `hjust`</code></pre>
<pre><code>Warning in (function (mapping = NULL, data = NULL, stat = &quot;identity&quot;, position =
&quot;identity&quot;, : Ignoring unknown aesthetics: angle</code></pre>
<pre class="r"><code>p.tr.auNG.circ.altcolor</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20auNG%20circular%20map%20color-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggsci_2.9         ggtext_0.1.2      ggtreeExtra_1.8.0 ggtree_3.7.1     
 [5] patchwork_1.1.2   ggpubr_0.4.0      magrittr_2.0.3    forcats_0.5.2    
 [9] stringr_1.4.1     dplyr_1.0.10      purrr_0.3.5       readr_2.1.3      
[13] tidyr_1.2.1       tibble_3.1.8      ggplot2_3.4.0     tidyverse_1.3.2  

loaded via a namespace (and not attached):
 [1] googledrive_2.0.0   ggnewscale_0.4.8    colorspace_2.0-3   
 [4] ggsignif_0.6.4      ellipsis_0.3.2      rprojroot_2.0.3    
 [7] markdown_1.3        fs_1.5.2            aplot_0.1.8        
[10] gridtext_0.1.5      rstudioapi_0.14     farver_2.1.1       
[13] bit64_4.0.5         fansi_1.0.3         lubridate_1.9.0    
[16] xml2_1.3.3          cachem_1.0.6        knitr_1.40         
[19] jsonlite_1.8.3      workflowr_1.7.0     broom_1.0.1        
[22] dbplyr_2.2.1        compiler_4.2.2      httr_1.4.4         
[25] backports_1.4.1     assertthat_0.2.1    fastmap_1.1.0      
[28] lazyeval_0.2.2      gargle_1.2.1        cli_3.4.1          
[31] later_1.3.0         htmltools_0.5.3     tools_4.2.2        
[34] gtable_0.3.1        glue_1.6.2          Rcpp_1.0.9         
[37] carData_3.0-5       cellranger_1.1.0    jquerylib_0.1.4    
[40] vctrs_0.5.0         ape_5.6-2           nlme_3.1-160       
[43] xfun_0.34           rvest_1.0.3         timechange_0.1.1   
[46] lifecycle_1.0.3     renv_0.16.0         rstatix_0.7.0      
[49] googlesheets4_1.0.1 scales_1.2.1        vroom_1.6.0        
[52] hms_1.1.2           promises_1.2.0.1    parallel_4.2.2     
[55] RColorBrewer_1.1-3  yaml_2.3.6          ggfun_0.0.8        
[58] yulab.utils_0.0.5   sass_0.4.2          stringi_1.7.8      
[61] highr_0.9           tidytree_0.4.1      rlang_1.0.6        
[64] pkgconfig_2.0.3     commonmark_1.8.1    evaluate_0.18      
[67] lattice_0.20-45     treeio_1.23.0       labeling_0.4.2     
[70] bit_4.0.4           tidyselect_1.2.0    R6_2.5.1           
[73] magick_2.7.3        generics_0.1.3      DBI_1.1.3          
[76] pillar_1.8.1        haven_2.5.1         withr_2.5.0        
[79] abind_1.4-5         modelr_0.1.9        crayon_1.5.2       
[82] car_3.1-1           utf8_1.2.2          tzdb_0.3.0         
[85] rmarkdown_2.17      grid_4.2.2          readxl_1.4.1       
[88] git2r_0.30.1        reprex_2.0.2        digest_0.6.30      
[91] httpuv_1.6.6        gridGraphics_0.5-1  munsell_0.5.0      
[94] ggplotify_0.1.0     bslib_0.4.1        </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
