<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Juan Manuel Vazquez" />


<title>QC Plots</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myotis Genome Analyses</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QC Plots</h1>
<h4 class="author">Juan Manuel Vazquez</h4>
<h4 class="date">6/27/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>R_workflowr/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree9d4271fb67db5a54ed7df5703ae96751ebb342cetargetblank9d4271fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/9d4271fb67db5a54ed7df5703ae96751ebb342ce" target="_blank">9d4271f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree9d4271fb67db5a54ed7df5703ae96751ebb342cetargetblank9d4271fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/9d4271fb67db5a54ed7df5703ae96751ebb342ce" target="_blank">9d4271f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    R_workflowr/.RData
    Ignored:    R_workflowr/.Rproj.user/
    Ignored:    R_workflowr/analysis/figure/
    Ignored:    R_workflowr/data/
    Ignored:    R_workflowr/output/pubFiles/
    Ignored:    R_workflowr/output/webGestalt/
    Ignored:    R_workflowr/renv/library/
    Ignored:    R_workflowr/renv/staging/
    Ignored:    R_workflowr/test.phy

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>R_workflowr/analysis/ng_curves.Rmd</code>)
and HTML (<code>R_workflowr/docs/ng_curves.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/6f1b4403ae2968d962ac387fe1b01ee443f05dd3/R_workflowr/analysis/ng_curves.Rmd" target="_blank">6f1b440</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-27
</td>
<td>
PANCOVA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sudmantlab/MyotisGenomeAssembly/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/ng_curves.html" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
<td>
wflow_build works
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/8f66d05c66fd494567461d74f4110a0876566489/R_workflowr/analysis/ng_curves.Rmd" target="_blank">8f66d05</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-21
</td>
<td>
clean version of Fig 1, now with LQ
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/analysis/ng_curves.Rmd" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
<td>
draft sub figs for F1&amp;2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sudmantlab/MyotisGenomeAssembly/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/ng_curves.html" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
<td>
draft sub figs for F1&amp;2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/89c89e617642e32aba25ce984ce60847f5c62550/R_workflowr/analysis/ng_curves.Rmd" target="_blank">89c89e6</a>
</td>
<td>
docmanny
</td>
<td>
2023-05-12
</td>
<td>
reorg
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/0408a6b776b857654cc39b442128c9dbde43be6c/R_workflowr/analysis/ng_curves.Rmd" target="_blank">0408a6b</a>
</td>
<td>
docmanny
</td>
<td>
2023-05-09
</td>
<td>
Figure 1 work
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/130db9241027e566774a78d8976548246a4c60cc/R_workflowr/analysis/ng_curves.Rmd" target="_blank">130db92</a>
</td>
<td>
docmanny
</td>
<td>
2023-05-09
</td>
<td>
workflowr::wflow_git_commit(all = T)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sudmantlab/MyotisGenomeAssembly/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/ng_curves.html" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
<td>
Adding workflowr
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/ca0752885e2873b157615bd4dca8dd3c8b11d587/R_workflowr/analysis/ng_curves.Rmd" target="_blank">ca07528</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
<td>
wflow_git_commit(".")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="assembly-stats" class="section level2">
<h2>Assembly Stats</h2>
<pre class="r"><code>our_genomes = c(
  &quot;Myotis_auriculus&quot;,
  &quot;Myotis_californicus&quot;,
  &quot;Myotis_occultus&quot;,
  &quot;Myotis_lucifugus&quot;,
  &quot;Myotis_yumanensis&quot;,
  &quot;Myotis_volans&quot;,
  &quot;Myotis_velifer&quot;,
  &quot;Myotis_evotis&quot;,
  &quot;Myotis_thysanodes&quot;
)</code></pre>
<pre class="r"><code>assembly_stats &lt;- read_tsv(&quot;../data/assembly_stats/all_assembly_stats.contigs.tsv&quot;, col_names = c(&quot;genomeName&quot;, &quot;stat&quot;, &quot;v1&quot;, &quot;v2&quot;)) %&gt;% 
  mutate(contig = str_detect(genomeName, &quot;contig&quot;)) %&gt;% 
  filter(str_detect(genomeName, &quot;M_|mmyo|\\.chr&quot;, negate=T)) %&gt;%
  filter(str_detect(genomeName, &quot;mMyo.+1(?!_unmasked)&quot;, negate=T)) %&gt;%
  separate(genomeName, c(&quot;name&quot;, &quot;scaffold&quot;), sep=&quot;\\.&quot;) %&gt;% 
  arrange(name, contig) %&gt;% 
  select(-scaffold) %&gt;% 
  mutate(name=name %&gt;% str_remove(&quot;_unmasked&quot;))</code></pre>
<pre class="r"><code>assembly_stats &lt;- read_tsv(&#39;../data/assembly_stats/all_assembly_stats.cleaned.tsv&#39;)</code></pre>
<pre><code>Rows: 30552 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): name, stat
dbl (2): v1, v2
lgl (1): contig

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>meta &lt;- read_tsv(&quot;../data/chiropteraGenomes-20230427.tsv&quot;) %&gt;% 
  mutate(`Organism Name` = `Organism Name` %&gt;% str_replace_all(&quot; &quot;, &quot;_&quot;))</code></pre>
<pre><code>Rows: 52 Columns: 33
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (19): Assembly Accession, Assembly Name, Organism Name, Organism Infrasp...
dbl (12): Assembly Stats Total Sequence Length, Assembly Stats Contig N50, A...
lgl  (2): Organism Infraspecific Names Strain, Organism Infraspecific Names ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>meta.origin &lt;- meta %&gt;% 
  select(name=`Organism Name`, Origin=Group, Citation) %&gt;% 
  group_by(name) %&gt;% 
  mutate(
    Origin = ifelse(Origin %in% c(&#39;Zoonomia&#39;, &#39;Bat1K&#39;, &#39;DNAZoo&#39;, &#39;CCGP&#39;), Origin, &#39;Other&#39;)
  ) %&gt;% 
  bind_rows(assembly_stats %&gt;% filter(str_detect(name, &quot;mMyo&quot;)) %&gt;% select(name) %&gt;% distinct %&gt;% mutate(Origin=&quot;This Paper&quot;, Citation = &quot;This Paper&quot;, name = name %&gt;% str_remove(&quot;_unmasked&quot;)))

assembly_stats.meta &lt;- assembly_stats %&gt;% 
  left_join(meta.origin, by=&quot;name&quot;) %&gt;% 
  replace_na(list(Origin=&quot;Other&quot;))%&gt;% mutate(
    species = name %&gt;% 
      str_replace(&quot;mMyoAui.*&quot;, &quot;Myotis_auriculus&quot;) %&gt;% 
      str_replace(&quot;mMyoCai.*&quot;, &quot;Myotis_californicus&quot;) %&gt;% 
      str_replace(&quot;mMyoOcc.*&quot;, &quot;Myotis_occultus&quot;) %&gt;% 
      str_replace(&quot;mMyoLuc.*&quot;, &quot;Myotis_lucifugus&quot;) %&gt;% 
      str_replace(&quot;mMyoYum.*&quot;, &quot;Myotis_yumanensis&quot;) %&gt;% 
      str_replace(&quot;mMyoVol.*&quot;, &quot;Myotis_volans&quot;) %&gt;% 
      str_replace(&quot;mMyoVel.*&quot;, &quot;Myotis_velifer&quot;) %&gt;% 
      str_replace(&quot;mMyoEvo.*&quot;, &quot;Myotis_evotis&quot;) %&gt;% 
      str_replace(&quot;mMyoThy.*&quot;, &quot;Myotis_thysanodes&quot;) %&gt;% 
      str_replace(&quot;_T2T&quot;,&quot;&quot;) %&gt;% 
      str_replace(&quot;myoLuc2&quot;, &quot;Myotis_lucifugus&quot;)
  )

assembly_stats.meta[str_detect(assembly_stats.meta$name, &quot;T2T&quot;),&quot;Origin&quot;] &lt;- &quot;T2T&quot;</code></pre>
<div id="table" class="section level3">
<h3>Table</h3>
<pre class="r"><code>table.auNG &lt;- assembly_stats.meta %&gt;% 
  filter(stat==&#39;auNG&#39;) %&gt;% 
  arrange(v2) %&gt;% 
  mutate(v2=scales::label_comma()(v2),
         species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
  select(-v1, -contig, -Citation, -name) %&gt;% 
  select(Species=species, everything()) %&gt;% 
  dplyr::rename(auNG=&#39;v2&#39;)
table.LG50 &lt;- assembly_stats.meta %&gt;% 
  filter(stat==&#39;LGx&#39;, v1==50) %&gt;% 
  arrange(v2) %&gt;% 
  mutate(v2=scales::label_comma()(v2),
         species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
  select(-v1, -contig, -Citation, -name) %&gt;% 
  select(Species=species, everything()) %&gt;% 
  dplyr::rename(LG50=&#39;v2&#39;)
table.NG50 &lt;- assembly_stats.meta %&gt;% 
  filter(stat==&#39;NGx&#39;, v1==50) %&gt;% 
  arrange(desc(v2)) %&gt;% 
  mutate(v2=scales::label_comma()(v2),
         species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
  select(-v1, -contig, -Citation, -name) %&gt;% 
  select(Species=species, everything()) %&gt;% 
  dplyr::rename(NG50=&#39;v2&#39;)

## TODO:
# table.NScaff &lt;- assembly_stats.meta %&gt;% 
#   filter(stat==&#39;N&#39;, v1==&#39;Scaffold&#39;) %&gt;% 
#   arrange(desc(v2)) %&gt;% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
#   select(-v1, -contig, -Citation, -name) %&gt;% 
#   select(species, everything()) %&gt;% 
#   dplyr::rename(Scaffolds=&#39;v2&#39;)
# table.NGap &lt;- assembly_stats.meta %&gt;% 
#   filter(stat==&#39;N&#39;, v1==&#39;Gap&#39;) %&gt;% 
#   arrange(desc(v2)) %&gt;% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
#   select(-v1, -contig, -Citation, -name) %&gt;% 
#   select(species, everything()) %&gt;% 
#   dplyr::rename(Gaps=&#39;v2&#39;)
# table.NST2T &lt;- assembly_stats.meta %&gt;% 
#   filter(stat==&#39;N&#39;, v1==&#39;Scaffolds_T2T&#39;) %&gt;% 
#   arrange(desc(v2)) %&gt;% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
#   select(-v1, -contig, -Citation, -name) %&gt;% 
#   select(species, everything()) %&gt;% 
#   dplyr::rename(Scaffolds_T2T=&#39;v2&#39;)
# table.NCT2T &lt;- assembly_stats.meta %&gt;% 
#   arrange(desc(v2)) %&gt;% 
#   mutate(v2=scales::label_comma()(v2),
#          species= species %&gt;% str_replace_all(&#39;_&#39;, &quot; &quot;)) %&gt;%
#   select(-v1, -contig, -Citation, -name) %&gt;% 
#   select(species, everything()) %&gt;% 
#   dplyr::rename(Complete_Scaffolds=&#39;v2&#39;)

table.stats &lt;- table.auNG %&gt;%
  # full_join(table.NScaff) %&gt;% 
  # full_join(table.NGap) %&gt;% 
  # full_join(table.NST2T) %&gt;% 
  # full_join(table.NCT2T) %&gt;% 
  full_join(table.NG50) %&gt;% 
  full_join(table.LG50)</code></pre>
<pre><code>Joining with `by = join_by(Species, stat, Origin)`
Joining with `by = join_by(Species, stat, Origin)`</code></pre>
</div>
<div id="notes-about-the-tree" class="section level3">
<h3>Notes about the tree</h3>
<p>In order to get TimeTree to play nicely with the phylogeny, a few
changes were needed: - Murina aurata feae (replaced with Murina aurata)
- Pteropus pselaphon (replaced with Pteropus mariannus) - Miniopterus
schreibersii (replace with Miniopterus schreibersii orianae) - Myotis
occultus (replaced with Myotis lucifugus lucifugus) - Myotis lucifugus
(replaced with Myotis lucifugus carissima) - Hipposideros pendleburyi
(replaced with Hipposideros turpis) - TimetreeV5 has split Eptesicus and
broken its nomenclature/link to NCBI, so you can’t find “Eptesicus
fuscus” anymore. Instead, you have to look for all of Chiroptera or
Vespertilliodinae and then look for Eptesicus fuscus hispaniolae.</p>
<p>If you do all these changes, then undo the changes in post, you get
this tree. Note that it was very annoying to splice the two trees
together by eye so avoid needing to redo this.</p>
<pre class="r"><code>tr &lt;- treeio::read.newick(&quot;../data/trees/TimeTree.org/genomeSpecies.timetree.nwk&quot;)

# setdiff(tr$tip.label, assembly_stats.meta$species)
# setdiff(assembly_stats.meta$species, tr$tip.label)</code></pre>
</div>
</div>
<div id="ngx-curves" class="section level2">
<h2>NGx curves</h2>
<pre class="r"><code>plot.NGx &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;NGx&quot;,
         contig==T
         ) %&gt;% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x=&quot;NGx&quot;, y = &quot;bp&quot;, title = &quot;Contig Contiguity&quot;) + 
  geom_point() + 
  geom_line() + 
  ylim(1e4,NA) + 
  scale_color_brewer(palette = &quot;Dark2&quot;) + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.NGx</code></pre>
<pre><code>Warning: Removed 850 rows containing missing values (`geom_point()`).</code></pre>
<pre><code>Warning: Removed 850 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/ng_curves.Rmd/NGx-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-NGx-1">
Past versions of NGx-1.png
</button>
</p>
<div id="fig-NGx-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/NGx-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/NGx-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/figure/ng_curves.Rmd/NGx-1.png" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>col.source &lt;- c(
  &quot;This Paper&quot; = &quot;#FFBA08&quot;,
  Bat1K = &quot;#B33951&quot;,
  T2T = &quot;#d1b1c8&quot;,
  Zoonomia = &quot;#4357AD&quot;,
  Other = &quot;#6f7c12&quot;
)

plot.NGx.altCol.log &lt;- assembly_stats.meta %&gt;% 
  mutate(Origin = Origin %&gt;% factor(levels=names(col.source))) %&gt;% 
  filter(stat == &quot;NGx&quot;,
         contig==T
         ) %&gt;% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x=&quot;NGx&quot;, y = &quot;bp&quot;, title = &quot;Contig Contiguity&quot;) + 
  # geom_point() + 
  geom_line(linewidth=0.5) + 
  scale_y_log10(lim=c(1e4,NA))+
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.NGx.altCol.log</code></pre>
<pre><code>Warning: Removed 850 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/ng_curves.Rmd/NGx%20alt%20log-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-NGx-alt-log-1">
Past versions of "NGx alt log-1.png"
</button>
</p>
<div id="fig-no-spaces-NGx-alt-log-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/NGx alt log-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/NGx alt log-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot.NGx.altCol &lt;- assembly_stats.meta %&gt;% 
  mutate(Origin = Origin %&gt;% factor(levels=names(col.source))) %&gt;% 
  filter(stat == &quot;NGx&quot;,
         contig==T
         ) %&gt;% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x=&quot;NGx&quot;, y = &quot;bp&quot;, title = &quot;Contig Contiguity&quot;) + 
  # geom_point() + 
  geom_line(linewidth=0.5) + 
  ylim(1e4,NA) +
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.NGx.altCol</code></pre>
<pre><code>Warning: Removed 850 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/ng_curves.Rmd/NGx%20alt%20linear-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-NGx-alt-linear-1">
Past versions of "NGx alt linear-1.png"
</button>
</p>
<div id="fig-no-spaces-NGx-alt-linear-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/NGx alt linear-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/NGx alt linear-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot.NGx.altCol %&gt;% ggsave(plot=., filename = &quot;../output/NGx_curves.pdf&quot;, width = 6, height = 3.71, dpi=900, units = &quot;in&quot;)</code></pre>
<pre class="r"><code>assembly_stats.plotme &lt;- assembly_stats.meta %&gt;% 
  mutate(Origin = Origin %&gt;% factor(levels=names(col.source))) %&gt;% 
  filter(stat == &quot;NGx&quot;,
         contig==T
         )
plot.NGx.altCol.t2tzoo &lt;- assembly_stats.plotme %&gt;% 
  ggplot(aes(x=v1, y=v2, color=Origin, group=name)) + 
  labs(x=&quot;NGx&quot;, y = &quot;bp&quot;, title = &quot;Contig Contiguity&quot;) + 
  # geom_point() + 
  geom_line(
    data=assembly_stats.plotme %&gt;% filter(Origin %in% c(&quot;T2T&quot;, &quot;Zoonomia&quot;)),
    linewidth=0.5
  ) + 
  ylim(1e4,NA) +
  scale_color_manual(values=col.source) + 
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.NGx.altCol.t2tzoo</code></pre>
<pre><code>Warning: Removed 333 rows containing missing values (`geom_line()`).</code></pre>
<p><img src="figure/ng_curves.Rmd/onlyT2T-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-onlyT2T-1">
Past versions of onlyT2T-1.png
</button>
</p>
<div id="fig-onlyT2T-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/onlyT2T-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/onlyT2T-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot.NGx.altCol.t2tzoo %&gt;% ggsave(plot=., filename = &quot;../output/NGx_curves_onlyT2TZoonomia.pdf&quot;, width = 6, height = 3.71, dpi=900, units = &quot;in&quot;)</code></pre>
</div>
<div id="aung" class="section level2">
<h2>auNG</h2>
<pre class="r"><code>plot.auNG &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;auNG&quot;,
         contig==T
         ) %&gt;% 
  arrange(desc(v2)) %&gt;% 
  mutate(name = factor(name, levels = name %&gt;% unique)) %&gt;% 
  ggplot(aes(x=v2, y=name, color=Origin)) + 
  labs(x=&quot;auNG&quot;, y = &quot;Assembly&quot;, title = &quot;auNG&quot;) + 
  geom_point() + 
  scale_color_brewer(palette = &quot;Dark2&quot;) + 
  # scale_y_log10(limits=c(1e0,NA)) +
  # scale_y_continuous(limits=c(1e0,NA)) +
  theme_pubclean() + 
  labs_pubr() + 
  theme(legend.position = &quot;bottom&quot;)
plot.auNG</code></pre>
<p><img src="figure/ng_curves.Rmd/AuNG-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-AuNG-1">
Past versions of AuNG-1.png
</button>
</p>
<div id="fig-AuNG-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/AuNG-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/AuNG-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/figure/ng_curves.Rmd/AuNG-1.png" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tree-plus-aung" class="section level2">
<h2>Tree plus auNG</h2>
<pre class="r"><code>auNG &lt;- assembly_stats.meta %&gt;% 
  filter(stat == &quot;auNG&quot;,
         contig==T
         ) %&gt;% 
  select(species, Origin, v2, v1) %&gt;% 
  rename(label=species, auNG=v2, genomeSize=v1) %&gt;% 
  filter(!(str_detect(label, &quot;lucifugus&quot;) &amp; Origin==&quot;Other&quot;)) %&gt;% 
  # a bug lead to an accidental dup when there was both a contig and a scaffold
  group_by(label, Origin) %&gt;% 
  filter(auNG==max(auNG)) %&gt;% 
  ungroup
  

color_scale_origin = RColorBrewer::brewer.pal(n=auNG %&gt;% pull(Origin) %&gt;% unique %&gt;% length, name=&quot;Dark2&quot;) %&gt;% set_names(., auNG %&gt;% pull(Origin) %&gt;% unique)

# tr.auNG &lt;- full_join(tr,auNG)

## Sometimes I hate ggtree
nodes_ourgenomes &lt;- tr %&gt;% 
  as_tibble %&gt;% 
  filter(label %in% our_genomes) %&gt;% 
  select(label, node) %&gt;% 
  deframe()</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.tr.auNG &lt;- tr %&gt;% 
  ggtree() + 
  geom_tiplab(
    aes(
      subset = (node %in% nodes_ourgenomes)
    ),
    align = T,
    fontface=&quot;bold&quot;
    ) +
  geom_tiplab(
    aes(
      subset = !(node %in% nodes_ourgenomes)
    ),
    align = T
    ) +
  xlim_tree(c(NA,250)) +
  geom_facet(
    panel = &quot;auNG (bar)&quot;,
    data = auNG,
    geom = geom_col,
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = &#39;y&#39;,
    width = .6
  )+
  geom_facet(
    panel = &quot;auNG (point)&quot;, 
    data = auNG, 
    geom = geom_point, 
    mapping=aes(
      # x = log(1),
      x = log(auNG),
      color=Origin,
      fill=Origin
    ), 
    size=2
  )+
  theme_tree2() +
  scale_color_manual(&quot;Origin&quot;, values = color_scale_origin) +
  scale_fill_manual(&quot;Origin&quot;, values = color_scale_origin) +
  # geom_facet(panel = &quot;Genome Size&quot;, data = auNG %&gt;% select(label, genomeSize, Origin), geom = geom_point, 
  #              mapping=aes(x = genomeSize, color=Origin))+ 
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent,node.
ℹ invalid tbl_tree object. Missing column: parent,node.
ℹ invalid tbl_tree object. Missing column: parent,node.
ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre class="r"><code>p.tr.auNG</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20plus%20auNG-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-phylo-plus-auNG-1">
Past versions of "phylo plus auNG-1.png"
</button>
</p>
<div id="fig-no-spaces-phylo-plus-auNG-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/phylo plus auNG-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/phylo plus auNG-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/figure/ng_curves.Rmd/phylo plus auNG-1.png" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>auNG.fancy &lt;- auNG %&gt;% 
  mutate(
    label_fancy = sapply(label, . %&gt;% if_else(. %in% our_genomes, str_c(&quot;**&quot;,.,&quot;**&quot;), .)) %&gt;% 
      str_remove(&quot;_mesoamericanus&quot;) %&gt;% 
      str_replace_all(&quot;_&quot;, &quot; &quot;)
  )</code></pre>
<pre class="r"><code>angle_rotate &lt;- function(angle){
    # subset1 &lt;- &quot;(angle &lt; 90 | angle &gt; 270)&quot;
    # subset2 &lt;- &quot;(angle &gt;= 90 &amp; angle &lt;=270)&quot;
    if (angle &lt; 90 | angle &gt; 270){
      return(angle)
    } else {
      return(angle+180)
    }
}
h_readjust &lt;- function(angle){
    if (angle &lt; 90 | angle &gt; 270){
      return(0)
    } else {
      return(1)
    }
}

p.tr.auNG.circ &lt;-
  tr %&gt;%
  ggtree(layout = &quot;fan&quot;, open.angle=180) +
  xlim_tree(c(NA,100))+ 
  geom_fruit(
    data = auNG.fancy,
    geom = geom_col,
    mapping=aes(
      y= label,
      x = log(auNG),
      color=Origin,
      fill=Origin
    ),
    orientation = &#39;y&#39;,
    width = .6
  ) +
  geom_fruit(
    data = auNG.fancy,
    geom = geom_richtext, 
    mapping = aes(
      y=label,
      label=label_fancy,
      color=Origin,
      angle=sapply(angle,angle_rotate),
      hjust=sapply(angle,h_readjust)
    ),
    fill = NA, 
    label.color = NA, 
    label.padding = grid::unit(rep(0, 4), &quot;pt&quot;),
    
  ) + 
  scale_color_manual(&quot;Origin&quot;, values = color_scale_origin, guide=&quot;none&quot;) +
  scale_fill_manual(&quot;Origin&quot;, values = color_scale_origin,
                    guide = guide_legend(nrow = 2, ncol=6, title.position = &quot;left&quot;)) +
  theme(legend.position = c(0.6,0.3), plot.margin = unit(c(4,0,0,0),&quot;cm&quot;))</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.
ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre class="r"><code>p.tr.auNG.circ</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20auNG%20circular-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-phylo-auNG-circular-1">
Past versions of "phylo auNG circular-1.png"
</button>
</p>
<div id="fig-no-spaces-phylo-auNG-circular-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular-1.png" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>species_color = ggsci::pal_d3(palette = &quot;category20&quot;)(length(our_genomes)+1) %&gt;% 
  set_names(., c(our_genomes, &quot;Other&quot;))

species_color[&quot;Myotis_evotis&quot;] = &quot;#17BECFFF&quot;
species_color[&quot;Other&quot;] = &quot;#7F7F7FFF&quot;

auNG.fancy.color &lt;- auNG.fancy %&gt;% 
  mutate(color_me = sapply(label, . %&gt;% ifelse(. %in% our_genomes, ., &quot;Other&quot;)))</code></pre>
<pre class="r"><code>p.tr.auNG.circ.altcolor &lt;-
  tr %&gt;%
  ggtree(layout = &quot;fan&quot;, open.angle=180) +
  xlim_tree(c(NA,100))+ 
  # geom_fruit_list(
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      orientation = &#39;y&#39;,
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA, 
      # offset=0.2,
      label.padding = grid::unit(rep(0,4), &quot;pt&quot;),
    ) + 
  # ) +
  scale_color_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  scale_fill_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  theme(plot.margin = unit(c(4,0,0,0),&quot;cm&quot;))</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.
ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre class="r"><code>p.tr.auNG.circ.altcolor</code></pre>
<p><img src="figure/ng_curves.Rmd/phylo%20auNG%20circular%20map%20color-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-phylo-auNG-circular-map-color-1">
Past versions of "phylo auNG circular map color-1.png"
</button>
</p>
<div id="fig-no-spaces-phylo-auNG-circular-map-color-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular map color-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular map color-1.png" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/5848dd7f7eac78b5bbb781b0b101adaa002a6faa/R_workflowr/docs/figure/ng_curves.Rmd/phylo auNG circular map color-1.png" target="_blank">5848dd7</a>
</td>
<td>
Juan M Vazquez (docmanny)
</td>
<td>
2023-01-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>t2t.base = p.tr.auNG.circ.altcolor$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(x)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>t2t.bartop = p.tr.auNG.circ.altcolor$layers[[4]]$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(new_xtmp) +
  p.tr.auNG.circ.altcolor$layers[[4]]$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(x)
t2t.start = min(p.tr.auNG.circ.altcolor$data$y)
t2t.end = max(p.tr.auNG.circ.altcolor$data$y)

p.tr.auNG.circ.altcolor.withLine &lt;- p.tr.auNG.circ.altcolor + 
  geom_segment(
    aes(
      x = t2t.base + t2t.bartop,
      xend = t2t.base + t2t.bartop,
      y = t2t.start, 
      yend = t2t.end
    )
  )</code></pre>
</div>
<div id="updated-phylo" class="section level2">
<h2>Updated Phylo</h2>
<pre class="r"><code>tr.536 &lt;- read.tree(&#39;../data/trees/536Mammals-rooted-nodeNames.nwk&#39;)

tr.536$tip.label &lt;- tr.536$tip.label %&gt;% 
            str_replace(&#39;Canis_lupus_familiaris&#39;, &#39;Canis_familiaris&#39;) #%&gt;% 
            # 

auNG.fancy.color$label &lt;- auNG.fancy.color$label %&gt;% 
  str_replace(&#39;Murina_aurata_feae&#39;, &#39;Murina_aurata&#39;) %&gt;% 
  str_replace(&#39;Pteronotus_parnellii_mesoamericanus&#39;, &#39;Pteronotus_parnellii&#39;)
  
t.tr.536 &lt;- tr.536 %&gt;% as_tibble

node.chiroptera = t.tr.536%&gt;% filter(label==&#39;Chiroptera&#39;) %&gt;% pull(node)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tipnodes.chiroptera = tidytree::offspring(
  tr.536, 
  .node=node.chiroptera, 
  type = &#39;tips&#39;
)

tipnodes.chiroptera &lt;- t.tr.536 %&gt;% 
    filter(node %in% tipnodes.chiroptera) %&gt;% 
    pull(label)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tips = c(
  # t.tr.536 %&gt;% 
  #   filter(node %in% tipnodes.chiroptera) %&gt;% 
  #   pull(label),
  tipnodes.chiroptera,
  &quot;Homo_sapiens&quot;,
  &quot;Mus_musculus&quot;,
  &quot;Canis_familiaris&quot;,
  &quot;Bos_taurus&quot;
)

tr.genomes &lt;- tr.536 %&gt;% 
  keep.tip(
    tips
  )</code></pre>
<pre class="r"><code>p.trNew.auNG.circ.altcolor &lt;-
  tr.genomes %&gt;%
  ggtree(layout = &quot;fan&quot;, open.angle=180, branch.length = &#39;none&#39;) +
  # xlim_tree(c(NA,10))+ 
  # geom_fruit_list(
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_col,
      mapping=aes(
        y= label,
        x = log(auNG),
        color=color_me,
        fill=color_me
      ),
      orientation = &#39;y&#39;,
      width = .6
    ) + 
    geom_fruit(
      data = auNG.fancy.color,
      geom = geom_richtext, 
      mapping = aes(
        y=label,
        label=label_fancy,
        color=color_me,
        angle=sapply(angle,angle_rotate),
        hjust=sapply(angle,h_readjust)
      ),
      fill = NA, 
      label.color = NA,
      fontface=&#39;italic&#39;,
      size=8/.pt,
      # offset=0.2,
      label.padding = grid::unit(rep(0,4), &quot;pt&quot;),
    ) + 
  # ) +
  scale_color_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  scale_fill_manual(&quot;Origin&quot;, values = species_color, guide=&quot;none&quot;) +
  theme(plot.margin = unit(c(4,0,0,0),&quot;cm&quot;))</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.
ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.

ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre class="r"><code>t2t.base = p.trNew.auNG.circ.altcolor$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(x)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>t2t.bartop = p.trNew.auNG.circ.altcolor$layers[[3]]$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(new_xtmp) +0.6#+   p.trNew.auNG.circ.altcolor$layers[[3]]$data %&gt;% filter(label == &#39;Homo_sapiens&#39;) %&gt;% pull(x)
t2t.start = min(p.trNew.auNG.circ.altcolor$data$y)
t2t.end = max(p.trNew.auNG.circ.altcolor$data$y)

p.trNew.auNG.circ.altcolor.withLine &lt;- p.trNew.auNG.circ.altcolor + 
  geom_segment(
    aes(
      x = t2t.base + t2t.bartop,
      xend = t2t.base + t2t.bartop,
      y = t2t.start, 
      yend = t2t.end
    ),
    linewidth=0.1, 
    # lty=&#39;dashed&#39;
  )

p.trNew.auNG.circ.altcolor.withLine</code></pre>
<p><img src="figure/ng_curves.Rmd/newPhylo-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-newPhylo-1">
Past versions of newPhylo-1.png
</button>
</p>
<div id="fig-newPhylo-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/d0ca892e789b96accec0b454ff063589f25cf47a/R_workflowr/docs/figure/ng_curves.Rmd/newPhylo-1.png" target="_blank">d0ca892</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># g &lt;- ggplotGrob(p.trNew.auNG.circ.altcolor.withLine)

p.trNew.auNG.circ.altcolor.withLine %&gt;% 
  ggsave(
    filename = &#39;../output/subfigs/phylo_auNG_circ_col_withLine.pdf&#39;,
    plot= ., 
    width = 7.5, 
    height = 4, 
    units = &#39;in&#39;, 
    dpi = 300
  )</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: America/La_Paz
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ragg_1.2.5        gridExtra_2.3     ggsci_3.0.0       ggtext_0.1.2     
 [5] ggtreeExtra_1.8.1 tidytree_0.4.5    ggtree_3.6.2      patchwork_1.1.3  
 [9] ggpubr_0.6.0      magrittr_2.0.3    ape_5.7-1         lubridate_1.9.2  
[13] forcats_1.0.0     stringr_1.5.0     dplyr_1.1.3       purrr_1.0.2      
[17] readr_2.1.4       tidyr_1.3.0       tibble_3.2.1      ggplot2_3.4.3    
[21] tidyverse_2.0.0  

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   farver_2.1.1       fastmap_1.1.1      lazyeval_0.2.2    
 [5] promises_1.2.1     digest_0.6.33      timechange_0.2.0   lifecycle_1.0.3   
 [9] compiler_4.3.1     rlang_1.1.1        sass_0.4.7         tools_4.3.1       
[13] utf8_1.2.3         yaml_2.3.7         knitr_1.44         ggsignif_0.6.4    
[17] labeling_0.4.3     bit_4.0.5          RColorBrewer_1.1-3 xml2_1.3.5        
[21] aplot_0.2.1        abind_1.4-5        workflowr_1.7.1    withr_2.5.0       
[25] grid_4.3.1         fansi_1.0.5        git2r_0.32.0       colorspace_2.1-0  
[29] scales_1.2.1       cli_3.6.1          crayon_1.5.2       rmarkdown_2.25    
[33] treeio_1.25.4      generics_0.1.3     rstudioapi_0.15.0  tzdb_0.4.0        
[37] commonmark_1.9.0   cachem_1.0.8       parallel_4.3.1     ggplotify_0.1.2   
[41] vctrs_0.6.3        yulab.utils_0.1.0  jsonlite_1.8.7     carData_3.0-5     
[45] car_3.1-2          gridGraphics_0.5-1 hms_1.1.3          bit64_4.0.5       
[49] rstatix_0.7.2      magick_2.8.1       systemfonts_1.0.4  ggnewscale_0.4.9  
[53] jquerylib_0.1.4    glue_1.6.2         stringi_1.7.12     gtable_0.3.4      
[57] later_1.3.1        munsell_0.5.0      pillar_1.9.0       htmltools_0.5.6   
[61] R6_2.5.1           textshaping_0.3.6  rprojroot_2.0.3    vroom_1.6.3       
[65] evaluate_0.21      lattice_0.21-8     markdown_1.8       backports_1.4.1   
[69] gridtext_0.1.5     memoise_2.0.1      broom_1.0.5        httpuv_1.6.11     
[73] ggfun_0.1.3        bslib_0.5.1        Rcpp_1.0.11        nlme_3.1-163      
[77] whisker_0.4.1      xfun_0.40          fs_1.6.3           pkgconfig_2.0.3   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
