<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="docmanny" />

<meta name="date" content="2023-10-23" />

<title>RICR</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myotis Genome Analyses</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RICR</h1>
<h4 class="author">docmanny</h4>
<h4 class="date">2023-10-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>R_workflowr/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree9d4271fb67db5a54ed7df5703ae96751ebb342cetargetblank9d4271fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/9d4271fb67db5a54ed7df5703ae96751ebb342ce" target="_blank">9d4271f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabMyotisGenomeAssemblytree9d4271fb67db5a54ed7df5703ae96751ebb342cetargetblank9d4271fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/tree/9d4271fb67db5a54ed7df5703ae96751ebb342ce" target="_blank">9d4271f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    R_workflowr/.RData
    Ignored:    R_workflowr/.Rproj.user/
    Ignored:    R_workflowr/analysis/figure/
    Ignored:    R_workflowr/data/
    Ignored:    R_workflowr/output/pubFiles/
    Ignored:    R_workflowr/output/webGestalt/
    Ignored:    R_workflowr/renv/library/
    Ignored:    R_workflowr/renv/staging/
    Ignored:    R_workflowr/test.phy

Unstaged changes:
    Modified:   R_workflowr/analysis/RICR.Rmd
    Modified:   R_workflowr/output/draft_figs/Myotis_size_life_LQ_upham.pdf
    Modified:   R_workflowr/output/subfigs/phylo_auNG_circ_col_withLine.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>R_workflowr/analysis/RICR.Rmd</code>) and
HTML (<code>R_workflowr/docs/RICR.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/1a4123ee985328e7ade6f18d00a63669ec0f1e38/R_workflowr/analysis/RICR.Rmd" target="_blank">1a4123e</a>
</td>
<td>
docmanny
</td>
<td>
2024-01-08
</td>
<td>
small changes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/8f66d05c66fd494567461d74f4110a0876566489/R_workflowr/analysis/RICR.Rmd" target="_blank">8f66d05</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-21
</td>
<td>
clean version of Fig 1, now with LQ
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/77bd637f40e2cced911c2e548ddda6f92674f59b/R_workflowr/analysis/RICR.Rmd" target="_blank">77bd637</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-19
</td>
<td>
added more figs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sudmantlab/MyotisGenomeAssembly/blob/3d9fd9c8d0d597a96d92151d9d96dbea6c34799e/R_workflowr/analysis/RICR.Rmd" target="_blank">3d9fd9c</a>
</td>
<td>
docmanny
</td>
<td>
2023-12-15
</td>
<td>
draft sub figs for F1&amp;2
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>So, huge caveat - in order to use the full power of the data I have,
I had to use a different time-calibrated tree (Upham <em>et al</em>,
2019) that has everyone for this analysis. Thus, the relatedness between
bats is messed up because they don’t have our HQ, genome-wide phylo.</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(magrittr)</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;

The following object is masked from &#39;package:purrr&#39;:

    set_names

The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre class="r"><code>library(ape)</code></pre>
<pre><code>
Attaching package: &#39;ape&#39;

The following object is masked from &#39;package:dplyr&#39;:

    where</code></pre>
<pre class="r"><code>library(tidytree)</code></pre>
<pre><code>If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Shuangbin Xu, Lin Li, Xiao Luo, Meijun Chen, Wenli Tang, Li Zhan, Zehan
Dai, Tommy T. Lam, Yi Guan, Guangchuang Yu. Ggtree: A serialized data
object for visualization of a phylogenetic tree and annotation data.
iMeta 2022, 1(4):e56. doi:10.1002/imt2.56

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

Attaching package: &#39;tidytree&#39;

The following objects are masked from &#39;package:ape&#39;:

    drop.tip, keep.tip

The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;

The following object is masked from &#39;package:ape&#39;:

    rotate</code></pre>
<pre class="r"><code>library(ggdist)
library(ggtree)</code></pre>
<pre><code>ggtree v3.6.2 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

Shuangbin Xu, Lin Li, Xiao Luo, Meijun Chen, Wenli Tang, Li Zhan, Zehan
Dai, Tommy T. Lam, Yi Guan, Guangchuang Yu. Ggtree: A serialized data
object for visualization of a phylogenetic tree and annotation data.
iMeta 2022, 1(4):e56. doi:10.1002/imt2.56

G Yu. Data Integration, Manipulation and Visualization of Phylogenetic
Trees (1st ed.). Chapman and Hall/CRC. 2022. ISBN: 9781032233574

Attaching package: &#39;ggtree&#39;

The following object is masked from &#39;package:ggpubr&#39;:

    rotate

The following object is masked from &#39;package:ape&#39;:

    rotate

The following object is masked from &#39;package:magrittr&#39;:

    inset

The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre class="r"><code>library(tidytree)
library(broom)
library(ggsci)
library(ggrepel)
library(kableExtra)</code></pre>
<pre><code>
Attaching package: &#39;kableExtra&#39;

The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<pre class="r"><code>library(rphylopic)</code></pre>
<pre><code>Thank you for using rphylopic! Don&#39;t forget to attribute the silhouettes that you use to their contributors using the `get_attribution()` function and/or the `verbose` argument in the `geom_phylopic()`, `add_phylopic()`, and `add_phylopic_base()` functions. We would also appreciate it if you cite this package in your work (use `citation(&quot;rphylopic&quot;)`).</code></pre>
<pre class="r"><code>library(ggnewscale)
library(patchwork)
library(aplot)</code></pre>
<pre class="r"><code>## StackOverflow: 
# https://stackoverflow.com/questions/7549694/add-regression-line-equation-and-r2-on-graph


lm_eqn = function(m) {

  l &lt;- list(a = format(coef(m)[1], digits = 2),
      b = format(abs(coef(m)[2]), digits = 2),
      r2 = format(summary(m)$r.squared, digits = 3));

  if (coef(m)[2] &gt;= 0)  {
    eq &lt;- substitute(italic(y) == a + b %.% italic(x)*&quot;, &quot;~~italic(r)^2~&quot;=&quot;~r2,l)
  } else {
    eq &lt;- substitute(italic(y) == a - b %.% italic(x)*&quot;, &quot;~~italic(r)^2~&quot;=&quot;~r2,l)    
  }

  as.character(as.expression(eq));                 
}</code></pre>
<pre class="r"><code>our_genomes = c(
  &quot;Myotis_auriculus&quot;,
  &quot;Myotis_californicus&quot;,
  &quot;Myotis_occultus&quot;,
  &quot;Myotis_lucifugus&quot;,
  &quot;Myotis_yumanensis&quot;,
  &quot;Myotis_volans&quot;,
  &quot;Myotis_velifer&quot;,
  &quot;Myotis_evotis&quot;,
  &quot;Myotis_thysanodes&quot;
)

tt.upham &lt;- read_tsv(&#39;../data/trees/UphamEtAl2019_harmonizednames.tsv&#39;) %&gt;% 
  select(newlab = label, label=submitted_name)</code></pre>
<pre><code>Rows: 4100 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (3): submitted_name, matched_name, label

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>tr.upham &lt;- read.tree(&#39;../data/trees/UphamEtAl2019_harmonizednames.nwk&#39;)
df.lq &lt;- read_tsv(&#39;../data/lifehistory/manny_mammal_agedata_harmonized.withLQ_hasAli.tsv&#39;)</code></pre>
<pre><code>Rows: 5463 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): label, Order
dbl (4): size, lifespan, predLife, LQ
lgl (1): has.ali

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># cancerSucceptibility.table &lt;- read_csv(&#39;../data/stableTraits/cancerRisk-UphamEtAl2019WithBS_Run1_2.csv&#39;)
# sizetable &lt;- read_csv(&#39;../data/stableTraits/bodysize-UphamEtAl2019_harmonizednames_withSizeLife_size_concat.csv&#39;) %&gt;% 
#   rename(
#     bs.branch.length=branch.length, 
#     bs.brownian=Brownian, 
#     bs.median=Median, 
#     bs.low=X95.CI_Low, 
#     bs.high=X95.CI_High
#   )
# lifetable &lt;- read_csv(&#39;../data/stableTraits/life-UphamEtAl2019_harmonizednames_withSizeLife_lifespan_concat.csv&#39;) %&gt;% 
#   rename(
#     lifespan.branch.length=branch.length, 
#     lifespan.brownian=Brownian, 
#     lifespan.median=Median, 
#     lifespan.low=X95.CI_Low, 
#     lifespan.high=X95.CI_High
#   )</code></pre>
</div>
<div id="more-lq" class="section level2">
<h2>More LQ</h2>
<pre class="r"><code>p.prediction &lt;- df.lq %&gt;%
  filter(!is.na(Order),
         !is.na(predLife),
         !is.na(lifespan)) %&gt;%
  group_by(Order) %&gt;%
  filter(n_distinct(label) &gt; 20) %&gt;%
  ungroup %&gt;%
  ggplot(
    aes(
      x=lifespan,
      y=predLife,
      color=Order
    )
  )+
  geom_point() +
  labs(x=&quot;Lifespan (yrs)&quot;,
       y=&quot;Predicted Lifespan (yrs)&quot;) +
  scale_color_d3(palette = &#39;category20&#39;) +
  facet_wrap(~Order, scales=&quot;free&quot;) +
  geom_smooth(method = &#39;lm&#39;) +
  theme_pubr() +
  labs_pubr() +
  theme(legend.position = &quot;none&quot;)
p.prediction</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/RICR.Rmd/lq%20prediction-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df.lq.lm &lt;- df.lq %&gt;% 
  filter(!is.na(Order),
         !is.na(predLife),
         !is.na(lifespan)) %&gt;% 
  group_by(Order) %&gt;% 
  filter(n_distinct(label) &gt; 20) %&gt;% 
  ungroup %&gt;%
  nest(data=-Order) %&gt;% 
  mutate(
    model = purrr::map(data, ~ lm(predLife ~ lifespan, .x)),
    lm.clean = purrr::map(model, tidy),
    lm.glance = purrr::map(model, glance),
    equation = sapply(model, lm_eqn)
  ) %&gt;% 
  unnest(lm.glance)


p.prediction.withEq &lt;- p.prediction + 
  geom_text(
    data= df.lq.lm, #%&gt;% 
      # select(Order, data, equation) %&gt;% 
      # unnest(data) %&gt;% 
      # group_by(Order) %&gt;% 
      # mutate(predLife = max(predLife),
      #        lifespan = min(lifespan)) %&gt;% 
      # select(Order, lifespan, predLife, equation) %&gt;% 
      # distinct %&gt;% 
      # ungroup,
    aes(
      x=-Inf,
      y=Inf,
      label=equation
    ),
    hjust=0,
    vjust=1,
    parse=T
  )

p.prediction.withEq</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/RICR.Rmd/manually%20add%20in%20equations%20to%20lq%20predictions-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>phylopic.codes &lt;- c(
  &#39;Artiodactyla&#39; = &#39;dc4c0e07-611f-41d0-b2fa-8c5bc3a22fba&#39;,
  &#39;Carnivora&#39; = &#39;b2dcd44f-1fb8-490e-b016-8279eb328f46&#39;,
  &#39;Didelphimorphia&#39; = &#39;162f2753-b795-4ecd-890a-2c6a2ff94ff6&#39;,
  &#39;Rodentia&#39; = &#39;c4572239-3b7c-4259-8049-3f44e6d52e6f&#39;,
  &#39;Primates&#39; = &#39;eccbb404-c99f-41f9-8785-01a7f57f1269&#39;,
  &#39;Diprotodontia&#39; = &#39;0904270e-b105-46e0-b81f-c4911d47d467&#39;,
  &#39;Cetacea&#39; = &#39;41158ddc-5aa9-41ef-99f9-99c3c10f1847&#39;,
  # &#39;Chiroptera&#39; = &#39;18bfd2fc-f184-4c3a-b511-796aafcc70f6&#39;,
  &#39;Chiroptera&#39; = &#39;21180755-3394-40bf-93eb-810954c0f7ba&#39;,
  &#39;Soricomorpha&#39; = &#39;f83c6893-f0ed-4ec4-b558-aa774c5c9b5b&#39;,
  &#39;Dasyuromorphia&#39; = &#39;9f505e8e-c431-4738-900d-600b99057b51&#39;
)


p.r2line &lt;- df.lq.lm %&gt;%
  mutate(uid = phylopic.codes[Order]) %&gt;% 
  arrange(r.squared) %&gt;% 
  mutate(y=0.02 %&gt;% multiply_by(rep(c(1,-1), times=5))) %&gt;%
  ggplot(
    aes(
      x=r.squared,
      y=0,
      color=Order,
      fill=Order
    )
  ) +
  geom_hline(yintercept=0) +
  geom_point(
    shape = &#39;|&#39;,
    size=8,
  ) +
  geom_phylopic(
    aes(
      y=y,
      uuid=uid
    ),
    size=0.03
  )  +
  # theme_void() +
   coord_fixed(
     ylim=c(-0.03,0.03),
     xlim=c(0,0.8)
   ) +
  labs(x=expression(paste(R^2))) + 
  theme_minimal() + 
  labs_pubr() + 
  theme(
    legend.position = &#39;none&#39;,
    axis.line.y=element_blank(),
    axis.text.y=element_blank(),
    axis.ticks.y=element_blank(),
    axis.title.y=element_blank(),
    panel.grid = element_blank()
  )

p.r2line</code></pre>
<p><img src="figure/RICR.Rmd/R2%20line%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="stabletraits" class="section level2">
<h2>StableTraits</h2>
<pre class="r"><code>life.anc &lt;- read_tsv(&#39;../data/stableTraits/UphamEtAl2019_harmonizednames_withSizeLifeLn_life_concat.ancstates&#39;) %&gt;% 
      filter(!str_starts(Parameter, &#39;stable&#39;)) %&gt;% 
  rename(
    label=Parameter,
    lnLife.brownian=Brownian, 
    lnLife.median=Median, 
    lnLife.low=`95%CI_Low`, 
    lnLife.high=`95%CI_High`
  )</code></pre>
<pre><code>Rows: 2315 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): Parameter
dbl (4): Brownian, Median, 95%CI_Low, 95%CI_High

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>tr.life &lt;- read.tree(&#39;../data/stableTraits/UphamEtAl2019_harmonizednames_withSizeLifeLn_life_concat.sqchange_tree&#39;) #%&gt;% 
  # as.treedata() %&gt;% 
  # full_join(
  #   life.anc,
  #   by=c(label=&#39;Parameter&#39;)
  # )

tr.size &lt;- read.tree(&#39;../data/stableTraits/UphamEtAl2019_harmonizednames_withSizeLifeLn_size_concat.sqchange_tree&#39;)

size.anc &lt;- read_tsv(&#39;../data/stableTraits/UphamEtAl2019_harmonizednames_withSizeLifeLn_size_concat.ancstates&#39;) %&gt;% 
      filter(!str_starts(Parameter, &#39;stable&#39;)) %&gt;% 
  rename(
    label=Parameter,
    lnSize.brownian=Brownian, 
    lnSize.median=Median, 
    lnSize.low=`95%CI_Low`, 
    lnSize.high=`95%CI_High`
  )</code></pre>
<pre><code>Rows: 2315 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (1): Parameter
dbl (4): Brownian, Median, 95%CI_Low, 95%CI_High

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>tr.og.size &lt;- read.tree(&#39;../data/stableTraits/UphamEtAl2019_harmonizednames_withSizeLifeLn_size_concat.tree&#39;)

# shared = intersect(tr.life$tip.label, tr.size$tip.label)
# 
# length(shared)
# length(tr.life$tip.label)
# length(tr.size$tip.label)
# 
# comparison &lt;- comparePhylo(
#   x=tr.life,
#   y=tr.size
# )

tr.life %&gt;% as_tibble %&gt;% full_join(tr.size %&gt;% as_tibble, by=&#39;label&#39;) %&gt;% filter(parent.x != parent.y)</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre><code># A tibble: 0 × 7
# ℹ 7 variables: parent.x &lt;int&gt;, node.x &lt;int&gt;, branch.length.x &lt;dbl&gt;,
#   label &lt;chr&gt;, parent.y &lt;int&gt;, node.y &lt;int&gt;, branch.length.y &lt;dbl&gt;</code></pre>
<pre class="r"><code>tr.sizedata &lt;- tr.size %&gt;% 
  full_join(size.anc)</code></pre>
<pre><code>Joining with `by = join_by(label)`</code></pre>
<pre class="r"><code>tr.lifedata &lt;- tr.life %&gt;% 
  full_join(life.anc)</code></pre>
<pre><code>Joining with `by = join_by(label)`</code></pre>
<pre class="r"><code>tr.sizelife &lt;- tr.og.size %&gt;% 
  as.treedata %&gt;% 
  full_join(
    size.anc
  ) %&gt;%
  full_join(
    life.anc
  ) %&gt;% 
  full_join(
    tr.lifedata %&gt;% as_tibble %&gt;% select(label,life.branch.length=branch.length)
  ) %&gt;% 
  full_join(
    tr.sizedata %&gt;% as_tibble %&gt;% select(label,size.branch.length=branch.length)
  )</code></pre>
<pre><code>Joining with `by = join_by(label)`
Joining with `by = join_by(label)`
ℹ invalid tbl_tree object. Missing column: parent,node.
Joining with `by = join_by(label)`
ℹ invalid tbl_tree object. Missing column: parent,node.
Joining with `by = join_by(label)`</code></pre>
<pre class="r"><code>### NOTE: Idionycteris_phyllotis is an artifact
tr.sizedata &lt;- tr.sizedata %&gt;% 
  tidytree::drop.tip(&#39;Idionycteris_phyllotis&#39;)</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.lifedata &lt;- tr.lifedata %&gt;% 
  tidytree::drop.tip(&#39;Idionycteris_phyllotis&#39;)</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.sizelife &lt;- tr.sizelife %&gt;% 
  tidytree::drop.tip(&#39;Idionycteris_phyllotis&#39;)</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>lq.lm &lt;- lm(
  lifespan ~ size, 
  df.lq %&gt;% 
    filter(Order != &quot;Chiroptera&quot;) %&gt;% 
    select(size, lifespan) %&gt;% 
    mutate_all(log)
)

tr.sizelife.lq &lt;- augment(
  lq.lm, 
  newdata = tr.sizelife %&gt;% as_tibble %&gt;% mutate(size=lnSize.median)
) %&gt;% 
  mutate(predLife.median = .fitted) %&gt;% 
  select(-.fitted)

tr.sizelife.lq &lt;-   augment(
  lq.lm, 
  newdata = tr.sizelife.lq %&gt;% mutate(size=lnSize.low)
) %&gt;% 
  mutate(predLife.low = .fitted)  %&gt;% 
  select(-.fitted)

tr.sizelife.lq &lt;- augment(
  lq.lm, 
  newdata = tr.sizelife.lq %&gt;% mutate(size=lnSize.high)
) %&gt;% 
  mutate(predLife.high = .fitted) %&gt;% 
  select(-.fitted) 

tr.sizelife.lq &lt;- augment(
  lq.lm, 
  newdata = tr.sizelife.lq %&gt;% mutate(size=size.branch.length)
) %&gt;% 
  mutate(predLife.branch.length = .fitted) %&gt;% 
  select(-.fitted)%&gt;% 
  mutate(
    life.median = exp(lnLife.median), 
    life.low = exp(lnLife.low), 
    life.high = exp(lnLife.high), 
    predLife.median = exp(predLife.median), 
    predLife.low = exp(predLife.low), 
    predLife.high = exp(predLife.high), 
    LQ.median = life.median/predLife.median,
    LQ.low = life.median/predLife.low,
    LQ.high = life.median/predLife.high,
    LQ.branch.length = life.branch.length/predLife.branch.length
  ) %&gt;% 
  as.treedata()

tr.sizelife.lq@phylo$tip.label=tr.sizelife.lq@extraInfo %&gt;% filter(!str_detect(label.y, &#39;\\d&#39;)) %&gt;% pull(label.y)</code></pre>
</div>
<div id="range-plots" class="section level2">
<h2>Range Plots</h2>
<pre class="r"><code>size.scale &lt;- tr.sizedata %&gt;% as_tibble %&gt;% pull(lnSize.median) %&gt;% unique %&gt;% as.numeric() %&gt;% sort
size.scale.min &lt;- size.scale[1]
size.scale.max &lt;- size.scale[length(size.scale)]
get.size.color &lt;- function(n, scale = size.scale){
  scale.colors = viridis::viridis(n=length(scale)) %&gt;% 
    set_names(., as.character(scale))
  scale.colors[as.character(n)]
}

lifespan.scale &lt;- tr.lifedata %&gt;% as_tibble %&gt;% pull(lnLife.median) %&gt;% unique %&gt;% as.numeric() %&gt;% sort
lifespan.scale.min &lt;- lifespan.scale[1]
lifespan.scale.max &lt;- lifespan.scale[length(lifespan.scale)]
get.lifespan.color &lt;- function(n, scale = lifespan.scale){
  scale.colors = viridis::viridis(n=length(scale), option = &quot;C&quot;) %&gt;% 
    set_names(., as.character(scale))
  scale.colors[as.character(n)]
}

LQ.scale &lt;- tr.sizelife.lq %&gt;% as_tibble %&gt;% pull(LQ.median) %&gt;% unique %&gt;% as.numeric() %&gt;% sort
LQ.scale.min &lt;- LQ.scale[1]
LQ.scale.max &lt;- LQ.scale[length(LQ.scale)]
get.LQ.color &lt;- function(n, scale = LQ.scale){
  scale.colors = viridis::viridis(n=length(scale), option = &quot;C&quot;) %&gt;% 
    set_names(., as.character(scale))
  scale.colors[as.character(n)]
}

size.scale.expanded &lt;- c(
  tr.sizedata %&gt;% as_tibble %&gt;% pull(lnSize.median),
  tr.sizedata %&gt;% as_tibble %&gt;% pull(lnSize.low),
  tr.sizedata %&gt;% as_tibble %&gt;% pull(lnSize.high)
) %&gt;% unique %&gt;% as.numeric() %&gt;% sort
size.scale.expanded.min &lt;- size.scale.expanded[1]
size.scale.expanded.max &lt;- size.scale.expanded[length(size.scale.expanded)]

lifespan.scale.expanded &lt;- c(
  tr.lifedata %&gt;% as_tibble %&gt;% pull(lnLife.median),
  tr.lifedata %&gt;% as_tibble %&gt;% pull(lnLife.low),
  tr.lifedata %&gt;% as_tibble %&gt;% pull(lnLife.high)
)  %&gt;% unique %&gt;% as.numeric() %&gt;% sort
lifespan.scale.expanded.min &lt;- lifespan.scale.expanded[1]
lifespan.scale.expanded.max &lt;- lifespan.scale.expanded[length(lifespan.scale.expanded)]

LQ.scale.expanded &lt;- c(
  tr.sizelife.lq %&gt;% as_tibble %&gt;% pull(LQ.median),
  tr.sizelife.lq %&gt;% as_tibble %&gt;% pull(LQ.low),
  tr.sizelife.lq %&gt;% as_tibble %&gt;% pull(LQ.high)
)  %&gt;% unique %&gt;% as.numeric() %&gt;% sort
LQ.scale.expanded.min &lt;- lifespan.scale.expanded[1]
LQ.scale.expanded.max &lt;- lifespan.scale.expanded[length(lifespan.scale.expanded)]</code></pre>
<div id="body-size" class="section level3">
<h3>Body Size</h3>
<div id="bats" class="section level4">
<h4>Bats</h4>
<pre class="r"><code>p.bodysize.bats &lt;- tr.sizedata  %&gt;% 
  tidytree::tree_subset(MRCA(tr.sizedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(lnSize.median) %&gt;% 
  mutate(
    size.median = exp(lnSize.median),
    size.low = exp(lnSize.low),
    size.high = exp(lnSize.high)
  ) %&gt;% 
  mutate(label = factor(label, levels=label)) %&gt;% 
  ggplot(
    aes(
      y=label,
      # x=bs.median, 
      # xmin=bs.low,
      # xmax=bs.high
      x=size.median, 
      xmin=size.low,
      xmax=size.high
      )
  ) +
  geom_pointrange(size=0.5) +
  geom_point(aes(color=lnSize.median), size=1) +
  geom_point(aes(x=size.low, color=lnSize.low), size=1.5, stroke=2, pch=&quot;[&quot;) +
  geom_point(aes(x=size.high, color=lnSize.high), size=1.5, stroke=2, pch=&quot;]&quot;) +
  scale_color_distiller(limits=c(size.scale.expanded.min, size.scale.expanded.max), 
                        palette = &quot;Blues&quot;, 
                        direction = 1,
                        guide = guide_colorbar(
                          title=&quot;Body size (log)&quot;,
                          title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
                          title.position = &quot;top&quot;,
                          title.hjust = 0.5
                          )
                        ) +
  labs(
    x=&quot;Body size (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div id="myotis" class="section level4">
<h4>Myotis</h4>
<pre class="r"><code>p.bodysize.myotis &lt;- tr.sizedata  %&gt;% 
  tidytree::tree_subset(MRCA(tr.sizedata, &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(lnSize.median) %&gt;% 
  mutate(
    size.median = exp(lnSize.median),
    size.low = exp(lnSize.low),
    size.high = exp(lnSize.high)
  ) %&gt;% 
  mutate(label = factor(label, levels=label)) %&gt;% 
  ggplot(
    aes(
      y=label,
      # x=bs.median, 
      # xmin=bs.low,
      # xmax=bs.high
      x=lnSize.median, 
      xmin=lnSize.low,
      xmax=lnSize.high
      )
  ) +
  geom_pointrange(size=0.5, stroke=0.1) +
  geom_point(aes(color=lnSize.median), size=0.5) +
  # geom_point(aes(x=lnSize.low, color=lnSize.low), size=1.5, stroke=2, pch=&quot;[&quot;) +
  # geom_point(aes(x=lnSize.high, color=lnSize.high), size=1.5, stroke=2, pch=&quot;]&quot;) +
  scale_color_distiller(limits=c(size.scale.expanded.min, size.scale.expanded.max), 
                        palette = &quot;Blues&quot;, 
                        direction = 1,
                        guide = guide_colorbar(
                          title=&quot;Body size (log)&quot;,
                          title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
                          title.position = &quot;top&quot;,
                          title.hjust = 0.5
                          )
                        ) +
  labs(
    x=&quot;Body size (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.bodysize.myotis</code></pre>
<p><img src="figure/RICR.Rmd/range%20plot%20size%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lifespan" class="section level3">
<h3>Lifespan</h3>
<div id="bats-1" class="section level4">
<h4>Bats</h4>
<pre class="r"><code>p.lifespan.bats &lt;- tr.lifedata %&gt;%
  tidytree::tree_subset(MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(lnLife.median) %&gt;% 
  mutate(
    life.median = exp(lnLife.median),
    life.low = exp(lnLife.low),
    life.high = exp(lnLife.high)
  ) %&gt;% 
  mutate(label = factor(label, levels=label)) %&gt;% 
  ggplot(
    aes(
      y=label,
      # x=lifespan.median, 
      # xmin=lifespan.low,
      # xmax=lifespan.high
      x=life.median, 
      xmin=life.low,
      xmax=life.high
      )
  ) +
  geom_pointrange(size=0.5, stroke=0.5) +
  geom_point(aes(color=lnLife.median), size=0.5) +
  # geom_point(aes(x=life.low, color=lnLife.low), size=1.5, stroke=2, pch=&quot;[&quot;) +
  # geom_point(aes(x=life.high, color=lnLife.high), size=1.5, stroke=2, pch=&quot;]&quot;) +
  scale_color_distiller(
    palette = &quot;Reds&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (yrs)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  labs(
    x=&quot;Lifespan (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.lifespan.bats</code></pre>
<p><img src="figure/RICR.Rmd/rangePlot%20Lifespan%20-%20Bats-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="myotis-1" class="section level4">
<h4>Myotis</h4>
<pre class="r"><code>p.lifespan.myotis &lt;- tr.lifedata %&gt;%
  tidytree::tree_subset(MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(lnLife.median) %&gt;% 
  mutate(
    life.median = exp(lnLife.median),
    life.low = exp(lnLife.low),
    life.high = exp(lnLife.high)
  ) %&gt;% 
  mutate(label = factor(label, levels=label)) %&gt;% 
  ggplot(
    aes(
      y=label,
      # x=lifespan.median, 
      # xmin=lifespan.low,
      # xmax=lifespan.high
      x=life.median, 
      xmin=life.low,
      xmax=life.high
      )
  ) +
  geom_pointrange(size=0.5, stroke=0.5) +
  geom_point(aes(color=life.median), size=0.5) +
  # geom_point(aes(x=life.low, color=life.low), size=1.5, stroke=2, pch=&quot;[&quot;) +
  # geom_point(aes(x=life.high, color=life.high), size=1.5, stroke=2, pch=&quot;]&quot;) +
  scale_color_distiller(
    palette = &quot;Reds&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (yrs)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  labs(
    x=&quot;Lifespan (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.lifespan.myotis</code></pre>
<p><img src="figure/RICR.Rmd/rangePlot%20Lifespan%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lq" class="section level3">
<h3>LQ</h3>
<div id="bats-2" class="section level4">
<h4>Bats</h4>
</div>
<div id="myotis-2" class="section level4">
<h4>Myotis</h4>
<pre class="r"><code>### Lifespan ###

p.tree.time &lt;- tr.upham %&gt;% 
  ggtree(
    # aes(color=lnLife.median)
  ) + 
  geom_highlight(node=MRCA(tr.upham, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;gold&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.upham, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Chiroptera&#39;) + 
  geom_highlight(node=MRCA(tr.upham, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), fill=&#39;lightgreen&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.upham, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), label = &#39;Cetacea&#39;) + 
  geom_highlight(node=MRCA(tr.upham, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;)) +
  ggtree::geom_cladelab(node=MRCA(tr.upham, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;), label = &#39;Primata&#39;) + 
  geom_highlight(node=MRCA(tr.upham, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), fill=&#39;beige&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.upham, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), label = &#39;Paenungulata&#39;) + 
  theme_tree2()+
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) + 
  # scale_color_distiller(
  #   palette = &quot;Reds&quot;, direction = 1,
  #   guide = guide_colorbar(
  #     title=&quot;Lifespan (log(y))&quot;,
  #     title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
  #     title.position = &quot;top&quot;,
  #     title.hjust = 0.5
  #   )
  # ) + 
  xlim(c(NA,400))
p.tree.time</code></pre>
<p><img src="figure/RICR.Rmd/Time%20tree%20-%20all%20mammals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Body Size ###

p.tree.size &lt;- tr.sizedata %&gt;% 
  ggtree(
    aes(color=as.numeric(lnSize.median))
  ) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;gold&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Bats&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), fill=&#39;lightgreen&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), label = &#39;Whales&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;)) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;), label = &#39;Primates&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), fill=&#39;beige&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), label = &#39;Elephants&#39;, align=T) +
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) + 
  scale_color_distiller(
    palette = &quot;Blues&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Body Size (log(g))&quot;,
      title.theme =  element_text(size = 15, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  xlim(c(NA,1300))
p.tree.size</code></pre>
<p><img src="figure/RICR.Rmd/Body%20size%20evolution%20-%20all%20mammals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Lifespan ###

p.tree.life &lt;- tr.lifedata %&gt;% 
  ggtree(
    aes(color=lnLife.median)
  ) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;gold&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Chiroptera&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), fill=&#39;lightgreen&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), label = &#39;Cetacea&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;)) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;), label = &#39;Primata&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), fill=&#39;beige&#39;) +
  ggtree::geom_cladelab(node=MRCA(tr.lifedata, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), label = &#39;Paenungulata&#39;, align=T) + 
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) + 
  scale_color_distiller(
    palette = &quot;Reds&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (log(y))&quot;,
      title.theme =  element_text(size = 15, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  xlim(c(NA,400))
p.tree.life</code></pre>
<p><img src="figure/RICR.Rmd/Lifespan%20evolution%20-%20all%20mammals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Lifespan ###
d.tree.life &lt;- p.tree.life$data

d.tree.life$x &lt;- max(d.tree.life$x) - d.tree.life$x

p.tree.life.flipped &lt;- ggplot(data=d.tree.life) + 
  geom_tree(aes(color=lnLife.median)) + 
  theme_tree() + 
  geom_highlight(node=MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;gold&#39;) +
  geom_highlight(node=MRCA(tr.lifedata, &#39;Balaena_mysticetus&#39;, &#39;Phocoena_phocoena&#39;), fill=&#39;lightgreen&#39;) +
  geom_highlight(node=MRCA(tr.lifedata, &#39;Lepilemur_leucopus&#39;, &#39;Homo_sapiens&#39;)) +
  geom_highlight(node=MRCA(tr.lifedata, &#39;Procavia_capensis&#39;, &#39;Loxodonta_africana&#39;), fill=&#39;beige&#39;) +
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) + 
  scale_color_distiller(
    palette = &quot;Reds&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (log(y))&quot;,
      title.theme =  element_text(size = 15, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  )
p.tree.life.flipped</code></pre>
<p><img src="figure/RICR.Rmd/Lifespan%20evolution%20-%20all%20mammals%20flipped-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tr.sizedata.chiroptera &lt;- tr.sizedata %&gt;% 
  tidytree::tree_subset(MRCA(tr.sizedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>myotis.nodelab &lt;- tr.sizedata.chiroptera@phylo %&gt;% keep.tip(tr.sizedata.chiroptera@phylo$tip.label[str_detect(tr.sizedata.chiroptera@phylo$tip.label, &#39;Myotis&#39;)]) %&gt;% as_tibble() %&gt;% filter(node==parent) %&gt;% pull(label)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>myotis.node &lt;- tr.sizedata.chiroptera %&gt;% as_tibble %&gt;% filter(label==myotis.nodelab) %&gt;% pull(node)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.tree.size.bats &lt;- tr.sizedata.chiroptera %&gt;% 
  ggtree(
    aes(color=as.numeric(lnSize.median),
        text=label)
  ) + 
  geom_highlight(node=myotis.node, fill=&#39;gold&#39;) + 
  geom_cladelab(node=myotis.node, label = &#39;Myotis&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.sizedata.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), fill=&#39;blue&#39;) + 
  geom_cladelab(node=MRCA(tr.sizedata.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), label = &#39;Eptesicus&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.sizedata.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;pink&#39;) + 
  geom_cladelab(node=MRCA(tr.sizedata.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Pteropodidae&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.sizedata.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), fill=&#39;lightblue&#39;) + 
  geom_cladelab(node=MRCA(tr.sizedata.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), label = &#39;Phyllostomidae&#39;, align=T) + 
  # geom_highlight(node=nodeid(tr.sizedata.chiroptera, &#39;Noctilio_leporinus&#39;), fill=&#39;lightblue&#39;) + 
  geom_cladelab(node=nodeid(tr.sizedata.chiroptera, &#39;Noctilio_leporinus&#39;), label = &#39;Noctilionidae&#39;, align=T) + 
  # geom_tiplab(aes(subset=(node==nodeid(tr.sizedata.chiroptera, &#39;Mystacina_tuberculata&#39;)), label = &#39;Mystacinidae&#39;), color=&#39;black&#39;, align=T) + 
  geom_cladelab(node=nodeid(tr.sizedata.chiroptera, &#39;Mystacina_tuberculata&#39;), label = &#39;Mystacinidae&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.sizedata.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), fill=&#39;lightgreen&#39;) + 
  geom_cladelab(node=MRCA(tr.sizedata.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), label = &#39;Rhinolophoidea&#39;, align=T) + 
  geom_highlight(node=MRCA(tr.sizedata.chiroptera, &#39;Molossus_molossus&#39;, &#39;Chaerephon_pumilus&#39;), fill=&#39;coral&#39;) + 
  geom_cladelab(node=MRCA(tr.sizedata.chiroptera, &#39;Molossus_molossus&#39;, &#39;Chaerephon_pumilus&#39;), label = &#39;Molossidae&#39;, align=T) + 
  # scale_color_distiller(
  #     palette = &quot;Blues&quot;, direction = 1,
  #     guide = guide_colorbar(
  #       title=&quot;Body Size (log(g))&quot;,
  #       title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
  #       title.position = &quot;top&quot;,
  #       title.hjust = 0.5
  #     )
  #   ) + 
  guides(color=guide_colorbar(&#39;Body size (log)&#39;)) + 
  theme(
    legend.position = c(0.1,0.8)
  ) +
  xlim(c(NA,70))
p.tree.size.bats</code></pre>
<p><img src="figure/RICR.Rmd/Body%20size%20evolution%20-%20Bats-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Lifespan ###

tr.lifedata.chiroptera &lt;- tr.lifedata %&gt;% 
  tidytree::tree_subset(MRCA(tr.lifedata, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.tree.life.bats &lt;- tr.lifedata.chiroptera %&gt;% 
  ggtree(
    aes(color=as.numeric(lnLife.median))
  ) + 
  geom_highlight(node=myotis.node, fill=&#39;gold&#39;) +
  geom_cladelab(node=myotis.node, label = &#39;Myotis&#39;, align=T) +
  geom_highlight(node=MRCA(tr.lifedata.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), fill=&#39;blue&#39;) +
  geom_cladelab(node=MRCA(tr.lifedata.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), label = &#39;Eptesicus&#39;, align=T) +
  geom_highlight(node=MRCA(tr.lifedata.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;pink&#39;) +
  geom_cladelab(node=MRCA(tr.lifedata.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Pteropodidae&#39;, align=T) +
  geom_highlight(node=MRCA(tr.lifedata.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), fill=&#39;lightblue&#39;) +
  geom_cladelab(node=MRCA(tr.lifedata.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), label = &#39;Phyllostomidae&#39;, align=T) +
  geom_cladelab(node=nodeid(tr.lifedata.chiroptera, &#39;Noctilio_leporinus&#39;), label = &#39;Noctilionidae&#39;, align=T) +
  geom_cladelab(node=nodeid(tr.lifedata.chiroptera, &#39;Mystacina_tuberculata&#39;), label = &#39;Mystacinidae&#39;, align=T) +
  geom_highlight(node=MRCA(tr.lifedata.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), fill=&#39;lightgreen&#39;) +
  geom_cladelab(node=MRCA(tr.lifedata.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), label = &#39;Rhinolophoidea&#39;, align=T) +
  geom_highlight(node=MRCA(tr.lifedata.chiroptera, &#39;Molossus_molossus&#39;, &#39;Chaerephon_pumilus&#39;), fill=&#39;coral&#39;) +
  geom_tiplab(
    aes(
      subset = label %in% c(our_genomes, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;),
      label = label %&gt;% str_replace_all(&quot;_&quot;, &quot; &quot;)
    ),
    fontface=&#39;italic&#39;,
    size=2,
    color=&#39;black&#39;
  ) +
  scale_color_distiller(palette = &quot;Blues&quot;,
                        direction = 1,
                        guide = guide_colorbar(
                          title=&quot;Body size (log)&quot;,
                          title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
                          title.position = &quot;top&quot;,
                          title.hjust = 0.5
                          )
                        ) +
  guides(color=guide_colorbar(&#39;Lifespan (log)&#39;)) +
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) +
  xlim(c(NA,450))
p.tree.life.bats #%&gt;% plotly::ggplotly()</code></pre>
<p><img src="figure/RICR.Rmd/Lifespan%20evolution%20-%20Bats-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Lifespan ###

tr.sizelife.lq.chiroptera &lt;- tr.sizelife.lq %&gt;% 
  tidytree::tree_subset(MRCA(tr.sizelife.lq, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.tree.LQ.bats &lt;- tr.sizelife.lq.chiroptera %&gt;% 
  ggtree(
    aes(
      color=as.numeric(LQ.median),
      # text=label
    ),
    branch.length = &quot;LQ.branch.length&quot;
  ) + 
  geom_highlight(node=myotis.node, fill=&#39;gold&#39;) +
  geom_cladelab(node=myotis.node, label = &#39;Myotis&#39;, align=T) +
  geom_highlight(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), fill=&#39;blue&#39;) +
  geom_cladelab(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Eptesicus_nilssonii&#39;, &#39;Eptesicus_fuscus&#39;), label = &#39;Eptesicus&#39;, align=T) +
  geom_highlight(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), fill=&#39;pink&#39;) +
  geom_cladelab(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Rousettus_aegyptiacus&#39;, &#39;Pteropus_vampyrus&#39;), label = &#39;Pteropodidae&#39;, align=T) +
  geom_highlight(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), fill=&#39;lightblue&#39;) +
  geom_cladelab(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Artibeus_jamaicensis&#39;, &#39;Macrotus_californicus&#39;), label = &#39;Phyllostomidae&#39;, align=T) +
  # geom_highlight(node=nodeid(tr.sizelife.lq.chiroptera, &#39;Noctilio_leporinus&#39;), fill=&#39;lightblue&#39;) +
  geom_cladelab(node=nodeid(tr.sizelife.lq.chiroptera, &#39;Noctilio_leporinus&#39;), label = &#39;Noctilionidae&#39;, align=T) +
  # geom_tiplab(aes(subset=(node==nodeid(tr.sizelife.lq.chiroptera, &#39;Mystacina_tuberculata&#39;)), label = &#39;Mystacinidae&#39;), color=&#39;black&#39;, align=T) +
  geom_cladelab(node=nodeid(tr.sizelife.lq.chiroptera, &#39;Mystacina_tuberculata&#39;), label = &#39;Mystacinidae&#39;, align=T) +
  geom_highlight(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), fill=&#39;lightgreen&#39;) +
  geom_cladelab(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Rhinolophus_euryale&#39;, &#39;Megaderma_lyra&#39;), label = &#39;Rhinolophoidea&#39;, align=T) +
  geom_highlight(node=MRCA(tr.sizelife.lq.chiroptera, &#39;Molossus_molossus&#39;, &#39;Chaerephon_pumilus&#39;), fill=&#39;coral&#39;) +
  geom_tiplab(
    aes(
      subset = label %in% c(our_genomes, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;),
      label = label %&gt;% str_replace_all(&quot;_&quot;, &quot; &quot;)
    ),
    fontface=&#39;italic&#39;,
    size=2,
    color=&#39;black&#39;
  ) +
  scale_color_distiller(palette = &quot;Blues&quot;,
                        direction = 1,
                        guide = guide_colorbar(
                          title=&quot;Body size (log)&quot;,
                          title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
                          title.position = &quot;top&quot;,
                          title.hjust = 0.5
                          )
                        ) +
  guides(color=guide_colorbar(&#39;LQ&#39;)) + 
  theme(
    legend.position = &#39;bottom&#39;,
    legend.direction = &#39;horizontal&#39;
  ) #+ 
  # xlim(c(NA,450))
p.tree.LQ.bats #%&gt;% plotly::ggplotly()</code></pre>
<p><img src="figure/RICR.Rmd/LQ%20evolution%20-%20Bats-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Double Tree ###
p.tree.life.base &lt;- tr.lifedata %&gt;% 
  ggtree() + 
  scale_x_sqrt()
p.tree.size.base &lt;- tr.sizedata %&gt;% 
  ggtree() + 
  scale_x_sqrt()

d.dbltree.mam.size &lt;- p.tree.size.base$data
d.dbltree.mam.life &lt;- p.tree.life.base$data

d.dbltree.mam.size$x &lt;- d.dbltree.mam.size$x %&gt;% divide_by(5)

# d.dbltree.mam.life$x &lt;- d.dbltree.mam.life$x %&gt;% divide_by(5)

d.dbltree.mam.life$x &lt;- max(d.dbltree.mam.life$x) - d.dbltree.mam.life$x + max(d.dbltree.mam.size$x) + 10

p.dbltree.sizelife &lt;-
  ggplot() +
  geom_tree(data=d.dbltree.mam.size, aes(color=lnSize.median), size=1) +
  # tr.sizedata %&gt;% 
  # ggtree(data=d.dbltree.mam.size, aes(color=lnSize.median), size=1) +
  # # scale_color_distiller(&quot;Body size (log)&quot;, palette = &quot;Blues&quot;, direction = 1) +
  scale_color_distiller(
    limits=c(tr.sizedata@extraInfo$lnSize.median %&gt;% min, tr.sizedata@extraInfo$lnSize.median %&gt;% max),
    palette = &quot;Blues&quot;, 
    direction = 1,
    guide = guide_colorbar(
      title=&quot;Body size (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5, 
      title.vjust = 0.75,
      order = 1
    )
  )+ 
  geom_nodepoint(data = d.dbltree.mam.size, aes(color=lnSize.median, x=x, y=y), inherit.aes = F, size=2) + 
  geom_tippoint(data = d.dbltree.mam.size, aes(color=lnSize.median, x=x, y=y), inherit.aes = F, size=2) +
  new_scale_color() + 
  geom_tree(
    data=d.dbltree.mam.life, 
    aes(color=lnLife.median), 
    inherit.aes = F, size=1
  ) + 
  geom_nodepoint(data=d.dbltree.mam.life, aes(color=lnLife.median, x=x, y=y), inherit.aes = F, size=2) + 
  geom_tippoint(data=d.dbltree.mam.life, aes(color=lnLife.median, x=x, y=y), inherit.aes = F, size=2) +
  scale_color_distiller(
    limits=c(tr.lifedata@extraInfo$lnLife.median %&gt;% min, tr.lifedata@extraInfo$lnLife.median %&gt;% max),
    palette = &quot;Reds&quot;,
    direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5,
      title.vjust = 0.75,
      order = 3
    )
  )  + 
  theme_tree() +
  theme(
    plot.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
    panel.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
  legend.text = element_text(
    size = 6, 
    face = &quot;plain&quot;, colour = &quot;black&quot;
  ),
    legend.background = element_blank(),
  legend.position = &quot;bottom&quot;,
  legend.box = &quot;horizontal&quot;,
  legend.direction = &quot;horizontal&quot;,
  legend.box.just = &quot;right&quot;,
  legend.spacing.y = unit(0.5,&quot;mm&quot;)
  # legend.direction = &quot;vertical&quot;
  )

p.dbltree.sizelife</code></pre>
<p><img src="figure/RICR.Rmd/DoubleTree%20Mammals%20Size%20Lifespan-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Double Tree ###
p.tree.life.bats.base &lt;- tr.lifedata.chiroptera %&gt;% 
  ggtree() + 
  scale_x_sqrt()
p.tree.size.bats.base &lt;- tr.sizedata.chiroptera %&gt;% 
  ggtree() + 
  scale_x_sqrt()

d.dbltree.3 &lt;- p.tree.size.bats.base$data
d.dbltree.4 &lt;- p.tree.life.bats.base$data

d.dbltree.4$x &lt;- max(d.dbltree.4$x) - d.dbltree.4$x + max(d.dbltree.3$x) + 10

p.dbltree.bats.sizelife &lt;-
  tr.sizedata.chiroptera %&gt;% 
  ggtree(data=d.dbltree.3, aes(color=lnSize.median), size=1) +
  # scale_color_distiller(&quot;Body size (log)&quot;, palette = &quot;Blues&quot;, direction = 1) +
  scale_color_distiller(
    limits=c(tr.sizedata.chiroptera@extraInfo$lnSize.median %&gt;% min, tr.sizedata.chiroptera@extraInfo$lnSize.median %&gt;% max),
    palette = &quot;Blues&quot;, 
    direction = 1,
    guide = guide_colorbar(
      title=&quot;Body size (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5, 
      title.vjust = 0.75,
      order = 1
    )
    # guide=guide_none()
  ) +
  # scale_fill_distiller(
  #   limits=c(tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% min, tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% max), 
  #   palette = &quot;Reds&quot;, 
  #   direction = 1,
  #   guide = guide_colorbar(
  #     title=&quot;Lifespan&quot;,
  #     title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
  #     title.position = &quot;left&quot;,
  #     title.hjust = 0.5, 
  #     title.vjust = 0.75,
  #     order = 3
  #   )
  #   # guide=guide_none()
  # ) +
  geom_nodepoint(size=2) + 
  geom_tippoint(size=2) +
  # geom_tiplab(aes(label = if_else(is.na(Genome), &quot;&quot;, label)), 
  #             x=(min(d.dbltree.4$x)+max(d.dbltree.3$x))/2, color=&quot;black&quot;, hjust = 0.5) +
  # geom_text_repel(
  #   aes(
  #     label = if_else(
  #       label %in% c(our_genomes, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;), 
  #       label, 
  #       NA
  #     ) %&gt;% str_replace_all(&quot;_&quot;, &quot; &quot;)
  #   ), 
  #   direction = &quot;y&quot;, 
  #   force = 0.5, 
  #   segment.size = NA,
  #   size=2.5,
  #   x=(min(d.dbltree.4$x)+max(d.dbltree.3$x))/2, 
  #   color=&quot;black&quot;, 
  #   hjust = 0.5
  # ) +
  new_scale_color() + 
  geom_tree(
    data=d.dbltree.4, 
    aes(color=lnLife.median), 
    inherit.aes = F, size=1
  ) + 
  geom_nodepoint(data=d.dbltree.4, aes(color=lnLife.median, x=x, y=y), inherit.aes = F, size=2) + 
  geom_tippoint(data=d.dbltree.4, aes(color=lnLife.median, x=x, y=y), inherit.aes = F, size=2) +
  # scale_color_viridis_c(option = &quot;D&quot;) + 
  # scale_color_distiller(
  #   palette = &quot;Reds&quot;, direction = 1,
  #   guide = guide_colorbar(
  #     title=&quot;Cancer Risk (log2)&quot;,
  #     title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
  #     title.position = &quot;top&quot;,
  #     title.hjust = 0.5
  #   )
  # ) + 
  scale_color_distiller(
    limits=c(tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% min, tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% max),
    palette = &quot;Reds&quot;,
    direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5,
      title.vjust = 0.75,
      order = 3
    )
    # guide=guide_none()
  ) +
  theme(
    plot.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
    panel.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
  legend.text = element_text(
    size = 6, 
    face = &quot;plain&quot;, colour = &quot;black&quot;
  ),
    legend.background = element_blank(),
  legend.position = c(0.5, 0.15),#&quot;bottom&quot;,
  legend.box = &quot;vertical&quot;,
  legend.direction = &quot;horizontal&quot;,
  legend.box.just = &quot;right&quot;,
  legend.spacing.y = unit(0.5,&quot;mm&quot;)
  # legend.direction = &quot;vertical&quot;
  )

p.dbltree.bats.sizelife</code></pre>
<p><img src="figure/RICR.Rmd/DoubleTree%20size%20+%20lifespan-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Double Tree ###
p.tree.life.bats.base &lt;- tr.lifedata.chiroptera %&gt;%
  ggtree()
p.tree.LQ.bats.base &lt;- tr.sizelife.lq.chiroptera %&gt;% 
  ggtree(branch.length = &quot;LQ.branch.length&quot;)

d.dbltree.3 &lt;- p.tree.life.bats.base$data
d.dbltree.4 &lt;- p.tree.LQ.bats.base$data

d.dbltree.4$x &lt;- max(d.dbltree.4$x) - d.dbltree.4$x + max(d.dbltree.3$x) + 50

p.dbltree.all &lt;-
  tr.lifedata.chiroptera %&gt;% 
  ggtree(data=d.dbltree.3, aes(color=lnLife.median), size=1) +
  # scale_color_distiller(&quot;Body size (log)&quot;, palette = &quot;Blues&quot;, direction = 1) +
  scale_color_distiller(
    limits=c(tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% min, tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% max),
    palette = &quot;Blues&quot;, 
    direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5, 
      title.vjust = 0.75,
      order = 1
    )
    # guide=guide_none()
  ) +
  # scale_fill_distiller(
  #   limits=c(tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% min, tr.lifedata.chiroptera@extraInfo$lnLife.median %&gt;% max), 
  #   palette = &quot;Reds&quot;, 
  #   direction = 1,
  #   guide = guide_colorbar(
  #     title=&quot;Lifespan&quot;,
  #     title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
  #     title.position = &quot;left&quot;,
  #     title.hjust = 0.5, 
  #     title.vjust = 0.75,
  #     order = 3
  #   )
  #   # guide=guide_none()
  # ) +
  geom_nodepoint(size=2) + 
  geom_tippoint(size=2) +
  # geom_tiplab(aes(subset = label %in% c(our_genomes, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;)),
  #             x=(min(d.dbltree.4$x)+max(d.dbltree.3$x))/2, color=&quot;black&quot;, hjust = -1) +
  geom_segment(
    data = d.dbltree.3 %&gt;% 
      filter(label %in% c(&quot;Myotis_lucifugus&quot;, &quot;Myotis_auriculus&quot;, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;)) %&gt;% 
      left_join(
        d.dbltree.4 %&gt;% select(label, xend=x), 
        by=&#39;label&#39;
      ), 
    aes(
      x=x, 
      xend=xend,
      yend=y
    ),
    color=&#39;black&#39;,
    lty=&#39;dotted&#39;
  ) + 
  geom_label(
    aes(
      label = if_else(
        label %in% c(&quot;Myotis_lucifugus&quot;, &quot;Myotis_auriculus&quot;, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;),
        label,
        NA
      ) %&gt;% str_replace_all(&quot;_&quot;, &quot; &quot;)
    ),
    direction = &quot;y&quot;,
    force = 0.5,
    segment.size = NA,
    size=2.5,
    x=(min(d.dbltree.4$x)+max(d.dbltree.3$x))/2,
    color=&quot;black&quot;,
    fill=&#39;grey&#39;,
    hjust = 0.5
  ) +
  new_scale_color() + 
  geom_tree(
    data=d.dbltree.4, 
    aes(color=LQ.median), 
    inherit.aes = F, size=1
  ) + 
  geom_nodepoint(data=d.dbltree.4, aes(color=LQ.median, x=x, y=y), inherit.aes = F, size=2) + 
  geom_tippoint(data=d.dbltree.4, aes(color=LQ.median, x=x, y=y), inherit.aes = F, size=2) +
  scale_color_distiller(
    limits=c(tr.sizelife.lq.chiroptera@extraInfo$LQ.median %&gt;% min, tr.sizelife.lq.chiroptera@extraInfo$LQ.median %&gt;% max),
    palette = &quot;Reds&quot;,
    direction = 1,
    guide = guide_colorbar(
      title=&quot;LQ&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;left&quot;,
      title.hjust = 0.5,
      title.vjust = 0.75,
      order = 3
    )
    # guide=guide_none()
  ) +
  theme(
    plot.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
    panel.background = element_rect(fill=&quot;grey&quot;, color=&quot;grey&quot;),
  legend.text = element_text(
    size = 6, 
    face = &quot;plain&quot;, colour = &quot;black&quot;
  ),
    legend.background = element_blank(),
  legend.position = &quot;bottom&quot;, #c(0.5, 0.15),
  legend.box = &quot;vertical&quot;,
  legend.direction = &quot;horizontal&quot;,
  legend.box.just = &quot;right&quot;,
  legend.spacing.y = unit(0.5,&quot;mm&quot;)
  # legend.direction = &quot;vertical&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent,node.
ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre><code>Warning in geom_label(aes(label = if_else(label %in% c(&quot;Myotis_lucifugus&quot;, :
Ignoring unknown parameters: `direction`, `force`, and `segment.size`</code></pre>
<pre class="r"><code>p.dbltree.all</code></pre>
<pre><code>Warning: Removed 189 rows containing missing values (`geom_label()`).</code></pre>
<p><img src="figure/RICR.Rmd/DoubleTree%20lifespan%20+%20LQ-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="all-together-figure---myotis" class="section level3">
<h3>All together figure - Myotis</h3>
<pre class="r"><code>p.tr.myotis &lt;- tr.upham %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label) %&gt;% 
  tidytree::tree_subset(MRCA(tr.upham %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label), &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;% 
  ggtree(
    size=1
  ) + 
  geom_tiplab(
    size=3
  ) + 
  geom_nodelab(
    aes(label=label)
  ) + 
  xlim(c(NA, 50))</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>fig.myotis.sizelifelq &lt;- (
  (
    p.tr.myotis
  ) | 
  (
    (p.bodysize.myotis + theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)) / 
    (p.lifespan.myotis + theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)) / 
    (p.LQ.myotis + theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0))
  ) 
) +
    # guide_area()  + 
  plot_layout(width=c(2,1))+
  plot_annotation(tag_levels = &#39;A&#39;) &amp; 
  # theme_pubr() &amp;
  theme(legend.box=&#39;vertical&#39;,
        legend.direction = &#39;vertical&#39;,
        legend.key.height = unit(0.3, &#39;in&#39;),
        legend.key.width = unit(0.25, &#39;in&#39;),
        legend.key.size = unit(0.25, &#39;in&#39;),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        plot.tag = element_text(size = 8, face = &#39;bold&#39;))

fig.myotis.sizelifelq</code></pre>
<p><img src="figure/RICR.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(
  &#39;../output/draft_figs/Myotis_size_life_LQ_upham.pdf&#39;, 
  plot=fig.myotis.sizelifelq,
  width = 5,
  height = 6,
  units = &#39;in&#39;,
  dpi=300
)</code></pre>
</div>
</div>
<div id="cancer-succeptibility-ricr" class="section level2">
<h2>Cancer Succeptibility &amp; RICR</h2>
<p>The consequences of changes in body size and lifespan include changes
in the number of cells in an organism’s body and the time they have for
any given cell to become a tumor. Peto’s Paradox implies that cancer
risk per cell must decrease as the body size and lifespan of a species
increases.</p>
<p>The cancer risk <span class="math inline">\(K\)</span> of an
individual can be summarized as:</p>
<p>$ K $</p>
<p>where <span class="math inline">\(t\)</span> is lifespan and <span
class="math inline">\(D\)</span> is body size. Given that we have
predictions of body size and lifespan at every node in the tree, we can
estimate <span class="math inline">\(K\)</span> for every node. Vazquez
&amp; Lynch (2020) defined the relative cancer risk for a node of
interest as <span
class="math inline">\(\log_2(\frac{K_{anc}}{K_{node}})\)</span>, where
<span class="math inline">\(K_{anc}\)</span> is the cancer risk <span
class="math inline">\(K\)</span> of the ancestral node, and <span
class="math inline">\(K_{node}\)</span> is the cancer risk <span
class="math inline">\(K\)</span> of the node of interest. In this
schema, positive RICR scores means that the cancer risk of the ancestor
is higher than that of the node of interest, and vice versa. If the RICR
score is negative, it indicates that the node of interest has an
increased risk of cancer than its ancestor. <em>These nodes with Reduced
Intrinsic Cancer Risk (RICR) are the nodes where we expect an
enhancement of tumor suppression.</em></p>
<!-- We saw in Vazquez & Lynch (2020) and Vazquez _et al._ (2022) that there is evidence for conservation of genes associated with increase tumor suppression at nodes with positive RICR scores -->
<pre class="r"><code># Recalculate log2CancerSucceptabilityChange with real lifespans
cancerSucceptibility.table &lt;- tr.sizelife.lq %&gt;%
  as_tibble %&gt;% 
  ### Cancer Susceptibility
  mutate(
    K1 = exp(lnLife.median)^6 * exp(lnSize.median),
    lnK1 = 6*lnLife.median + lnSize.median
  )

cancerSucceptibility.table &lt;- cancerSucceptibility.table %&gt;% 
  mutate(
    Ancestor = label %&gt;%
      sapply(
        .,
        function(n, d=cancerSucceptibility.table){
          if(is.na(n)){return(NA)}
          #print(n)
          #print(length(n))
          p.node &lt;- d %&gt;% filter(label==n) %&gt;% pull(parent) %&gt;% unique
          d %&gt;% filter(node == p.node) %&gt;% pull(label) %&gt;% unique %&gt;% return
        }
      ) %&gt;% str_replace_all(&quot;\\.&quot;, &quot; &quot;) %&gt;%
      str_replace(&quot;Root&quot;, &quot;chiroptera&quot;),
    K2 = parent %&gt;%
      sapply(
        .,
        function(n, d=cancerSucceptibility.table){
          d %&gt;% filter(node==n) %&gt;% pull(K1) %&gt;% unique
        }
      ),
    lnK2 = parent %&gt;%
      sapply(
        .,
        function(n, d=cancerSucceptibility.table){
          d %&gt;% filter(node==n) %&gt;% pull(lnK1) %&gt;% unique
        }
      ),
    CancerSucceptabilityChange = K2/K1,
    log2CancerSucceptabilityChange = log2(K2)-log2(K1),
    label = label %&gt;% str_replace_all(&quot;\\.&quot;, &quot; &quot;)
  )

tr.cancerSucceptibility &lt;- cancerSucceptibility.table %&gt;% 
  as.treedata()

cancerSucceptibility.table %&gt;% 
  filter(str_detect(label, &quot;Myotis&quot;)) %&gt;% 
  select(parent,node,label, lnSize.median, lnLife.median, K1, K2, log2CancerSucceptabilityChange) %&gt;% 
  arrange(log2CancerSucceptabilityChange)

cancerSucceptibility.table %&gt;% 
  filter(label %in% tr.sizelife.lq.chiroptera@phylo$tip.label) %&gt;% 
  select(parent,node,label, lnSize.median, lnLife.median, K1, K2, log2CancerSucceptabilityChange) %&gt;% 
  arrange(log2CancerSucceptabilityChange)</code></pre>
<pre class="r"><code>tr.cancerSucceptibility &lt;- treeio::read.beast(&#39;../output/Upham2019-RICR-Data.nexus&#39;)
tr.cancerSucceptibility@data &lt;- tr.cancerSucceptibility@data %&gt;% 
  mutate_at(vars(-label.y, -node,-Ancestor),as.numeric)</code></pre>
<pre class="r"><code>p.cancerSucceptibility &lt;- tr.cancerSucceptibility %&gt;% 
  ggtree(
    aes(color=log2CancerSucceptabilityChange),
    layout = &#39;fan&#39;,
    open.angle = 180
  ) + 
  geom_tippoint(
    aes(
      fill=lnK1
    ),
    shape=21,
    color=&#39;black&#39;
  ) + 
  geom_nodepoint(
    aes(
      fill=lnK1
    ),
    shape=21,
    color=&#39;black&#39;
  ) + 
  scale_color_viridis_c(
    &quot;RICR (log2)&quot;,
    option = &#39;C&#39;
  ) + 
  scale_fill_viridis_c(
    &quot;Cancer Risk (log)&quot;,
    option = &#39;D&#39;
  ) + 
  theme(
    legend.position = c(0.5, 0.4),
    legend.direction = &#39;horizontal&#39;,
    legend.box = &#39;horizontal&#39;
  )</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.</code></pre>
<pre class="r"><code>p.cancerSucceptibility</code></pre>
<p><img src="figure/RICR.Rmd/Plot%20RICR%20(time-scaled)%20-%20Mammals-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tr.RICR.chiroptera &lt;- tr.cancerSucceptibility %&gt;% 
  tidytree::tree_subset(MRCA(tr.cancerSucceptibility, &#39;Myotis_lucifugus&#39;, &#39;Pteropus_vampyrus&#39;), levels_back = 0)</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.RICR.bats &lt;- tr.RICR.chiroptera %&gt;% 
  ggtree(
    aes(color=log2CancerSucceptabilityChange),
    layout = &#39;fan&#39;,
    open.angle = 180,
    size=1.5
  ) + 
  geom_tippoint(
    aes(
      fill=lnK1
    ),
    shape=21,
    color=&#39;black&#39;
  ) + 
  geom_nodepoint(
    aes(
      fill=lnK1
    ),
    shape=22,
    color=&#39;black&#39;
  ) + 
  scale_color_viridis_c(
    &quot;RICR (log2)&quot;,
    option = &#39;C&#39;
  ) + 
  scale_fill_viridis_c(
    &quot;Cancer Risk (log)&quot;,
    option = &#39;D&#39;
  ) + 
  geom_tiplab2(
    aes(
      subset = label %in% c(our_genomes, &quot;Myotis_grisescens&quot;, &quot;Rhinolophus_rouxii&quot;, &quot;Myotis_brandtii&quot;, &quot;Nycticeius_humeralis&quot;, &quot;Natalus_stramineus&quot;, &quot;Desmodus_rotundus&quot;, &quot;Pteropus_giganteus&quot;, &quot;Eonycteris_spelaea&quot;),
      label = label %&gt;% str_replace_all(&quot;_&quot;, &quot; &quot;)
    ),
    fontface=&#39;italic&#39;, offset=1,
    size=2,
    color=&#39;black&#39;
  ) +
  theme(
    legend.position = c(0.5, 0.4),
    legend.direction = &#39;horizontal&#39;,
    legend.box = &#39;horizontal&#39;
  )</code></pre>
<pre><code>Scale for y is already present.
Adding another scale for y, which will replace the existing scale.</code></pre>
<pre class="r"><code>p.RICR.bats</code></pre>
<p><img src="figure/RICR.Rmd/Plot%20RICR%20(time-scaled)%20-%20Bats-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It seems like overall, there’s not much change in cancer risk at
nodes within the tree - all the changes happen at the tips. Within the
root of <em>Myotis</em>, the RICR changes are small (&lt;1). In both
megabats and microbats, changes in RICR within the tree are relatively
small but dynamic - this is probably due to the density of species in
such a short period of time compared to what I’m used to in elephants
and other large-bodied mammals.</p>
<p>At the common ancestor of Grey myotis and the other bats, there’s a
series of small increases in RICR (3.8, 0.99, 4.24, 0.65) from the
common ancestor until you get to <em>grisescens</em> - then there’s a
big drop (-12.7) in RICR. Brandt’s bat is the longest-lived bat and the
daughter of a short-lived node, so it makes sense that there’s a big
drop there (-0.22 to -6.8). All of our <em>Myotis</em> show drops in
RICR, especially for <em>lucifugus</em> (-4.1). Note that for <em>Myotis
myotis</em>, while it has a low drop in RICR (-1.0), its most recent
ancestor had a larger drop (-3.2). The second greatest drop in RICR in
in Megabats, in <em>Pteropus gigantus</em> (-7.3).</p>
<div id="ricr-based-version-of-bat-fig" class="section level3">
<h3>RICR-based version of bat fig</h3>
<pre class="r"><code>p.RICR.myotis &lt;- tr.cancerSucceptibility %&gt;%
  tidytree::tree_subset(MRCA(tr.cancerSucceptibility, &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(log2CancerSucceptabilityChange) %&gt;% 
  mutate(label = factor(label, levels=label)) %&gt;% 
  ggplot(
    aes(
      y=label,
      x=log2CancerSucceptabilityChange
      )
  ) +
  geom_segment(
    aes(y=label, yend=label,
        x=0, xend=log2CancerSucceptabilityChange,
        color=log2CancerSucceptabilityChange)
  ) + 
  geom_point(aes(color=log2CancerSucceptabilityChange), size=1) +
  # geom_point(aes(x=LQ.low, color=LQ.low), size=1.5, stroke=2, pch=&quot;]&quot;) +
  # geom_point(aes(x=LQ.high, color=LQ.high), size=1.5, stroke=2, pch=&quot;[&quot;) +
  scale_color_viridis_c(
    option = &#39;D&#39;,
    guide = guide_colorbar(
      title=&quot;RICR&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) +
  labs(
    x=&quot;RICR&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.RICR.myotis</code></pre>
<p><img src="figure/RICR.Rmd/RangePlot%20RICR%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>tbl.RICR.myotis2 &lt;- tr.cancerSucceptibility %&gt;%
  tidytree::tree_subset(MRCA(tr.cancerSucceptibility, &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  arrange(lnK1) %&gt;% 
  mutate(label = str_replace_all(label, &quot;_&quot;, &quot; &quot;)) %&gt;% 
  mutate(label = factor(label, levels=label))</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>high.K &lt;- tbl.RICR.myotis2$label[str_detect(tbl.RICR.myotis2$label, &quot;Myotis myotis&quot;)]
mid.K &lt;- tbl.RICR.myotis2$label[str_detect(
  tbl.RICR.myotis2$label, 
  # &quot;Myotis_septentrionalis&quot;
  &quot;1668&quot;
)]
low.K = tbl.RICR.myotis2$label[str_detect(tbl.RICR.myotis2$label, &quot;Myotis auriculus&quot;)]

p.RICR.myotis2 &lt;- tbl.RICR.myotis2 %&gt;% 
  ggplot(
    aes(
      y=label,
      x=lnK1
      )
  ) +
  geom_segment(
    aes(y=label, yend=label,
        x=lnK2, xend=lnK1,
        color=log2CancerSucceptabilityChange),
    arrow = arrow(length = unit(0.1, &#39;cm&#39;),type=&#39;closed&#39;)
  ) + 
  geom_point(aes(x=lnK2), size=2, shape=&quot;|&quot;) +
  geom_segment(
    aes(
      y=high.K,
      yend = mid.K,
      x=-Inf,
      xend=-Inf
    ),
    color=&#39;blue&#39;,
    size=2
  )+
  geom_segment(
    aes(
      y=mid.K,
      yend = low.K,
      x=-Inf,
      xend=-Inf
    ),
    color=&#39;red&#39;,
    size=2
  )+
  # geom_point(aes(x=LQ.low, color=LQ.low), size=1.5, stroke=2, pch=&quot;]&quot;) +
  # geom_point(aes(x=LQ.high, color=LQ.high), size=1.5, stroke=2, pch=&quot;[&quot;) +
  scale_color_viridis_c(
    option = &#39;D&#39;,
    guide = guide_colorbar(
      title=&quot;RICR&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) +
  labs(
    x=&quot;Cancer Risk&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  guides(
    shape = guide_legend()
  ) +
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>p.RICR.myotis2</code></pre>
<p><img src="figure/RICR.Rmd/RangePlot%20RICR%20v2%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p.RICR.myotis3.plot &lt;- tbl.RICR.myotis2 %&gt;% 
  ggplot(
    aes(
      y=label,
      x=lnK1
      )
  ) +
  geom_segment(
    aes(y=label, yend=label,
        x=lnK2, xend=lnK1,
        color=log2CancerSucceptabilityChange),
    arrow = arrow(length = unit(0.2, &#39;cm&#39;),type=&#39;closed&#39;)
  ) + 
  geom_point(aes(x=lnK2), size=2, shape=&quot;|&quot;) +
  scale_color_viridis_c(
    option = &#39;D&#39;,
    guide = guide_colorbar(
      title=&quot;RICR&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) +
  scale_x_continuous(limits=c(9.5,30)) +
  labs(
    x=&quot;Cancer Risk&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() +
  theme(
    text = element_text(face = &quot;plain&quot;,
                        colour = &quot;black&quot;, lineheight = 0.9,
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(),
                        debug = FALSE),
    axis.text = element_text(size = 6,
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8,
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8,
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;),
    axis.text.y = element_blank(), 
    axis.ticks.y = element_blank(),
    axis.title.y = element_blank(),
    legend.text = element_text(size = 6,
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )

p.RICR.myotis.tiles &lt;- tbl.RICR.myotis2 %&gt;% 
  ggplot(
    aes(
      x=0,
      y=label,
      fill=lnK1
    )
  ) + 
    geom_tile(
    width=0.25,
    height=0.5,
    color=&#39;black&#39;
  ) +
    scale_fill_viridis_c(
    option = &#39;B&#39;,
    guide = guide_colorbar(
      title=&quot;Cancer Risk (log)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  theme_void() +
  theme(
    legend.position = &#39;bottom&#39;
  )

p.RICR.myotis.text &lt;- tbl.RICR.myotis2 %&gt;% 
  ggplot(
    aes(
      x=0,
      y=label,
      label=label
    )
  ) + 
    geom_text(
      hjust=1, 
      size=2.5,
      fontface=&#39;bold.italic&#39;
    ) +  
  xlim(c(-0.025,0)) + 
  theme_void() +
  theme(
    legend.position = &#39;bottom&#39;
  )

p.RICR.myotis3 &lt;- (p.RICR.myotis.text + 
  p.RICR.myotis.tiles + 
  p.RICR.myotis3.plot) + 
  # guide_area() + 
  plot_layout(guides=&#39;collect&#39;,widths = c(0.3,0.01,1)) &amp; 
  theme(
    legend.position = &#39;bottom&#39;,
    legend.text = element_text(size = 6,
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
  )
   
p.RICR.myotis3</code></pre>
<p><img src="figure/RICR.Rmd/RangePlot%20RICR%20v3%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(
  &#39;../output/subfigs/myotis_RICR_v3.pdf&#39;,
  p.RICR.myotis3,
  height = 5,
  width = 8,
  units = &#39;in&#39;,
  dpi=300
)
ggsave(
  &#39;../output/subfigs/myotis_RICR_v3.png&#39;,
  p.RICR.myotis3,
  height = 5,
  width = 8,
  units = &#39;in&#39;,
  dpi=300
)</code></pre>
<pre class="r"><code>p.tr.myotis &lt;- tr.cancerSucceptibility %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label) %&gt;% 
  tidytree::tree_subset(MRCA(tr.upham %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label), &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;% 
  ggtree(
    size=1
  ) + 
  geom_tiplab(
    size=3
  ) + 
  geom_nodelab(
    aes(label=label),
    geom = &#39;shadowtext&#39;,
    color=&#39;white&#39;,
    fill=&#39;black&#39;,
    hjust=1.5,
    vjust=-0.2,
    size=2.5,
  ) + 
  xlim(c(NA, 50))</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>p.tr.myotis2 &lt;- (tr.cancerSucceptibility %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label) %&gt;%
  tidytree::tree_subset(MRCA(tr.upham %&gt;%
  keep.tip(tr.lifedata.chiroptera@phylo$tip.label), &#39;Myotis_lucifugus&#39;, &#39;Nycticeius_humeralis&#39;), levels_back = 0) %&gt;%
  ggtree(
    size=1
  ) + 
  geom_tiplab(
    size=3
  ) + 
  geom_nodelab(
    aes(label=label),
    geom = &#39;shadowtext&#39;,
    color=&#39;white&#39;,
    fill=&#39;black&#39;,
    hjust=1.5,
    vjust=-0.2,
    size=2.5,
  )) #%&gt;% </code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>  # collapse(., node, &#39;max&#39;)




fig.myotis.sizelifericr &lt;- (
  (
    p.tr.myotis
  ) | 
  (
    (p.bodysize.myotis + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     ) / 
    (p.lifespan.myotis + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     ) / 
    (p.RICR.myotis2 + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     )
  ) 
) +
    # guide_area()  + 
  plot_layout(width=c(1,2))+
  plot_annotation(tag_levels = &#39;A&#39;) &amp; 
  theme(legend.box=&#39;vertical&#39;,
        legend.direction = &#39;vertical&#39;,
        legend.key.height = unit(0.3, &#39;in&#39;),
        legend.key.width = unit(0.25, &#39;in&#39;),
        legend.key.size = unit(0.25, &#39;in&#39;),
        # axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        plot.tag = element_text(size = 8, face = &#39;bold&#39;))

fig.myotis.sizelifericr</code></pre>
<p><img src="figure/RICR.Rmd/myotis-only%20size-life-RICR%20fig-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(
  &#39;../output/draft_figs/Myotis_size_life_RICR_upham.pdf&#39;, 
  plot=fig.myotis.sizelifericr,
  width = 8,
  height = 6,
  units = &#39;in&#39;,
  dpi=300
)</code></pre>
<pre class="r"><code># tr.RICR.chiroptera %&gt;% ggtree(layout=&#39;circular&#39;) + geom_tiplab2()

node.myotis = MRCA(tr.RICR.chiroptera, &#39;Myotis_lucifugus&#39;, &#39;Myotis_emarginatus&#39;)
node.vespertilionidae = MRCA(tr.RICR.chiroptera, &#39;Myotis_lucifugus&#39;, &#39;Lasiurus_cinereus&#39;)
node.molossidae = MRCA(tr.RICR.chiroptera, &#39;Molossus_molossus&#39;, &#39;Chaerephon_pumilus&#39;)
node.phyllostomidae = MRCA(tr.RICR.chiroptera, &#39;Macrotus_waterhousii&#39;, &#39;Artibeus_jamaicensis&#39;)
node.yinpterochiroptera = MRCA(tr.RICR.chiroptera, &#39;Macroderma_gigas&#39;, &#39;Pteropus_giganteus&#39;)
node.yangochiroptera = MRCA(tr.RICR.chiroptera, &#39;Myotis_lucifugus&#39;, &#39;Mystacina_tuberculata&#39;) 

tr.RICR.myotis &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.myotis, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Myotis&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.vespertilionidae &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.vespertilionidae, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Vespertilionidae&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.molossidae &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.molossidae, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Molossidae&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.phyllostomidae &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.phyllostomidae, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Phyllostomidae&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.yinpterochiroptera &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.yinpterochiroptera, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Yinpterochiroptera&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.yangochiroptera &lt;- tr.RICR.chiroptera %&gt;%
  tidytree::tree_subset(node.yangochiroptera, levels_back = 0) %&gt;%
  as_tibble %&gt;% 
  summarize(
    label=&#39;Yangochiroptera&#39;,
    lnSize.median = mean(lnSize.median),
    lnLife.median = mean(lnLife.median),
    lnSize.low = mean(lnSize.low),
    lnLife.low = mean(lnLife.low),
    lnSize.high = mean(lnSize.high),
    lnLife.high = mean(lnLife.high),
    lnK1 = mean(lnK1),
    lnK2 = mean(lnK2),
    log2CancerSucceptabilityChange = mean(log2CancerSucceptabilityChange)
  )</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.
ℹ invalid tbl_tree object. Missing column: parent.</code></pre>
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
<pre class="r"><code>tr.RICR.summary &lt;- 
  bind_rows(
    tr.RICR.myotis,
    tr.RICR.vespertilionidae,
    tr.RICR.molossidae,
    tr.RICR.phyllostomidae,
    tr.RICR.yinpterochiroptera,
    tr.RICR.yangochiroptera,
  )


phyloOrder &lt;- p.tr.myotis$data %&gt;% 
  select(label, y) %&gt;% 
  arrange(desc(y)) %&gt;% 
  pull(label) %&gt;% 
  c(tr.RICR.summary$label) %&gt;% 
  str_replace_all(&quot;_&quot;, &quot; &quot;) %&gt;% 
  rev</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre><code>ℹ invalid tbl_tree object. Missing column: parent,node.</code></pre>
<pre class="r"><code>tbl.RICR.myotis.withSummary &lt;- 
  tbl.RICR.myotis2 %&gt;% 
  select(label, lnSize.median, lnLife.median, lnSize.low, lnLife.low, lnSize.high, lnLife.high, lnK1, lnK2, log2CancerSucceptabilityChange) %&gt;% 
  bind_rows(tr.RICR.summary) %&gt;% 
  mutate(label=factor(label, levels=phyloOrder)) %&gt;% 
  arrange(label)</code></pre>
<pre class="r"><code>p.RICR.myotis.sum &lt;- tbl.RICR.myotis.withSummary %&gt;% 
  ggplot(
    aes(
      y=label,
      x=lnK1
      )
  ) +
  geom_rect(
    aes(
      ymin=factor(&quot;Myotis&quot;, levels=phyloOrder),
      ymax=factor(&quot;Yangochiroptera&quot;, levels=phyloOrder),
      xmin=-Inf, 
      xmax=Inf
    ), 
    color=&#39;beige&#39;,
    fill=&#39;beige&#39;
  ) + 
  geom_segment(
    aes(y=label, yend=label,
        x=lnK2, xend=lnK1,
        color=log2CancerSucceptabilityChange),
    arrow = arrow(length = unit(0.1, &#39;cm&#39;),type=&#39;closed&#39;)
  ) + 
  geom_point(aes(x=lnK2), size=2, shape=&quot;|&quot;) +
  scale_color_viridis_c(
    option = &#39;D&#39;,
    guide = guide_colorbar(
      title=&quot;RICR&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) +
  labs(
    x=&quot;Cancer Risk&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubclean()+
  labs_pubr() + 
  guides(
    shape = guide_legend()
  ) +
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )

p.RICR.myotis.sum</code></pre>
<p><img src="figure/RICR.Rmd/RICR%20with%20Summarized%20Data%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p.bodysize.myotis.sum &lt;- tbl.RICR.myotis.withSummary %&gt;% 
  ggplot(
    aes(
      y=label,
      x=lnSize.median, 
      xmin=lnSize.low,
      xmax=lnSize.high
      )
  ) +
  geom_rect(
    aes(
      ymin=factor(&quot;Myotis&quot;, levels=phyloOrder),
      ymax=factor(&quot;Yangochiroptera&quot;, levels=phyloOrder),
      xmin=-Inf, 
      xmax=Inf
    ), 
    color=&#39;beige&#39;,
    fill=&#39;beige&#39;
  ) + 
  geom_pointrange(size=0.5, stroke=0.1) +
  geom_point(aes(color=lnSize.median), size=0.5) +
  scale_color_distiller(limits=c(size.scale.expanded.min, size.scale.expanded.max), 
                        palette = &quot;Blues&quot;, 
                        direction = 1,
                        guide = guide_colorbar(
                          title=&quot;Body size (log)&quot;,
                          title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
                          title.position = &quot;top&quot;,
                          title.hjust = 0.5
                          )
                        ) +
  labs(
    x=&quot;Body size (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )
p.bodysize.myotis.sum</code></pre>
<p><img src="figure/RICR.Rmd/Body%20Size%20with%20Summarized%20Data%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p.lifespan.myotis.sum &lt;- tbl.RICR.myotis.withSummary %&gt;% 
  mutate(
    life.median = exp(lnLife.median),
    life.low = exp(lnLife.low),
    life.high = exp(lnLife.high)
  ) %&gt;% 
  ggplot(
    aes(
      y=label,
      x=life.median, 
      xmin=life.low,
      xmax=life.high
      )
  ) +
  geom_rect(
    aes(
      ymin=factor(&quot;Myotis&quot;, levels=phyloOrder),
      ymax=factor(&quot;Yangochiroptera&quot;, levels=phyloOrder),
      xmin=-Inf, 
      xmax=Inf
    ), 
    color=&#39;beige&#39;,
    fill=&#39;beige&#39;
  ) + 
  geom_pointrange(size=0.5, stroke=0.5) +
  geom_point(aes(color=life.median), size=0.5) +
  scale_color_distiller(
    palette = &quot;Reds&quot;, direction = 1,
    guide = guide_colorbar(
      title=&quot;Lifespan (yrs)&quot;,
      title.theme =  element_text(size = 6, face = &quot;bold&quot;, colour = &quot;black&quot;),
      title.position = &quot;top&quot;,
      title.hjust = 0.5
    )
  ) + 
  labs(
    x=&quot;Lifespan (log)&quot;,
    y=&quot;Node&quot;
  ) + 
  theme_pubr()+
  labs_pubr() + 
  theme(
    text = element_text(face = &quot;plain&quot;, 
                        colour = &quot;black&quot;, lineheight = 0.9, 
                        hjust = 0.5, vjust = 0.5, angle = 0, margin = margin(), 
                        debug = FALSE), 
    axis.text = element_text(size = 6, 
                               colour = &quot;black&quot;, face = &quot;bold&quot;),
    axis.title = element_text(size = 8, 
                              colour = &quot;black&quot;, face = &quot;bold&quot;),
    plot.title = element_text(size = 8, 
                              colour = &quot;black&quot;, lineheight = 1, face = &quot;bold&quot;), 
    legend.text = element_text(size = 6, 
                               face = &quot;plain&quot;, colour = &quot;black&quot;),
    legend.position = &quot;bottom&quot;
  )
p.lifespan.myotis.sum</code></pre>
<p><img src="figure/RICR.Rmd/lifespan%20with%20summarized%20data%20-%20Myotis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fig.myotis.sizelifericr.sum &lt;- (
  (
    p.tr.myotis
  ) | 
  (
    (p.bodysize.myotis.sum + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     ) / 
    (p.lifespan.myotis.sum + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     ) / 
    (p.RICR.myotis.sum + 
       # theme(legend.position = &#39;left&#39;, legend.direction = &#39;vertical&#39;, legend.title.align = 0)
       theme(legend.position = &#39;none&#39;)
     )
  ) 
) +
    # guide_area()  + 
  plot_layout(width=c(1,2))+
  plot_annotation(tag_levels = &#39;A&#39;) &amp; 
  theme(legend.box=&#39;vertical&#39;,
        legend.direction = &#39;vertical&#39;,
        legend.key.height = unit(0.3, &#39;in&#39;),
        legend.key.width = unit(0.25, &#39;in&#39;),
        legend.key.size = unit(0.25, &#39;in&#39;),
        # axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        plot.tag = element_text(size = 8, face = &#39;bold&#39;))

fig.myotis.sizelifericr.sum</code></pre>
<p><img src="figure/RICR.Rmd/Myotis+Summary%20Size-Life-RICR%20fig-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(
  &#39;../output/draft_figs/Myotis_size_life_RICR_upham_withSummary.pdf&#39;, 
  plot=fig.myotis.sizelifericr.sum,
  width = 7,
  height = 8.5,
  units = &#39;in&#39;,
  dpi=300
)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: America/La_Paz
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] aplot_0.2.1      patchwork_1.1.3  ggnewscale_0.4.9 rphylopic_1.2.1 
 [5] kableExtra_1.3.4 ggrepel_0.9.4    ggsci_3.0.0      broom_1.0.5     
 [9] ggtree_3.6.2     ggdist_3.3.0     ggpubr_0.6.0     tidytree_0.4.5  
[13] ape_5.7-1        magrittr_2.0.3   lubridate_1.9.2  forcats_1.0.0   
[17] stringr_1.5.0    dplyr_1.1.3      purrr_1.0.2      readr_2.1.4     
[21] tidyr_1.3.0      tibble_3.2.1     ggplot2_3.4.3    tidyverse_2.0.0 

loaded via a namespace (and not attached):
 [1] shadowtext_0.1.2     pbapply_1.7-2        rlang_1.1.1         
 [4] git2r_0.32.0         compiler_4.3.1       mgcv_1.9-0          
 [7] png_0.1-8            systemfonts_1.0.4    vctrs_0.6.3         
[10] rvest_1.0.3          pkgconfig_2.0.3      crayon_1.5.2        
[13] fastmap_1.1.1        backports_1.4.1      labeling_0.4.3      
[16] utf8_1.2.3           promises_1.2.1       rmarkdown_2.25      
[19] tzdb_0.4.0           ragg_1.2.5           bit_4.0.5           
[22] xfun_0.40            cachem_1.0.8         jsonlite_1.8.7      
[25] later_1.3.1          jpeg_0.1-10          parallel_4.3.1      
[28] R6_2.5.1             RColorBrewer_1.1-3   bslib_0.5.1         
[31] stringi_1.7.12       car_3.1-2            jquerylib_0.1.4     
[34] Rcpp_1.0.11          knitr_1.44           base64enc_0.1-3     
[37] Matrix_1.6-1.1       httpuv_1.6.11        splines_4.3.1       
[40] timechange_0.2.0     tidyselect_1.2.0     rstudioapi_0.15.0   
[43] abind_1.4-5          yaml_2.3.7           curl_5.0.2          
[46] lattice_0.21-8       treeio_1.25.4        withr_2.5.0         
[49] evaluate_0.21        gridGraphics_0.5-1   xml2_1.3.5          
[52] pillar_1.9.0         carData_3.0-5        whisker_0.4.1       
[55] ggfun_0.1.3          distributional_0.3.2 generics_0.1.3      
[58] vroom_1.6.3          rprojroot_2.0.3      grImport2_0.3-0     
[61] hms_1.1.3            munsell_0.5.0        scales_1.2.1        
[64] glue_1.6.2           lazyeval_0.2.2       tools_4.3.1         
[67] webshot_0.5.5        ggsignif_0.6.4       fs_1.6.3            
[70] XML_3.99-0.14        grid_4.3.1           colorspace_2.1-0    
[73] nlme_3.1-163         cli_3.6.1            textshaping_0.3.6   
[76] workflowr_1.7.1      rsvg_2.6.0           fansi_1.0.5         
[79] viridisLite_0.4.2    svglite_2.1.1        gtable_0.3.4        
[82] rstatix_0.7.2        yulab.utils_0.1.0    sass_0.4.7          
[85] digest_0.6.33        ggplotify_0.1.2      farver_2.1.1        
[88] memoise_2.0.1        htmltools_0.5.6      lifecycle_1.0.3     
[91] httr_1.4.7           bit64_4.0.5         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
