<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Juan Manuel Vazquez" />

<meta name="date" content="2023-05-02" />

<title>Orthogroup Duplications</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Myotis Genome Analyses</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Orthogroup Duplications</h1>
<h4 class="author">Juan Manuel Vazquez</h4>
<h4 class="date">2023-05-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>R_workflowr/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230501code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230501)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230501code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230501)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabmyotisGenomeAssemblytree04466091830fe3dd3b93c5fb2720151a739edaf6targetblank0446609a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sudmantlab/myotisGenomeAssembly/tree/04466091830fe3dd3b93c5fb2720151a739edaf6" target="_blank">0446609</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsudmantlabmyotisGenomeAssemblytree04466091830fe3dd3b93c5fb2720151a739edaf6targetblank0446609a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sudmantlab/myotisGenomeAssembly/tree/04466091830fe3dd3b93c5fb2720151a739edaf6" target="_blank">0446609</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    R_workflowr/.RData
    Ignored:    R_workflowr/.Rhistory
    Ignored:    R_workflowr/.Rproj.user/
    Ignored:    R_workflowr/data/
    Ignored:    R_workflowr/output/
    Ignored:    R_workflowr/renv/library/
    Ignored:    R_workflowr/renv/staging/
    Ignored:    analyses/phylo/output/
    Ignored:    analyses/thex/.RData
    Ignored:    analyses/thex/.Rhistory
    Ignored:    analyses/thex/data/
    Ignored:    analyses/thex/logs/
    Ignored:    analyses/thex/output/
    Ignored:    data/assembly_stats/

Untracked files:
    Untracked:  R_workflowr/analysis/OrthoFinder_dup.Rmd
    Untracked:  R_workflowr/analysis/phylogeny.Rmd

Unstaged changes:
    Modified:   R_workflowr/.Rprofile
    Modified:   R_workflowr/_workflowr.yml
    Modified:   R_workflowr/analysis/_site.yml
    Modified:   R_workflowr/analysis/ng_curves.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="a-very-simple-attempt-at-looking-at-gene-duplications"
class="section level2">
<h2>A very simple attempt at looking at gene duplications</h2>
<div id="method" class="section level3">
<h3>Method</h3>
<p>All of this is based on
<code>OrthofinderResults/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv</code>.</p>
<p>First, a list of all Orthogroups associated with a COSMIC Tier 1 gene
was generated by subsetting one of the Ortholog files by COSMIC ENSGs
with the command:</p>
<pre><code>grep -f COSMIC_ENSG.txt ../../Results_Mar11/Orthologues/Orthologues_human/human__v__auriculus.tsv | cut -f1 | sort -Vu &gt; COSMIC_Myotis_orthogroups.txt</code></pre>
<p>Then, I subsetted the <code>Duplications_per_Orthogroup</code> file
to only include the COSMIC orthogroups:</p>
<pre><code>grep -f COSMIC_Myotis_orthogroups.txt ../../Results_Mar11/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv &gt; Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv</code></pre>
<p>For all counts related to COSMIC, I used <code>cut</code> and
<code>grep</code> to look for duplication event counts in the 3rd
column, “Duplications (50% support)”:</p>
<pre><code>### No dups
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -w &quot;0&quot; | wc -l

### 1 dup
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -w &quot;1&quot; | wc -l

### 2+ dup
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | grep -vw &quot;0&quot; | grep -vw &quot;1&quot; | wc -l

### Total
cut -f 3 Orthofinder_DuplicationsPerOrthogroup_COSMIC.tsv | wc -l</code></pre>
<p>I repeated this exactly using the original
<code>Duplications_per_Orthogroup.tsv</code> file as a baseline.</p>
<p>To make it a more fair fight, I subsetted the original
<code>Duplications_per_Orthogroup.tsv</code> file to match the total
group count of the COSMIC file and repeated the counts:</p>
<pre><code>tail -n +2 ../../Results_Mar11/Comparative_Genomics_Statistics/Duplications_per_Orthogroup.tsv | shuf -n 537 &gt; Orthofinder_DuplicationsPerOrthogroup_All.tsv</code></pre>
</div>
<div id="table-of-counts" class="section level3">
<h3>Table of counts</h3>
<pre class="r"><code>dup &lt;- read_table(&#39;../../data/tmp_Orthogroup_duplication.tsv&#39;)</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  Set = col_character(),
  Dup = col_character(),
  Count = col_double()
)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Set</th>
<th align="left">Dup</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="left">noDup</td>
<td align="right">445</td>
</tr>
<tr class="even">
<td align="left">COSMIC</td>
<td align="left">1Dup</td>
<td align="right">59</td>
</tr>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="left">2+Dup</td>
<td align="right">33</td>
</tr>
<tr class="even">
<td align="left">COSMIC</td>
<td align="left">Total</td>
<td align="right">537</td>
</tr>
<tr class="odd">
<td align="left">OrthoGroup</td>
<td align="left">noDup</td>
<td align="right">18106</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup</td>
<td align="left">1Dup</td>
<td align="right">2527</td>
</tr>
<tr class="odd">
<td align="left">OrthoGroup</td>
<td align="left">2+Dup</td>
<td align="right">1812</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup</td>
<td align="left">Total</td>
<td align="right">22445</td>
</tr>
<tr class="odd">
<td align="left">OrthoGroup537rando</td>
<td align="left">noDup</td>
<td align="right">421</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup537rando</td>
<td align="left">1Dup</td>
<td align="right">62</td>
</tr>
<tr class="odd">
<td align="left">OrthoGroupr537rando</td>
<td align="left">2+Dup</td>
<td align="right">54</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup536rando</td>
<td align="left">Total</td>
<td align="right">537</td>
</tr>
</tbody>
</table>
</div>
<div id="cosmic-vs-all" class="section level3">
<h3>COSMIC vs All</h3>
</div>
<div id="dups-vs-no-dups" class="section level3">
<h3>Dups vs no Dups</h3>
<pre class="r"><code>dup.table1 &lt;- data.frame(row.names = c(&quot;COSMIC&quot;,&quot;OrthoGroup&quot;), notDup = c(445,18106), dup = c(59+33, 2527+1812))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">notDup</th>
<th align="right">dup</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="right">445</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup</td>
<td align="right">18106</td>
<td align="right">4339</td>
</tr>
</tbody>
</table>
<pre class="r"><code>mosaicplot(dup.table1,
           main = &quot;COSMIC vs All&quot;,
           sub = &quot;Any duplications vs no duplications&quot;,
           color = TRUE
)</code></pre>
<p><img src="figure/OrthoFinder_dup.Rmd/COSMICvAll%20Mosaic%20Plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>test1 &lt;- fisher.test(dup.table1, alternative = &quot;g&quot;)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  dup.table1
p-value = 0.1099
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.9544619       Inf
sample estimates:
odds ratio 
  1.159141 </code></pre>
<div id="onedup-to-moredup" class="section level4">
<h4>oneDup to moreDup</h4>
<pre class="r"><code>dup.table1b &lt;- data.frame(row.names = c(&quot;COSMIC&quot;,&quot;OrthoGroup&quot;), oneDup = c(59,2527), moreDup = c(33, 1812))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">oneDup</th>
<th align="right">moreDup</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="right">59</td>
<td align="right">33</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup</td>
<td align="right">2527</td>
<td align="right">1812</td>
</tr>
</tbody>
</table>
<pre class="r"><code>mosaicplot(dup.table1b,
           main = &quot;COSMIC vs All&quot;,
           sub = &quot;One duplication vs many duplications&quot;,
           color = TRUE
)</code></pre>
<p><img src="figure/OrthoFinder_dup.Rmd/COSMICvAll%20Mosaic%20Plot%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>test1b &lt;- fisher.test(dup.table1b, alternative = &quot;g&quot;)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  dup.table1b
p-value = 0.152
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.8764278       Inf
sample estimates:
odds ratio 
  1.281938 </code></pre>
</div>
</div>
<div id="cosmic-vs-equal-size-subsample" class="section level3">
<h3>COSMIC vs equal-size subsample;</h3>
<div id="dup-vs-no-dups" class="section level4">
<h4>Dup vs No Dups</h4>
<pre class="r"><code>dup.table2 &lt;- data.frame(row.names = c(&quot;COSMIC&quot;,&quot;OrthoGroup_537&quot;), notDup = c(445,421), dup = c(59+33, 62+54))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">notDup</th>
<th align="right">dup</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="right">445</td>
<td align="right">92</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup_537</td>
<td align="right">421</td>
<td align="right">116</td>
</tr>
</tbody>
</table>
<pre class="r"><code>mosaicplot(dup.table2,
           main = &quot;COSMIC vs Subsample&quot;,
           sub = &quot;Any duplications vs no duplications&quot;,
           color = TRUE
)</code></pre>
<p><img src="figure/OrthoFinder_dup.Rmd/COSMICvSub%20Mosaic%20Plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>test2 &lt;- fisher.test(dup.table2, alternative = &quot;g&quot;)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  dup.table2
p-value = 0.03778
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 1.020602      Inf
sample estimates:
odds ratio 
  1.332382 </code></pre>
</div>
<div id="onedup-vs-manydup" class="section level4">
<h4>oneDup vs manyDup</h4>
<pre class="r"><code>dup.table2b &lt;- data.frame(row.names = c(&quot;COSMIC&quot;,&quot;OrthoGroup&quot;), oneDup = c(59,62), moreDup = c(33, 54))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">oneDup</th>
<th align="right">moreDup</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">COSMIC</td>
<td align="right">59</td>
<td align="right">33</td>
</tr>
<tr class="even">
<td align="left">OrthoGroup</td>
<td align="right">62</td>
<td align="right">54</td>
</tr>
</tbody>
</table>
<pre class="r"><code>mosaicplot(dup.table2b,
           main = &quot;COSMIC vs Subsample&quot;,
           sub = &quot;One duplication vs many duplications&quot;,
           color = TRUE
)</code></pre>
<p><img src="figure/OrthoFinder_dup.Rmd/COSMICvSub%20Mosaic%20Plot%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>test2b &lt;- fisher.test(dup.table2b, alternative = &quot;g&quot;)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  dup.table2b
p-value = 0.0791
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.9362507       Inf
sample estimates:
odds ratio 
  1.553831 </code></pre>
</div>
</div>
<div id="conclusion" class="section level3">
<h3>Conclusion</h3>
<p>As an overall rule, genes in COSMIC orthogroups are
<strong><em>less</em></strong> likely to have duplicated in our
<em>Myotis</em> species relative to all other orthogroups.</p>
<p>On the one hand, this shouldn’t be terribly surprising: bats have
non-allometric differences in lifespan, and so they shouldn’t have a
need for an expanded repetoire of TSGs. Furthermore, bat genomes are
small relative to other clades [@citeme], which in combination with our
overall gene counts for our species suggests that we shouldn’t expect
rampant gene duplication.</p>
<p>On the other hand, based on the multistage model of cancer, cancer
risk scales linearly with body size and to the sixth-power for lifespan;
thus in cases where there’s increases in lifespan, duplication should be
the fastest way to deal with it. However, this assumes that 1) there
weren’t already duplications or other optimizations that preempt the
increase in cancer risk in the common ancestor of long- and short-lived
myotis; and 2) that the short-lived Myotis really are that short-lived.
There is one thing that moves faster than duplication of genes: rewiring
of gene regulatory networks. It could very well be that the TEs are
changing about and driving things vs the genes.</p>
<p>Then there is the matter of the sensitivity of this analysis. Here, I
am lumping all COSMIC genes together, both oncogenes and TSGs. I’m also
lumping together all duplications everywhere. A more nuanced approach is
called for, that looks at TSGs and oncogenes separately; and that also
look at where in the phylogeny things change.</p>
</div>
</div>
<div id="to-do" class="section level2">
<h2>To Do</h2>
<ul>
<li>Look at what happens if you do this for another clade with a human
outgroup. Ideally, another group of 10 species in the same genus.</li>
<li>Break up COSMIC genes into pro- and anti-cancerous, and/or by pro-
and anti- cancer hallmark.</li>
<li>Redo this using the properly-segregated COSMIC orthogroups.</li>
<li>Refine the topology of duplications using NOTUNG and CAFE.</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] stargazer_5.2.3 lubridate_1.9.2 forcats_1.0.0   stringr_1.5.0  
 [5] dplyr_1.1.2     purrr_1.0.1     readr_2.1.4     tidyr_1.3.0    
 [9] tibble_3.2.1    ggplot2_3.4.2   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0 xfun_0.39        bslib_0.4.2      colorspace_2.1-0
 [5] vctrs_0.6.2      generics_0.1.3   htmltools_0.5.5  yaml_2.3.7      
 [9] utf8_1.2.3       rlang_1.1.0      jquerylib_0.1.4  later_1.3.0     
[13] pillar_1.9.0     glue_1.6.2       withr_2.5.0      lifecycle_1.0.3 
[17] munsell_0.5.0    gtable_0.3.3     workflowr_1.7.0  evaluate_0.20   
[21] knitr_1.42       tzdb_0.3.0       fastmap_1.1.1    httpuv_1.6.9    
[25] fansi_1.0.4      highr_0.10       Rcpp_1.0.10      promises_1.2.0.1
[29] scales_1.2.1     cachem_1.0.7     jsonlite_1.8.4   fs_1.6.2        
[33] hms_1.1.3        digest_0.6.31    stringi_1.7.12   grid_4.2.3      
[37] rprojroot_2.0.3  cli_3.6.1        tools_4.2.3      magrittr_2.0.3  
[41] sass_0.4.5       crayon_1.5.2     pkgconfig_2.0.3  timechange_0.2.0
[45] rmarkdown_2.21   rstudioapi_0.14  R6_2.5.1         git2r_0.32.0    
[49] compiler_4.2.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
